{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import Seq\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Change the following parameters to your output directory (where you want output files to be written), your metafile, and TNseq output. \n",
    "\n",
    "If these files aren't in the same directory as this notebook, you need use the full path. For example, the full path of 'barseq_output' would be '/usr2/people/clairedubin/barseq/barseq_output'\n",
    "\n",
    "Also change the control and experimental temperatures if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where the BarSeq FASTQs are\n",
    "FASTQ_directory = '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Barseq metafile\n",
    "\n",
    "You'll need to create your own metafile - easiest way is edit in Excel. A sample metafile is located at: /usr2/people/clairedubin/barseq/Kluyv_BarSeq_metadata.txt\n",
    "\n",
    "    Fastq: path to barseq FASTQ for each competition\n",
    "    SampleName: unique name for each competition, MUST include ctrl_temp or exp_temp as specified above\n",
    "    DualIndex: index from sequencing (in read name lines of FASTQ and the sequencing info file that Adam sends)\n",
    "    \n",
    "    UsePrecounted: True if you've already run the Barseq counting script on a competition, False otherwise\n",
    "    \n",
    "    poolfile: path to Tnseq output file, should be the same for each competition\n",
    "    output_dir: path to output directory\n",
    "    minRandom: number of bases between start of read and DualIndex, usually in sequence of 1,2,3,4,1,2,3,4 for each sample\n",
    "    maxRandom: same as minRandom\n",
    "    \n",
    "    BeforeBarcode: sequence preceding all barcodes\n",
    "    AfterBarcode: sequence following all barcodes\n",
    "    BarcodeLengths: lengths of barcodes to search for, with expected length of barcode first\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT085_S277_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT086_S278_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT087_S279_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT088_S280_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT089_S281_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT090_S282_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT091_S283_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT092_S284_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT093_S285_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT094_S286_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT095_S287_L003_R1_001.fastq\r\n",
      "/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT096_S288_L003_R1_001.fastq\r\n"
     ]
    }
   ],
   "source": [
    "!ls {FASTQ_directory}/*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format: sample_name, fastq_path, dual_index, offset\n",
    "#add or delete more lines as needed\n",
    "\n",
    "sample_info = [('28C_1', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT085_S277_L003_R1_001.fastq', 'GCCTTA', '1'),\n",
    "               ('28C_2', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT086_S278_L003_R1_001.fastq', 'GCTCCA', '2'),\n",
    "               ('28C_3', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT087_S279_L003_R1_001.fastq', 'GGCACA', '3'),\n",
    "               ('28C_4', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT088_S280_L003_R1_001.fastq', 'GGCCTG', '4'),\n",
    "               ('28C_5', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT089_S281_L003_R1_001.fastq', 'TCTACC', '1'),\n",
    "               ('28C_6', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT090_S282_L003_R1_001.fastq', 'TGAATG', '2'),\n",
    "               ('42C_1', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT091_S283_L003_R1_001.fastq', 'TGCCAT', '3'),\n",
    "               ('42C_2', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT092_S284_L003_R1_001.fastq', 'TGCTGG', '4'),\n",
    "               ('42C_3', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT093_S285_L003_R1_001.fastq', 'TGGCGC', '1'),\n",
    "               ('42C_4', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT094_S286_L003_R1_001.fastq', 'TTCGAA', '2'),\n",
    "               ('42C_5', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT095_S287_L003_R1_001.fastq', 'TTCTCC', '3'),\n",
    "               ('42C_6', '/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT096_S288_L003_R1_001.fastq', 'AGGTTT', '4'),\n",
    "              ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@K00364:269:HNCLMBBXY:3:1101:29488:1349 1:N:0:GCCTTA\r\n",
      "TNTTCCGGTCGACCTGCAGCGTACGCGTTCAGACCTTCCGTAATCAGAGAC\r\n",
      "+\r\n",
      "A#<<AFF<-A<AJJJAJJ<JF7AJ7FFF<AF--7<--FA-FJJFJ<<A-AF\r\n",
      "@K00364:269:HNCLMBBXY:3:1101:2077:1367 1:N:0:GCCTTA\r\n",
      "GATTCCGGTCGACCTGCAGCGTACGTGGGTCCGTATGGAAGTCAAAGAGAC\r\n",
      "+\r\n",
      "AAAAFJ-FAJJFJJFJJJJJJFJJ<JJF7-7F77AF--F<JF-FJJJJFFJ\r\n",
      "@K00364:269:HNCLMBBXY:3:1101:2808:1367 1:N:0:GCCTTA\r\n",
      "GATTCCGGTCGACCTGCAGCGTACGATGAGGGTGCTGAGTTGGCAAGAGAC\r\n"
     ]
    }
   ],
   "source": [
    "!head /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT085_S277_L003_R1_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if there are any samples where you already counted barcodes, add the sample names here\n",
    "precounted_samples = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these variables should be the same for each replicate\n",
    "\n",
    "metafile_name = 'BarSeq_metafile.txt'\n",
    "logfile_name = 'BarSeq_counting.log'\n",
    "poolfile = '30_to_41_pools_poolfile_combined_annotated.csv'\n",
    "output_dir = 'TNSeq_barseq_outputs'\n",
    "before_barcode = 'GTCGACCTGCAGCGTACG'\n",
    "after_barcode = 'AGAGACCTC'\n",
    "barcode_lengths = '20,19,18,21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['FileIndex','SampleName', 'Fastq', 'DualIndex', 'Poolfile', 'OutputDir', 'minRandom', 'maxRandom',\n",
    "                   'BeforeBarcode', 'AfterBarcode', 'BarcodeLengths', 'UsePrecounted']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for sample_name, fastq, dual_index, offset in sample_info:\n",
    "        \n",
    "        #DUAL INDECES ARE BACKWARDS FROM WHAT THE FLOWCELL OUTPUT SAYS\n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index[::-1], poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'False'])\n",
    "        f.write(to_write+'\\n')\n",
    "        \n",
    "    for precounted_sample_name in precounted_samples:\n",
    "        \n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index, poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'True'])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileIndex</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>DualIndex</th>\n",
       "      <th>Poolfile</th>\n",
       "      <th>OutputDir</th>\n",
       "      <th>minRandom</th>\n",
       "      <th>maxRandom</th>\n",
       "      <th>BeforeBarcode</th>\n",
       "      <th>AfterBarcode</th>\n",
       "      <th>BarcodeLengths</th>\n",
       "      <th>UsePrecounted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>28C_1</td>\n",
       "      <td>28C_1</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT085_S277_L003_R1_001.fastq</td>\n",
       "      <td>ATTCCG</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>28C_2</td>\n",
       "      <td>28C_2</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT086_S278_L003_R1_001.fastq</td>\n",
       "      <td>ACCTCG</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>28C_3</td>\n",
       "      <td>28C_3</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT087_S279_L003_R1_001.fastq</td>\n",
       "      <td>ACACGG</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>28C_4</td>\n",
       "      <td>28C_4</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT088_S280_L003_R1_001.fastq</td>\n",
       "      <td>GTCCGG</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>28C_5</td>\n",
       "      <td>28C_5</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT089_S281_L003_R1_001.fastq</td>\n",
       "      <td>CCATCT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>28C_6</td>\n",
       "      <td>28C_6</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT090_S282_L003_R1_001.fastq</td>\n",
       "      <td>GTAAGT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>42C_1</td>\n",
       "      <td>42C_1</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT091_S283_L003_R1_001.fastq</td>\n",
       "      <td>TACCGT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>42C_2</td>\n",
       "      <td>42C_2</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT092_S284_L003_R1_001.fastq</td>\n",
       "      <td>GGTCGT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>42C_3</td>\n",
       "      <td>42C_3</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT093_S285_L003_R1_001.fastq</td>\n",
       "      <td>CGCGGT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>42C_4</td>\n",
       "      <td>42C_4</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT094_S286_L003_R1_001.fastq</td>\n",
       "      <td>AAGCTT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>42C_5</td>\n",
       "      <td>42C_5</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT095_S287_L003_R1_001.fastq</td>\n",
       "      <td>CCTCTT</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>42C_6</td>\n",
       "      <td>42C_6</td>\n",
       "      <td>/usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT096_S288_L003_R1_001.fastq</td>\n",
       "      <td>TTTGGA</td>\n",
       "      <td>30_to_41_pools_poolfile_combined_annotated.csv</td>\n",
       "      <td>TNSeq_barseq_outputs</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FileIndex SampleName  \\\n",
       "0   28C_1     28C_1       \n",
       "1   28C_2     28C_2       \n",
       "2   28C_3     28C_3       \n",
       "3   28C_4     28C_4       \n",
       "4   28C_5     28C_5       \n",
       "5   28C_6     28C_6       \n",
       "6   42C_1     42C_1       \n",
       "7   42C_2     42C_2       \n",
       "8   42C_3     42C_3       \n",
       "9   42C_4     42C_4       \n",
       "10  42C_5     42C_5       \n",
       "11  42C_6     42C_6       \n",
       "\n",
       "                                                                                                        Fastq  \\\n",
       "0   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT085_S277_L003_R1_001.fastq   \n",
       "1   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT086_S278_L003_R1_001.fastq   \n",
       "2   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT087_S279_L003_R1_001.fastq   \n",
       "3   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT088_S280_L003_R1_001.fastq   \n",
       "4   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT089_S281_L003_R1_001.fastq   \n",
       "5   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT090_S282_L003_R1_001.fastq   \n",
       "6   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT091_S283_L003_R1_001.fastq   \n",
       "7   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT092_S284_L003_R1_001.fastq   \n",
       "8   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT093_S285_L003_R1_001.fastq   \n",
       "9   /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT094_S286_L003_R1_001.fastq   \n",
       "10  /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT095_S287_L003_R1_001.fastq   \n",
       "11  /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT096_S288_L003_R1_001.fastq   \n",
       "\n",
       "   DualIndex                                        Poolfile  \\\n",
       "0   ATTCCG    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "1   ACCTCG    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "2   ACACGG    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "3   GTCCGG    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "4   CCATCT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "5   GTAAGT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "6   TACCGT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "7   GGTCGT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "8   CGCGGT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "9   AAGCTT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "10  CCTCTT    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "11  TTTGGA    30_to_41_pools_poolfile_combined_annotated.csv   \n",
       "\n",
       "               OutputDir  minRandom  maxRandom       BeforeBarcode  \\\n",
       "0   TNSeq_barseq_outputs  1          1          GTCGACCTGCAGCGTACG   \n",
       "1   TNSeq_barseq_outputs  2          2          GTCGACCTGCAGCGTACG   \n",
       "2   TNSeq_barseq_outputs  3          3          GTCGACCTGCAGCGTACG   \n",
       "3   TNSeq_barseq_outputs  4          4          GTCGACCTGCAGCGTACG   \n",
       "4   TNSeq_barseq_outputs  1          1          GTCGACCTGCAGCGTACG   \n",
       "5   TNSeq_barseq_outputs  2          2          GTCGACCTGCAGCGTACG   \n",
       "6   TNSeq_barseq_outputs  3          3          GTCGACCTGCAGCGTACG   \n",
       "7   TNSeq_barseq_outputs  4          4          GTCGACCTGCAGCGTACG   \n",
       "8   TNSeq_barseq_outputs  1          1          GTCGACCTGCAGCGTACG   \n",
       "9   TNSeq_barseq_outputs  2          2          GTCGACCTGCAGCGTACG   \n",
       "10  TNSeq_barseq_outputs  3          3          GTCGACCTGCAGCGTACG   \n",
       "11  TNSeq_barseq_outputs  4          4          GTCGACCTGCAGCGTACG   \n",
       "\n",
       "   AfterBarcode BarcodeLengths  UsePrecounted  \n",
       "0   AGAGACCTC    20,19,18,21    False          \n",
       "1   AGAGACCTC    20,19,18,21    False          \n",
       "2   AGAGACCTC    20,19,18,21    False          \n",
       "3   AGAGACCTC    20,19,18,21    False          \n",
       "4   AGAGACCTC    20,19,18,21    False          \n",
       "5   AGAGACCTC    20,19,18,21    False          \n",
       "6   AGAGACCTC    20,19,18,21    False          \n",
       "7   AGAGACCTC    20,19,18,21    False          \n",
       "8   AGAGACCTC    20,19,18,21    False          \n",
       "9   AGAGACCTC    20,19,18,21    False          \n",
       "10  AGAGACCTC    20,19,18,21    False          \n",
       "11  AGAGACCTC    20,19,18,21    False          "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Barseq counting\n",
    "\n",
    "Might take a few hours!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2022-05-27 12:13:31 RBseq_Count_BarCodes.py\n",
      "2022-05-27 12:13:31 Version: 1.1.4\n",
      "2022-05-27 12:13:31 Release Date: July 16, 2020\n",
      "2022-05-27 12:13:31 Options passed:  metafile:BarSeq_metafile.txt  logFile:BarSeq_counting.log  minQual:10  matchBefore:6  matchAfter:6  quietMode:False \n",
      "2022-05-27 12:13:31 Logging status updates in BarSeq_counting.log\n",
      "2022-05-27 12:13:31 Loading TnSeq library metadata from BarSeq_metafile.txt\n",
      "2022-05-27 12:13:31 Setting output directory as: TNSeq_barseq_outputs (from first line of metadata file)\n",
      "2022-05-27 12:13:31 Loading mapped barcodes in mutant pool from: 30_to_41_pools_poolfile_combined_annotated.csv (from first line of metadata file)\n",
      "2022-05-27 12:13:33 Read 270652 barcodes from 30_to_41_pools_poolfile_combined_annotated.csv\n",
      "2022-05-27 12:13:33 Finding barcodes in fastqs and counting occurances\n",
      "2022-05-27 12:13:33 ---------------------\n",
      "2022-05-27 12:13:33   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT085_S277_L003_R1_001.fastq\n",
      "2022-05-27 12:13:33   Filtering reads on second index: ATTCCG\n",
      "2022-05-27 12:13:33   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:16:39     12303013 reads processed.\n",
      "2022-05-27 12:16:39     303378 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:16:39     11577275 reads with compliant barcodes.\n",
      "2022-05-27 12:16:39     725738 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:16:39       432809 reads without expected Dual Index.\n",
      "2022-05-27 12:16:39       177603 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:16:39       115306 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:16:39       1763 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:16:39       1837 reads with quality scores less than 10\n",
      "2022-05-27 12:16:39   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/28C_1.counts\n",
      "2022-05-27 12:16:40   Total barcodes seen (incudes sequencing errors): 480156\n",
      "2022-05-27 12:16:40   Matching barcodes to poolfile\n",
      "2022-05-27 12:16:40     Number of barcodes from poolfile seen: 23653\n",
      "2022-05-27 12:16:40     Reads with barcodes from poolfile: 450029\n",
      "2022-05-27 12:16:40   Most abundant barcode: TAGGTTAGATGAGTCGTTGT seen 6650 times.\n",
      "2022-05-27 12:16:40   Number of reads that differ from this barcode by one base pair (likely sequencing errors):157\n",
      "2022-05-27 12:16:40   Estimated sequencing error rate for barcodes: 2.3%\n",
      "2022-05-27 12:16:40   Barcodes seen once (highly inflated by sequencing errors): 201761\n",
      "2022-05-27 12:16:40   Barcodes seen twice (slightly inflated by sequencing errors): 9375\n",
      "2022-05-27 12:16:40   Barcodes seen three times or more: 269020\n",
      "2022-05-27 12:16:40   Chao estimate of population size (ones^2/2*twos): 2200000\n",
      "2022-05-27 12:16:40 ---------------------\n",
      "2022-05-27 12:16:40   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT086_S278_L003_R1_001.fastq\n",
      "2022-05-27 12:16:40   Filtering reads on second index: ACCTCG\n",
      "2022-05-27 12:16:40   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:20:03     13482660 reads processed.\n",
      "2022-05-27 12:20:03     13129113 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:20:03     12756820 reads with compliant barcodes.\n",
      "2022-05-27 12:20:03     725840 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:20:03       471389 reads without expected Dual Index.\n",
      "2022-05-27 12:20:03       172849 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:20:03       81576 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:20:03       1999 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:20:03       2143 reads with quality scores less than 10\n",
      "2022-05-27 12:20:03   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/28C_2.counts\n",
      "2022-05-27 12:20:04   Total barcodes seen (incudes sequencing errors): 550753\n",
      "2022-05-27 12:20:04   Matching barcodes to poolfile\n",
      "2022-05-27 12:20:05     Number of barcodes from poolfile seen: 31075\n",
      "2022-05-27 12:20:05     Reads with barcodes from poolfile: 496753\n",
      "2022-05-27 12:20:05   Most abundant barcode: ATTATGCCTGGTGTGGTGTC seen 2198 times.\n",
      "2022-05-27 12:20:05   Number of reads that differ from this barcode by one base pair (likely sequencing errors):32\n",
      "2022-05-27 12:20:05   Estimated sequencing error rate for barcodes: 1.4%\n",
      "2022-05-27 12:20:05   Barcodes seen once (highly inflated by sequencing errors): 235325\n",
      "2022-05-27 12:20:05   Barcodes seen twice (slightly inflated by sequencing errors): 15592\n",
      "2022-05-27 12:20:05   Barcodes seen three times or more: 299836\n",
      "2022-05-27 12:20:05   Chao estimate of population size (ones^2/2*twos): 1800000\n",
      "2022-05-27 12:20:05 ---------------------\n",
      "2022-05-27 12:20:05   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT087_S279_L003_R1_001.fastq\n",
      "2022-05-27 12:20:05   Filtering reads on second index: ACACGG\n",
      "2022-05-27 12:20:05   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:22:54     11312373 reads processed.\n",
      "2022-05-27 12:22:54     11132414 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:22:54     10668294 reads with compliant barcodes.\n",
      "2022-05-27 12:22:54     644079 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:22:54       455340 reads without expected Dual Index.\n",
      "2022-05-27 12:22:54       154164 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:22:54       34564 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:22:54       500 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:22:54       516 reads with quality scores less than 10\n",
      "2022-05-27 12:22:54   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/28C_3.counts\n",
      "2022-05-27 12:22:55   Total barcodes seen (incudes sequencing errors): 463486\n",
      "2022-05-27 12:22:55   Matching barcodes to poolfile\n",
      "2022-05-27 12:22:55     Number of barcodes from poolfile seen: 17894\n",
      "2022-05-27 12:22:55     Reads with barcodes from poolfile: 408840\n",
      "2022-05-27 12:22:55   Most abundant barcode: CTATGCGCCGAGGTTGCTTA seen 84379 times.\n",
      "2022-05-27 12:22:55   Number of reads that differ from this barcode by one base pair (likely sequencing errors):1725\n",
      "2022-05-27 12:22:55   Estimated sequencing error rate for barcodes: 2.0%\n",
      "2022-05-27 12:22:55   Barcodes seen once (highly inflated by sequencing errors): 213583\n",
      "2022-05-27 12:22:55   Barcodes seen twice (slightly inflated by sequencing errors): 11142\n",
      "2022-05-27 12:22:55   Barcodes seen three times or more: 238761\n",
      "2022-05-27 12:22:55   Chao estimate of population size (ones^2/2*twos): 2000000\n",
      "2022-05-27 12:22:55 ---------------------\n",
      "2022-05-27 12:22:55   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT088_S280_L003_R1_001.fastq\n",
      "2022-05-27 12:22:55   Filtering reads on second index: GTCCGG\n",
      "2022-05-27 12:22:55   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:25:16     9355649 reads processed.\n",
      "2022-05-27 12:25:16     9375455 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:25:16     8837366 reads with compliant barcodes.\n",
      "2022-05-27 12:25:16     518283 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:25:16       366023 reads without expected Dual Index.\n",
      "2022-05-27 12:25:16       127651 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:25:16       24605 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:25:16       334 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:25:16       336 reads with quality scores less than 10\n",
      "2022-05-27 12:25:16   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/28C_4.counts\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2022-05-27 12:25:17   Total barcodes seen (incudes sequencing errors): 363194\n",
      "2022-05-27 12:25:17   Matching barcodes to poolfile\n",
      "2022-05-27 12:25:17     Number of barcodes from poolfile seen: 12082\n",
      "2022-05-27 12:25:17     Reads with barcodes from poolfile: 343329\n",
      "2022-05-27 12:25:17   Most abundant barcode: ATTATGCCTGGTGTGGTGTC seen 1409 times.\n",
      "2022-05-27 12:25:17   Number of reads that differ from this barcode by one base pair (likely sequencing errors):30\n",
      "2022-05-27 12:25:17   Estimated sequencing error rate for barcodes: 2.1%\n",
      "2022-05-27 12:25:17   Barcodes seen once (highly inflated by sequencing errors): 162757\n",
      "2022-05-27 12:25:17   Barcodes seen twice (slightly inflated by sequencing errors): 7985\n",
      "2022-05-27 12:25:17   Barcodes seen three times or more: 192452\n",
      "2022-05-27 12:25:17   Chao estimate of population size (ones^2/2*twos): 1700000\n",
      "2022-05-27 12:25:17 ---------------------\n",
      "2022-05-27 12:25:17   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT089_S281_L003_R1_001.fastq\n",
      "2022-05-27 12:25:17   Filtering reads on second index: CCATCT\n",
      "2022-05-27 12:25:17   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:28:42     13709335 reads processed.\n",
      "2022-05-27 12:28:42     350993 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:28:42     12937662 reads with compliant barcodes.\n",
      "2022-05-27 12:28:42     771673 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:28:42       441343 reads without expected Dual Index.\n",
      "2022-05-27 12:28:42       192464 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:28:42       137825 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:28:42       2221 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:28:42       2315 reads with quality scores less than 10\n",
      "2022-05-27 12:28:42   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/28C_5.counts\n",
      "2022-05-27 12:28:43   Total barcodes seen (incudes sequencing errors): 634327\n",
      "2022-05-27 12:28:43   Matching barcodes to poolfile\n",
      "2022-05-27 12:28:44     Number of barcodes from poolfile seen: 54533\n",
      "2022-05-27 12:28:44     Reads with barcodes from poolfile: 506652\n",
      "2022-05-27 12:28:44   Most abundant barcode: ATTATGCCTGGTGTGGTGTC seen 2089 times.\n",
      "2022-05-27 12:28:44   Number of reads that differ from this barcode by one base pair (likely sequencing errors):44\n",
      "2022-05-27 12:28:44   Estimated sequencing error rate for barcodes: 2.1%\n",
      "2022-05-27 12:28:44   Barcodes seen once (highly inflated by sequencing errors): 264293\n",
      "2022-05-27 12:28:44   Barcodes seen twice (slightly inflated by sequencing errors): 35455\n",
      "2022-05-27 12:28:44   Barcodes seen three times or more: 334579\n",
      "2022-05-27 12:28:44   Chao estimate of population size (ones^2/2*twos): 990000\n",
      "2022-05-27 12:28:44 ---------------------\n",
      "2022-05-27 12:28:44   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT090_S282_L003_R1_001.fastq\n",
      "2022-05-27 12:28:44   Filtering reads on second index: GTAAGT\n",
      "2022-05-27 12:28:44   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:31:16     10091838 reads processed.\n",
      "2022-05-27 12:31:16     9804300 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:31:16     9540116 reads with compliant barcodes.\n",
      "2022-05-27 12:31:16     551722 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:31:16       364859 reads without expected Dual Index.\n",
      "2022-05-27 12:31:16       132342 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:31:16       54502 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:31:16       1216 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:31:16       1323 reads with quality scores less than 10\n",
      "2022-05-27 12:31:16   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/28C_6.counts\n",
      "2022-05-27 12:31:16   Total barcodes seen (incudes sequencing errors): 406222\n",
      "2022-05-27 12:31:16   Matching barcodes to poolfile\n",
      "2022-05-27 12:31:17     Number of barcodes from poolfile seen: 18492\n",
      "2022-05-27 12:31:17     Reads with barcodes from poolfile: 370177\n",
      "2022-05-27 12:31:17   Most abundant barcode: TTACCGGGATATCAAGAAGG seen 1542 times.\n",
      "2022-05-27 12:31:17   Number of reads that differ from this barcode by one base pair (likely sequencing errors):21\n",
      "2022-05-27 12:31:17   Estimated sequencing error rate for barcodes: 1.3%\n",
      "2022-05-27 12:31:17   Barcodes seen once (highly inflated by sequencing errors): 159251\n",
      "2022-05-27 12:31:17   Barcodes seen twice (slightly inflated by sequencing errors): 7385\n",
      "2022-05-27 12:31:17   Barcodes seen three times or more: 239586\n",
      "2022-05-27 12:31:17   Chao estimate of population size (ones^2/2*twos): 1700000\n",
      "2022-05-27 12:31:17 ---------------------\n",
      "2022-05-27 12:31:17   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT091_S283_L003_R1_001.fastq\n",
      "2022-05-27 12:31:17   Filtering reads on second index: TACCGT\n",
      "2022-05-27 12:31:17   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:32:22     4482747 reads processed.\n",
      "2022-05-27 12:32:22     4354144 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:32:22     4186725 reads with compliant barcodes.\n",
      "2022-05-27 12:32:22     296022 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:32:22       248668 reads without expected Dual Index.\n",
      "2022-05-27 12:32:22       38101 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:32:22       9249 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:32:22       164 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:32:22       165 reads with quality scores less than 10\n",
      "2022-05-27 12:32:22   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/42C_1.counts\n",
      "2022-05-27 12:32:23   Total barcodes seen (incudes sequencing errors): 348696\n",
      "2022-05-27 12:32:23   Matching barcodes to poolfile\n",
      "2022-05-27 12:32:23     Number of barcodes from poolfile seen: 23375\n",
      "2022-05-27 12:32:23     Reads with barcodes from poolfile: 161218\n",
      "2022-05-27 12:32:23   Most abundant barcode: GTGAGCCCCAAAGGCAAGCA seen 1654 times.\n",
      "2022-05-27 12:32:23   Number of reads that differ from this barcode by one base pair (likely sequencing errors):40\n",
      "2022-05-27 12:32:23   Estimated sequencing error rate for barcodes: 2.4%\n",
      "2022-05-27 12:32:23   Barcodes seen once (highly inflated by sequencing errors): 100203\n",
      "2022-05-27 12:32:23   Barcodes seen twice (slightly inflated by sequencing errors): 26993\n",
      "2022-05-27 12:32:23   Barcodes seen three times or more: 221500\n",
      "2022-05-27 12:32:23   Chao estimate of population size (ones^2/2*twos): 190000\n",
      "2022-05-27 12:32:23 ---------------------\n",
      "2022-05-27 12:32:23   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT092_S284_L003_R1_001.fastq\n",
      "2022-05-27 12:32:23   Filtering reads on second index: GGTCGT\n",
      "2022-05-27 12:32:23   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:33:53     6006801 reads processed.\n",
      "2022-05-27 12:33:53     6018393 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:33:53     5695628 reads with compliant barcodes.\n",
      "2022-05-27 12:33:53     311173 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:33:53       243862 reads without expected Dual Index.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2022-05-27 12:33:53       55164 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:33:53       12143 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:33:53       183 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:33:53       187 reads with quality scores less than 10\n",
      "2022-05-27 12:33:53   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/42C_2.counts\n",
      "2022-05-27 12:33:54   Total barcodes seen (incudes sequencing errors): 424325\n",
      "2022-05-27 12:33:54   Matching barcodes to poolfile\n",
      "2022-05-27 12:33:54     Number of barcodes from poolfile seen: 31819\n",
      "2022-05-27 12:33:54     Reads with barcodes from poolfile: 219540\n",
      "2022-05-27 12:33:54   Most abundant barcode: CTATGCGCCGAGGTTGCTTA seen 1337 times.\n",
      "2022-05-27 12:33:54   Number of reads that differ from this barcode by one base pair (likely sequencing errors):27\n",
      "2022-05-27 12:33:54   Estimated sequencing error rate for barcodes: 2.0%\n",
      "2022-05-27 12:33:54   Barcodes seen once (highly inflated by sequencing errors): 137266\n",
      "2022-05-27 12:33:54   Barcodes seen twice (slightly inflated by sequencing errors): 32626\n",
      "2022-05-27 12:33:54   Barcodes seen three times or more: 254433\n",
      "2022-05-27 12:33:54   Chao estimate of population size (ones^2/2*twos): 290000\n",
      "2022-05-27 12:33:54 ---------------------\n",
      "2022-05-27 12:33:54   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT093_S285_L003_R1_001.fastq\n",
      "2022-05-27 12:33:54   Filtering reads on second index: CGCGGT\n",
      "2022-05-27 12:33:54   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:34:55     4107570 reads processed.\n",
      "2022-05-27 12:34:55     94228 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:34:55     3819910 reads with compliant barcodes.\n",
      "2022-05-27 12:34:55     287660 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:34:55       212689 reads without expected Dual Index.\n",
      "2022-05-27 12:34:55       41366 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:34:55       33601 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:34:55       526 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:34:55       537 reads with quality scores less than 10\n",
      "2022-05-27 12:34:55   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/42C_3.counts\n",
      "2022-05-27 12:34:55   Total barcodes seen (incudes sequencing errors): 326699\n",
      "2022-05-27 12:34:55   Matching barcodes to poolfile\n",
      "2022-05-27 12:34:55     Number of barcodes from poolfile seen: 22010\n",
      "2022-05-27 12:34:55     Reads with barcodes from poolfile: 148770\n",
      "2022-05-27 12:34:55   Most abundant barcode: CTATGCGCCGAGGTTGCTTA seen 3162 times.\n",
      "2022-05-27 12:34:55   Number of reads that differ from this barcode by one base pair (likely sequencing errors):63\n",
      "2022-05-27 12:34:55   Estimated sequencing error rate for barcodes: 2.0%\n",
      "2022-05-27 12:34:55   Barcodes seen once (highly inflated by sequencing errors): 87139\n",
      "2022-05-27 12:34:55   Barcodes seen twice (slightly inflated by sequencing errors): 25492\n",
      "2022-05-27 12:34:55   Barcodes seen three times or more: 214068\n",
      "2022-05-27 12:34:55   Chao estimate of population size (ones^2/2*twos): 150000\n",
      "2022-05-27 12:34:55 ---------------------\n",
      "2022-05-27 12:34:55   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT094_S286_L003_R1_001.fastq\n",
      "2022-05-27 12:34:55   Filtering reads on second index: AAGCTT\n",
      "2022-05-27 12:34:55   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:35:32     2535205 reads processed.\n",
      "2022-05-27 12:35:32     2378840 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:35:32     2319051 reads with compliant barcodes.\n",
      "2022-05-27 12:35:32     216154 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:35:32       183827 reads without expected Dual Index.\n",
      "2022-05-27 12:35:32       20709 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:35:32       11608 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:35:32       359 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:35:32       395 reads with quality scores less than 10\n",
      "2022-05-27 12:35:32   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/42C_4.counts\n",
      "2022-05-27 12:35:32   Total barcodes seen (incudes sequencing errors): 207464\n",
      "2022-05-27 12:35:32   Matching barcodes to poolfile\n",
      "2022-05-27 12:35:33     Number of barcodes from poolfile seen: 9320\n",
      "2022-05-27 12:35:33     Reads with barcodes from poolfile: 88910\n",
      "2022-05-27 12:35:33   Most abundant barcode: GTGAGCCCCAAAGGCAAGCA seen 2301 times.\n",
      "2022-05-27 12:35:33   Number of reads that differ from this barcode by one base pair (likely sequencing errors):40\n",
      "2022-05-27 12:35:33   Estimated sequencing error rate for barcodes: 1.7%\n",
      "2022-05-27 12:35:33   Barcodes seen once (highly inflated by sequencing errors): 47484\n",
      "2022-05-27 12:35:33   Barcodes seen twice (slightly inflated by sequencing errors): 12248\n",
      "2022-05-27 12:35:33   Barcodes seen three times or more: 147732\n",
      "2022-05-27 12:35:33   Chao estimate of population size (ones^2/2*twos): 92000\n",
      "2022-05-27 12:35:33 ---------------------\n",
      "2022-05-27 12:35:33   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT095_S287_L003_R1_001.fastq\n",
      "2022-05-27 12:35:33   Filtering reads on second index: CCTCTT\n",
      "2022-05-27 12:35:33   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:35:53     1457202 reads processed.\n",
      "2022-05-27 12:35:53     1289337 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:35:53     1239870 reads with compliant barcodes.\n",
      "2022-05-27 12:35:53     217332 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:35:53       203957 reads without expected Dual Index.\n",
      "2022-05-27 12:35:53       10825 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:35:53       2550 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:35:53       44 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:35:53       44 reads with quality scores less than 10\n",
      "2022-05-27 12:35:53   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/42C_5.counts\n",
      "2022-05-27 12:35:53   Total barcodes seen (incudes sequencing errors): 142041\n",
      "2022-05-27 12:35:53   Matching barcodes to poolfile\n",
      "2022-05-27 12:35:53     Number of barcodes from poolfile seen: 5603\n",
      "2022-05-27 12:35:53     Reads with barcodes from poolfile: 47167\n",
      "2022-05-27 12:35:53   Most abundant barcode: TAGGTTAGATGAGTCGTTGT seen 1035 times.\n",
      "2022-05-27 12:35:53   Number of reads that differ from this barcode by one base pair (likely sequencing errors):24\n",
      "2022-05-27 12:35:53   Estimated sequencing error rate for barcodes: 2.3%\n",
      "2022-05-27 12:35:53   Barcodes seen once (highly inflated by sequencing errors): 30973\n",
      "2022-05-27 12:35:53   Barcodes seen twice (slightly inflated by sequencing errors): 10732\n",
      "2022-05-27 12:35:53   Barcodes seen three times or more: 100336\n",
      "2022-05-27 12:35:53   Chao estimate of population size (ones^2/2*twos): 45000\n",
      "2022-05-27 12:35:53 ---------------------\n",
      "2022-05-27 12:35:53   Mapping reads from /usr2/people/calabel1/NovaSeq_Cinco_de_Mayo/BarSeq_Cinco_de_Mayo/FEBA_BS_365_IT096_S288_L003_R1_001.fastq\n",
      "2022-05-27 12:35:53   Filtering reads on second index: TTTGGA\n",
      "2022-05-27 12:35:53   Looking for sequence barcode preceeded by CGTACG\n",
      "2022-05-27 12:38:57     12167850 reads processed.\n",
      "2022-05-27 12:38:57     12152836 reads were too short to check 6 bases after barcodes as specified with --matchAfter.  For these reads all remaining bases will be checked against the expected sequence. Reads with less than two remaining bases after the barcode will be counted as non-complaint.\n",
      "2022-05-27 12:38:57     11479288 reads with compliant barcodes.\n",
      "2022-05-27 12:38:57     688562 reads without recognizable, compliant barcodes. Of those:\n",
      "2022-05-27 12:38:57       511481 reads without expected Dual Index.\n",
      "2022-05-27 12:38:57       149663 reads without expected sequence before the barcode region.\n",
      "2022-05-27 12:38:57       27410 reads without expeced sequence after the barcode region.\n",
      "2022-05-27 12:38:57       358 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2022-05-27 12:38:57       344 reads with quality scores less than 10\n",
      "2022-05-27 12:38:57   Saving barcode counts to TNSeq_barseq_outputs/countsFiles/42C_6.counts\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2022-05-27 12:38:58   Total barcodes seen (incudes sequencing errors): 573962\n",
      "2022-05-27 12:38:58   Matching barcodes to poolfile\n",
      "2022-05-27 12:38:58     Number of barcodes from poolfile seen: 51530\n",
      "2022-05-27 12:38:58     Reads with barcodes from poolfile: 447034\n",
      "2022-05-27 12:38:58   Most abundant barcode: GTGAGCCCCAAAGGCAAGCA seen 2685 times.\n",
      "2022-05-27 12:38:58   Number of reads that differ from this barcode by one base pair (likely sequencing errors):41\n",
      "2022-05-27 12:38:58   Estimated sequencing error rate for barcodes: 1.5%\n",
      "2022-05-27 12:38:58   Barcodes seen once (highly inflated by sequencing errors): 215536\n",
      "2022-05-27 12:38:58   Barcodes seen twice (slightly inflated by sequencing errors): 37525\n",
      "2022-05-27 12:38:58   Barcodes seen three times or more: 320901\n",
      "2022-05-27 12:38:58   Chao estimate of population size (ones^2/2*twos): 620000\n",
      "2022-05-27 12:38:58 ---------------------\n",
      "2022-05-27 12:38:58 Saving summary statistics for fastqs to: TNSeq_barseq_outputs/fastqSummaryStats.txt\n",
      "2022-05-27 12:38:58 Processed 12 sequence files corresponding to 12 biological samples.\n",
      "2022-05-27 12:38:58 Counted 3688419 total reads for 86856 mapped barcodes from the mutant pool.\n",
      "2022-05-27 12:38:58 Saving poolcount file to: TNSeq_barseq_outputs/poolCount.txt\n"
     ]
    }
   ],
   "source": [
    "!python3 /usr2/people/clairedubin/barseq/latest_pipeline/RBseq_Count_BarCodes_v2.py -m {metafile_name} -l {logfile_name}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
