{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Modified from https://github.com/melanieabrams-pub/RH-seq_with_barcoding/blob/main/annotate_poolfile_for_barseq_v1.2_JSkerker.ipynb and \n",
    "https://github.com/melanieabrams-pub/kluyv_RH-seq_with_barcoding/blob/main/annotate_poolfile_kluyv.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load GFF annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#you don't need to change the path below - this is the most recent annotation file (5/12/22)\n",
    "gff_file = '/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar_CDS_orthologs_only.gff'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff = pd.read_csv(gff_file, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1' 'chr2' 'chr3' 'chr4' 'chr5' 'chr6' 'chr10' 'chr11' 'chr12' 'chr13'\n",
      " 'chr14' 'chr15' 'chr23' 'chr24' 'chr32' 'chr7' 'chr8' 'chr9']\n"
     ]
    }
   ],
   "source": [
    "t = gff['scaffold'].unique()\n",
    "print(t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>22024</td>\n",
       "      <td>25698</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KLLA0A00286g\"; transcript_id \"KLLA0A00286g\"</td>\n",
       "      <td>klYJL074C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>29717</td>\n",
       "      <td>32755</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KLLA0A00396g\"; transcript_id \"KLLA0A00396g\"</td>\n",
       "      <td>klYIL156W;YKR098C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>33084</td>\n",
       "      <td>35012</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KLLA0A00418g\"; transcript_id \"KLLA0A00418g\"</td>\n",
       "      <td>klYIL155C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>37134</td>\n",
       "      <td>38201</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KLLA0A00440g\"; transcript_id \"KLLA0A00440g\"</td>\n",
       "      <td>klYIL154C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>38766</td>\n",
       "      <td>39827</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KLLA0A00462g\"; transcript_id \"KLLA0A00462g\"</td>\n",
       "      <td>klYIL153W</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  scaffold     source type  start   stop strand  \\\n",
       "0  chr1     YGOB/YGAP  CDS  22024  25698  -       \n",
       "1  chr1     YGOB/YGAP  CDS  29717  32755  +       \n",
       "2  chr1     YGOB/YGAP  CDS  33084  35012  -       \n",
       "3  chr1     YGOB/YGAP  CDS  37134  38201  -       \n",
       "4  chr1     YGOB/YGAP  CDS  38766  39827  +       \n",
       "\n",
       "                                                   info                 ID  \n",
       "0  gene_id \"KLLA0A00286g\"; transcript_id \"KLLA0A00286g\"  klYJL074C          \n",
       "1  gene_id \"KLLA0A00396g\"; transcript_id \"KLLA0A00396g\"  klYIL156W;YKR098C  \n",
       "2  gene_id \"KLLA0A00418g\"; transcript_id \"KLLA0A00418g\"  klYIL155C          \n",
       "3  gene_id \"KLLA0A00440g\"; transcript_id \"KLLA0A00440g\"  klYIL154C          \n",
       "4  gene_id \"KLLA0A00462g\"; transcript_id \"KLLA0A00462g\"  klYIL153W          "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>22024</td>\n",
       "      <td>25698</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KLLA0A00286g\"; transcript_id \"KLLA0A00286g\"</td>\n",
       "      <td>klYJL074C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>29717</td>\n",
       "      <td>32755</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KLLA0A00396g\"; transcript_id \"KLLA0A00396g\"</td>\n",
       "      <td>klYIL156W;YKR098C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>33084</td>\n",
       "      <td>35012</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KLLA0A00418g\"; transcript_id \"KLLA0A00418g\"</td>\n",
       "      <td>klYIL155C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>37134</td>\n",
       "      <td>38201</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KLLA0A00440g\"; transcript_id \"KLLA0A00440g\"</td>\n",
       "      <td>klYIL154C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>38766</td>\n",
       "      <td>39827</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KLLA0A00462g\"; transcript_id \"KLLA0A00462g\"</td>\n",
       "      <td>klYIL153W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8960</th>\n",
       "      <td>chr9</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1534846</td>\n",
       "      <td>1535949</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KM020C07240\"; transcript_id \"KM020C07240\"</td>\n",
       "      <td>kmYIL153W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8961</th>\n",
       "      <td>chr9</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1536581</td>\n",
       "      <td>1537672</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KM020C07250\"; transcript_id \"KM020C07250\"</td>\n",
       "      <td>kmYIL154C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8962</th>\n",
       "      <td>chr9</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1539622</td>\n",
       "      <td>1541550</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KM020C07260\"; transcript_id \"KM020C07260\"</td>\n",
       "      <td>kmYIL155C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8963</th>\n",
       "      <td>chr9</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1541776</td>\n",
       "      <td>1544832</td>\n",
       "      <td>-</td>\n",
       "      <td>gene_id \"KM020C07270\"; transcript_id \"KM020C07270\"</td>\n",
       "      <td>kmYIL156W;YKR098C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8964</th>\n",
       "      <td>chr9</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1548740</td>\n",
       "      <td>1552414</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KM020C07300\"; transcript_id \"KM020C07300\"</td>\n",
       "      <td>kmYJL074C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8965 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     scaffold     source type    start     stop strand  \\\n",
       "0     chr1     YGOB/YGAP  CDS  22024    25698    -       \n",
       "1     chr1     YGOB/YGAP  CDS  29717    32755    +       \n",
       "2     chr1     YGOB/YGAP  CDS  33084    35012    -       \n",
       "3     chr1     YGOB/YGAP  CDS  37134    38201    -       \n",
       "4     chr1     YGOB/YGAP  CDS  38766    39827    +       \n",
       "...    ...           ...  ...    ...      ...   ..       \n",
       "8960  chr9     YGOB/YGAP  CDS  1534846  1535949  -       \n",
       "8961  chr9     YGOB/YGAP  CDS  1536581  1537672  +       \n",
       "8962  chr9     YGOB/YGAP  CDS  1539622  1541550  +       \n",
       "8963  chr9     YGOB/YGAP  CDS  1541776  1544832  -       \n",
       "8964  chr9     YGOB/YGAP  CDS  1548740  1552414  +       \n",
       "\n",
       "                                                      info                 ID  \n",
       "0     gene_id \"KLLA0A00286g\"; transcript_id \"KLLA0A00286g\"  klYJL074C          \n",
       "1     gene_id \"KLLA0A00396g\"; transcript_id \"KLLA0A00396g\"  klYIL156W;YKR098C  \n",
       "2     gene_id \"KLLA0A00418g\"; transcript_id \"KLLA0A00418g\"  klYIL155C          \n",
       "3     gene_id \"KLLA0A00440g\"; transcript_id \"KLLA0A00440g\"  klYIL154C          \n",
       "4     gene_id \"KLLA0A00462g\"; transcript_id \"KLLA0A00462g\"  klYIL153W          \n",
       "...                                                    ...        ...          \n",
       "8960  gene_id \"KM020C07240\"; transcript_id \"KM020C07240\"    kmYIL153W          \n",
       "8961  gene_id \"KM020C07250\"; transcript_id \"KM020C07250\"    kmYIL154C          \n",
       "8962  gene_id \"KM020C07260\"; transcript_id \"KM020C07260\"    kmYIL155C          \n",
       "8963  gene_id \"KM020C07270\"; transcript_id \"KM020C07270\"    kmYIL156W;YKR098C  \n",
       "8964  gene_id \"KM020C07300\"; transcript_id \"KM020C07300\"    kmYJL074C          \n",
       "\n",
       "[8965 rows x 8 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Match barcode insert locations to gene annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/auto/sahara/namib/home/kayleeec/rhseq\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE IF NEEDED: load COMBINED poolfile generated in TNseq mapping step\n",
    "\n",
    "poolfile = '/usr2/people/kayleeec/rhseq/Test_TNSeq/Test_Output_formatting/POOL30_poolfile'\n",
    "poolfile = 'Test_TNSeq/Test_Output_formatting/POOL30_poolfile'\n",
    "\n",
    "\n",
    "#CHANGE IF NEEDED: annotated pool file path to output\n",
    "\n",
    "annotated_poolfile_path ='/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/kaylee_file_annotated.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '/usr2/people/kayleeec/rhseq/Test_TNSeq/Test_Output_formatting/POOL30_poolfile'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[0;32mIn [18]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpoolfile\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      2\u001b[0m df\u001b[38;5;241m.\u001b[39mhead()\n",
      "File \u001b[0;32m~/.conda/envs/jupyter/lib/python3.10/site-packages/pandas/io/parsers/readers.py:912\u001b[0m, in \u001b[0;36mread_csv\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[1;32m    899\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[1;32m    900\u001b[0m     dialect,\n\u001b[1;32m    901\u001b[0m     delimiter,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    908\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[1;32m    909\u001b[0m )\n\u001b[1;32m    910\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[0;32m--> 912\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.conda/envs/jupyter/lib/python3.10/site-packages/pandas/io/parsers/readers.py:577\u001b[0m, in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    574\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[1;32m    576\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[0;32m--> 577\u001b[0m parser \u001b[38;5;241m=\u001b[39m \u001b[43mTextFileReader\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    579\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[1;32m    580\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[0;32m~/.conda/envs/jupyter/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1407\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m   1404\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[1;32m   1406\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1407\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.conda/envs/jupyter/lib/python3.10/site-packages/pandas/io/parsers/readers.py:1661\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[0;34m(self, f, engine)\u001b[0m\n\u001b[1;32m   1659\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[1;32m   1660\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m-> 1661\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1662\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1663\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1664\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1665\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1666\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1667\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1668\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1669\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1670\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1671\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m   1672\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[0;32m~/.conda/envs/jupyter/lib/python3.10/site-packages/pandas/io/common.py:859\u001b[0m, in \u001b[0;36mget_handle\u001b[0;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[1;32m    854\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    855\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[1;32m    856\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[1;32m    857\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[1;32m    858\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[0;32m--> 859\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[1;32m    860\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    861\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    862\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    863\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    864\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m    865\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    866\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    867\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[1;32m    868\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/usr2/people/kayleeec/rhseq/Test_TNSeq/Test_Output_formatting/POOL30_poolfile'"
     ]
    }
   ],
   "source": [
    "df = pd.read_csv(poolfile)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that each barcode in the dataframe is unique (the two numbers below should be the same):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of insertions in poolfile: 2406\n"
     ]
    }
   ],
   "source": [
    "print('Number of insertions in poolfile: {}'.format(df.shape[0]))\n",
    "#print('Number of unique barcodes in poolfile: {}'.format(len(df['barcode'].unique())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.set_index('transposonLocation')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['rcbarcode', 'nTot', 'n', 'scaffold', 'strand', 'pos', 'type',\n",
       "       'nMainLocation', 'nInsert', 'All genomic mappings',\n",
       "       'All insert mappings'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "      <th>All insert mappings</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transposonLocation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr9:937375:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr9</td>\n",
       "      <td>+</td>\n",
       "      <td>937375</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr9:937375:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2:338609:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr2</td>\n",
       "      <td>+</td>\n",
       "      <td>338609</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr2:338609:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4:413026:+</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>+</td>\n",
       "      <td>413026</td>\n",
       "      <td>Single</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr4:413026:+', 2)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:596883:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr14</td>\n",
       "      <td>+</td>\n",
       "      <td>596883</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr14:596883:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:899947:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>+</td>\n",
       "      <td>899947</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr13:899947:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    nTot  n scaffold strand     pos    type  nMainLocation  \\\n",
       "transposonLocation                                                           \n",
       "chr9:937375:+       1     1  chr9     +      937375  Single  1               \n",
       "chr2:338609:+       1     1  chr2     +      338609  Single  1               \n",
       "chr4:413026:+       2     2  chr4     +      413026  Single  2               \n",
       "chr14:596883:+      1     1  chr14    +      596883  Single  1               \n",
       "chr13:899947:+      1     1  chr13    +      899947  Single  1               \n",
       "\n",
       "                    nInsert     All genomic mappings All insert mappings  \n",
       "transposonLocation                                                        \n",
       "chr9:937375:+       0        [('chr9:937375:+', 1)]   {'Null:0:+': 0}     \n",
       "chr2:338609:+       0        [('chr2:338609:+', 1)]   {'Null:0:+': 0}     \n",
       "chr4:413026:+       0        [('chr4:413026:+', 2)]   {'Null:0:+': 0}     \n",
       "chr14:596883:+      0        [('chr14:596883:+', 1)]  {'Null:0:+': 0}     \n",
       "chr13:899947:+      0        [('chr13:899947:+', 1)]  {'Null:0:+': 0}     "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df=df.drop(df.columns[-1],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df.columns = ['nTot', 'n', 'scaffold', 'strand', 'pos', 'type',\n",
    "       'nMainLocation', 'nInsert', 'All genomic mappings',\n",
    "       'All insert mappings']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "      <th>All insert mappings</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr9:937375:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr9</td>\n",
       "      <td>+</td>\n",
       "      <td>937375</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr9:937375:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2:338609:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr2</td>\n",
       "      <td>+</td>\n",
       "      <td>338609</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr2:338609:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4:413026:+</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>+</td>\n",
       "      <td>413026</td>\n",
       "      <td>Single</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr4:413026:+', 2)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:596883:+</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>chr14</td>\n",
       "      <td>+</td>\n",
       "      <td>596883</td>\n",
       "      <td>Single</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr14:596883:+', 2)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:899947:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>+</td>\n",
       "      <td>899947</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr13:899947:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                nTot  n scaffold strand     pos    type  nMainLocation  \\\n",
       "barcode                                                                  \n",
       "chr9:937375:+   1     1  chr9     +      937375  Single  1               \n",
       "chr2:338609:+   1     1  chr2     +      338609  Single  1               \n",
       "chr4:413026:+   2     2  chr4     +      413026  Single  2               \n",
       "chr14:596883:+  2     2  chr14    +      596883  Single  2               \n",
       "chr13:899947:+  1     1  chr13    +      899947  Single  1               \n",
       "\n",
       "                nInsert     All genomic mappings All insert mappings  \n",
       "barcode                                                               \n",
       "chr9:937375:+   0        [('chr9:937375:+', 1)]   {'Null:0:+': 0}     \n",
       "chr2:338609:+   0        [('chr2:338609:+', 1)]   {'Null:0:+': 0}     \n",
       "chr4:413026:+   0        [('chr4:413026:+', 2)]   {'Null:0:+': 0}     \n",
       "chr14:596883:+  0        [('chr14:596883:+', 2)]  {'Null:0:+': 0}     \n",
       "chr13:899947:+  0        [('chr13:899947:+', 1)]  {'Null:0:+': 0}     "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapping 122 barcodes on chr1 to 218 genes on chr1\n",
      "Mapping 150 barcodes on chr10 to 282 genes on chr10\n",
      "Mapping 114 barcodes on chr11 to 248 genes on chr11\n",
      "Mapping 124 barcodes on chr12 to 229 genes on chr12\n",
      "Mapping 94 barcodes on chr13 to 196 genes on chr13\n",
      "Mapping 74 barcodes on chr14 to 179 genes on chr14\n",
      "Mapping 24 barcodes on chr15 to 40 genes on chr15\n",
      "Mapping 171 barcodes on chr2 to 283 genes on chr2\n",
      "Mapping 0 barcodes on chr23 to 2 genes on chr23\n",
      "Mapping 0 barcodes on chr24 to 1 genes on chr24\n",
      "Mapping 226 barcodes on chr3 to 373 genes on chr3\n",
      "Mapping 0 barcodes on chr32 to 1 genes on chr32\n",
      "Mapping 224 barcodes on chr4 to 356 genes on chr4\n",
      "Mapping 278 barcodes on chr5 to 500 genes on chr5\n",
      "Mapping 325 barcodes on chr6 to 535 genes on chr6\n",
      "Mapping 170 barcodes on chr7 to 365 genes on chr7\n",
      "Mapping 135 barcodes on chr8 to 357 genes on chr8\n",
      "Mapping 175 barcodes on chr9 to 318 genes on chr9\n"
     ]
    }
   ],
   "source": [
    "#this takes a few minutes\n",
    "def get_gene_by_bc_position(pos):\n",
    "    \n",
    "    gene_list = [scaffold_gff.loc[interval]['ID'] for interval in idx if pos in interval]\n",
    "    \n",
    "    #barcode position not in any genes\n",
    "    if not gene_list:\n",
    "        return ''\n",
    "    #barcode position is in one gene\n",
    "    if len(gene_list) == 1:\n",
    "        return gene_list[0]\n",
    "    #barcode position is in multiple genes (should be rare)\n",
    "    else:\n",
    "        return ';'.join(gene_list)\n",
    "\n",
    "barcode_to_gene_dict = {}\n",
    "\n",
    "for scaffold in sorted(gff['scaffold'].unique()):\n",
    "    \n",
    "    scaffold_gff = gff[gff['scaffold']==scaffold]\n",
    "    scaffold_barcodes = df[df['scaffold']==scaffold]\n",
    "    \n",
    "    #divide scaffold_gff.shape by 2 here to get number of genes (original shape includes promoters)\n",
    "    print('Mapping {} barcodes on {} to {} genes on {}'.format(scaffold_barcodes.shape[0], scaffold,\n",
    "                                                              round(scaffold_gff.shape[0]/2), scaffold))\n",
    "\n",
    "    #turn the start and stop of each gene into an interval\n",
    "    v = scaffold_gff.loc[:, 'start':'stop'].apply(tuple, 1).tolist()\n",
    "    idx = pd.IntervalIndex.from_tuples(v, closed='both') \n",
    "    scaffold_gff = scaffold_gff.set_index(idx)\n",
    "    \n",
    "    #for each barcode, check if it's in each start/stop interval\n",
    "    barcode_to_gene_dict.update(scaffold_barcodes['pos'].apply(get_gene_by_bc_position).to_dict())\n",
    "    \n",
    "df['gene'] = df.index.map(barcode_to_gene_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "      <th>All insert mappings</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transposonLocation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr9:937375:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr9</td>\n",
       "      <td>+</td>\n",
       "      <td>937375</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr9:937375:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "      <td>kmYHR034C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2:338609:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr2</td>\n",
       "      <td>+</td>\n",
       "      <td>338609</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr2:338609:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "      <td>klYJR144W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4:413026:+</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>chr4</td>\n",
       "      <td>+</td>\n",
       "      <td>413026</td>\n",
       "      <td>Single</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr4:413026:+', 2)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:596883:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr14</td>\n",
       "      <td>+</td>\n",
       "      <td>596883</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr14:596883:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:899947:+</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>+</td>\n",
       "      <td>899947</td>\n",
       "      <td>Single</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr13:899947:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    nTot  n scaffold strand     pos    type  nMainLocation  \\\n",
       "transposonLocation                                                           \n",
       "chr9:937375:+       1     1  chr9     +      937375  Single  1               \n",
       "chr2:338609:+       1     1  chr2     +      338609  Single  1               \n",
       "chr4:413026:+       2     2  chr4     +      413026  Single  2               \n",
       "chr14:596883:+      1     1  chr14    +      596883  Single  1               \n",
       "chr13:899947:+      1     1  chr13    +      899947  Single  1               \n",
       "\n",
       "                    nInsert     All genomic mappings All insert mappings  \\\n",
       "transposonLocation                                                         \n",
       "chr9:937375:+       0        [('chr9:937375:+', 1)]   {'Null:0:+': 0}      \n",
       "chr2:338609:+       0        [('chr2:338609:+', 1)]   {'Null:0:+': 0}      \n",
       "chr4:413026:+       0        [('chr4:413026:+', 2)]   {'Null:0:+': 0}      \n",
       "chr14:596883:+      0        [('chr14:596883:+', 1)]  {'Null:0:+': 0}      \n",
       "chr13:899947:+      0        [('chr13:899947:+', 1)]  {'Null:0:+': 0}      \n",
       "\n",
       "                         gene  \n",
       "transposonLocation             \n",
       "chr9:937375:+       kmYHR034C  \n",
       "chr2:338609:+       klYJR144W  \n",
       "chr4:413026:+                  \n",
       "chr14:596883:+                 \n",
       "chr13:899947:+                 "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check that a 'gene' column has been added to df\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1861 barcodes not in coding sequences\n",
      "\n",
      "545 barcodes in coding sequences\n",
      "293 barcodes in Kl coding sequences\n",
      "252 barcodes in Km coding sequences\n"
     ]
    }
   ],
   "source": [
    "print('{} barcodes not in coding sequences'.format(df[df['gene']==''].shape[0]))\n",
    "print()\n",
    "\n",
    "print('{} barcodes in coding sequences'.format(df[df['gene']!=''].shape[0]))\n",
    "print('{} barcodes in Kl coding sequences'.format(df[df['gene'].str.startswith('kl',na=False)].shape[0]))\n",
    "print('{} barcodes in Km coding sequences'.format(df[df['gene'].str.startswith('km',na=False)].shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(annotated_poolfile_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8700</th>\n",
       "      <td>chr9</td>\n",
       "      <td>YGOB/YGAP</td>\n",
       "      <td>CDS</td>\n",
       "      <td>937252</td>\n",
       "      <td>938259</td>\n",
       "      <td>+</td>\n",
       "      <td>gene_id \"KM020C04300\"; transcript_id \"KM020C04300\"</td>\n",
       "      <td>kmYHR034C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     scaffold     source type   start    stop strand  \\\n",
       "8700  chr9     YGOB/YGAP  CDS  937252  938259  +       \n",
       "\n",
       "                                                    info         ID  \n",
       "8700  gene_id \"KM020C04300\"; transcript_id \"KM020C04300\"  kmYHR034C  "
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff[gff['ID']=='kmYHR034C']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "solaenv",
   "language": "python",
   "name": "solaenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
