{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Note: the mapping script takes days to run! You should be running this notebook using screen, so if you close this page the script will keep running.\n",
    "\n",
    "If you close the notebook while using screen, the log output within the notebook might stop being printed. But the logfile specified in the line where you call the script will keep being written.\n",
    "\n",
    "How to run a Jupyter notebook with screen: https://docs.google.com/document/d/1lLcl3HhDBrn87M4-IhkpDLOmvmmf09-KeLSgn3U21hE/edit?usp=sharing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make metafile for TNseq mapping script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Info that will be the same for all pools\n",
    "\n",
    "metafile_name: Name of file where we're going to write the metadata\n",
    "\n",
    "combined_pool_name: What we're going to call the combination of individual pools\n",
    "\n",
    "ReadModel: Full or relative path to a text file describing the expected structure of TnSeq reads.  The file consists of 4 lines.  Line 1 denotes the number of initial variable bases to ignore as 'N's.  Line 2 is the sequence preceding the barcode.  Line 3 is the expected length of sequence barcode denoted as 'N's.  Line 4 is the sequence between the barcode and the junction with the genome.  Example:\n",
    "\n",
    "    nnnnnn\n",
    "    GATGTCCACGAGGTCTCT\n",
    "    NNNNNNNNNNNNNNNNNNNN\n",
    "    CGTACGCTGCAGGTCGACAATGATCCAAACTATCAGTGTTTGA\t\n",
    "\n",
    "InsertionSequence: Full or relative path to a text file describing giving the full sequence of each insertion.  This should be a fasta format file with two entries labeled 'insert' and 'plasmid' giving the full sequence of the expected insertion into the genome and and the remaining source plasmid sequence, if appropriate. Example:\n",
    "\n",
    "    >insert\n",
    "    agtcac...\n",
    "    >plasmid\n",
    "    actgact... \n",
    "      \n",
    "GenomeSequence: Full or relative path to a .fasta file of genome sequences for all species in pool.\n",
    "\n",
    "OutputDir: Full or relative path to the desired directory to save output files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED\n",
    "\n",
    "metafile_name = 'TNSeq_mapping_metafile_pools30_to_41_individual.txt'\n",
    "mapping_log_file = 'TNSeq_mapping_pools30_to_41_individual.log'\n",
    "\n",
    "ReadModel = '/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt'\n",
    "InsertionSequence = '/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt'\n",
    "GenomeSequence = '/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt'\n",
    "OutputDir = 'TNSeq_mapping_output_new_genome'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pool-specific information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where you saved the merged FASTQs in TNseq_prepare_files.ipynb\n",
    "FASTQ_directory = 'TNSeq_merge_FASTQ'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD31_S11_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD32_S10_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD33_S9_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD34_S8_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD35_S7_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD36_S6_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD37_S5_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD38_S4_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD39_S3_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD40_S2_L002_merged.fastq\r\n",
      "TNSeq_merge_FASTQ/REPEAT-RBAD41_S1_L002_merged.fastq\r\n"
     ]
    }
   ],
   "source": [
    "#get the list of fastqs we generated in the BBmerge step above\n",
    "#copy and paste these into the cell below this one\n",
    "\n",
    "!ls {FASTQ_directory}/*_merged.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each pool, come up with a UNIQUE short name for the pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED BUT KEEP THE [('NAME', 'PATH/TO/FASTQ.FASTQ')] FORMAT\n",
    "\n",
    "short_names_and_fastqs = [('POOL30', 'TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq'),\n",
    "                          ('POOL31', 'TNSeq_merge_FASTQ/REPEAT-RBAD31_S11_L002_merged.fastq'),\n",
    "                          ('POOL32', 'TNSeq_merge_FASTQ/REPEAT-RBAD32_S10_L002_merged.fastq'),\n",
    "                          ('POOL33', 'TNSeq_merge_FASTQ/REPEAT-RBAD33_S9_L002_merged.fastq'),\n",
    "                          ('POOL34', 'TNSeq_merge_FASTQ/REPEAT-RBAD34_S8_L002_merged.fastq'),\n",
    "                          ('POOL35', 'TNSeq_merge_FASTQ/REPEAT-RBAD35_S7_L002_merged.fastq'),\n",
    "                          ('POOL36', 'TNSeq_merge_FASTQ/REPEAT-RBAD36_S6_L002_merged.fastq'),\n",
    "                          ('POOL37', 'TNSeq_merge_FASTQ/REPEAT-RBAD37_S5_L002_merged.fastq'),\n",
    "                          ('POOL38', 'TNSeq_merge_FASTQ/REPEAT-RBAD38_S4_L002_merged.fastq'),\n",
    "                          ('POOL39', 'TNSeq_merge_FASTQ/REPEAT-RBAD39_S3_L002_merged.fastq'),\n",
    "                          ('POOL40', 'TNSeq_merge_FASTQ/REPEAT-RBAD40_S2_L002_merged.fastq'),\n",
    "                          ('POOL41', 'TNSeq_merge_FASTQ/REPEAT-RBAD41_S1_L002_merged.fastq'),]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Writing the above info into a metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['Pool', 'ShortName', 'Fastq', 'ReadModel', 'InsertionSequence', 'GenomeSequence', 'OutputDir']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for pool, fastq in short_names_and_fastqs:\n",
    "        \n",
    "        #we write the pool name twice because we're first processing each pool separately then combining them\n",
    "        #if we were combining them all at once then the first pool variable would be the name of the combined pool\n",
    "        to_write = '\\t'.join([pool, pool, fastq, ReadModel, InsertionSequence, GenomeSequence, OutputDir])\n",
    "        f.write(to_write+'\\n')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pool</th>\n",
       "      <th>ShortName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>ReadModel</th>\n",
       "      <th>InsertionSequence</th>\n",
       "      <th>GenomeSequence</th>\n",
       "      <th>OutputDir</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>POOL30</td>\n",
       "      <td>POOL30</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>POOL31</td>\n",
       "      <td>POOL31</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD31_S11_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>POOL32</td>\n",
       "      <td>POOL32</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD32_S10_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>POOL33</td>\n",
       "      <td>POOL33</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD33_S9_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>POOL34</td>\n",
       "      <td>POOL34</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD34_S8_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>POOL35</td>\n",
       "      <td>POOL35</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD35_S7_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>POOL36</td>\n",
       "      <td>POOL36</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD36_S6_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>POOL37</td>\n",
       "      <td>POOL37</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD37_S5_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POOL38</td>\n",
       "      <td>POOL38</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD38_S4_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>POOL39</td>\n",
       "      <td>POOL39</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD39_S3_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>POOL40</td>\n",
       "      <td>POOL40</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD40_S2_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>POOL41</td>\n",
       "      <td>POOL41</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD41_S1_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Pool ShortName                                                  Fastq  \\\n",
       "0   POOL30  POOL30    TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq   \n",
       "1   POOL31  POOL31    TNSeq_merge_FASTQ/REPEAT-RBAD31_S11_L002_merged.fastq   \n",
       "2   POOL32  POOL32    TNSeq_merge_FASTQ/REPEAT-RBAD32_S10_L002_merged.fastq   \n",
       "3   POOL33  POOL33    TNSeq_merge_FASTQ/REPEAT-RBAD33_S9_L002_merged.fastq    \n",
       "4   POOL34  POOL34    TNSeq_merge_FASTQ/REPEAT-RBAD34_S8_L002_merged.fastq    \n",
       "5   POOL35  POOL35    TNSeq_merge_FASTQ/REPEAT-RBAD35_S7_L002_merged.fastq    \n",
       "6   POOL36  POOL36    TNSeq_merge_FASTQ/REPEAT-RBAD36_S6_L002_merged.fastq    \n",
       "7   POOL37  POOL37    TNSeq_merge_FASTQ/REPEAT-RBAD37_S5_L002_merged.fastq    \n",
       "8   POOL38  POOL38    TNSeq_merge_FASTQ/REPEAT-RBAD38_S4_L002_merged.fastq    \n",
       "9   POOL39  POOL39    TNSeq_merge_FASTQ/REPEAT-RBAD39_S3_L002_merged.fastq    \n",
       "10  POOL40  POOL40    TNSeq_merge_FASTQ/REPEAT-RBAD40_S2_L002_merged.fastq    \n",
       "11  POOL41  POOL41    TNSeq_merge_FASTQ/REPEAT-RBAD41_S1_L002_merged.fastq    \n",
       "\n",
       "                                                           ReadModel  \\\n",
       "0   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "1   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "2   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "3   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "4   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "5   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "6   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "7   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "8   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "9   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "10  /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "11  /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "\n",
       "                                                                InsertionSequence  \\\n",
       "0   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "1   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "2   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "3   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "4   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "5   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "6   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "7   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "8   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "9   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "10  /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "11  /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "\n",
       "                                                              GenomeSequence  \\\n",
       "0   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "1   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "2   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "3   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "4   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "5   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "6   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "7   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "8   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "9   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "10  /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "11  /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "\n",
       "                          OutputDir  \n",
       "0   TNSeq_mapping_output_new_genome  \n",
       "1   TNSeq_mapping_output_new_genome  \n",
       "2   TNSeq_mapping_output_new_genome  \n",
       "3   TNSeq_mapping_output_new_genome  \n",
       "4   TNSeq_mapping_output_new_genome  \n",
       "5   TNSeq_mapping_output_new_genome  \n",
       "6   TNSeq_mapping_output_new_genome  \n",
       "7   TNSeq_mapping_output_new_genome  \n",
       "8   TNSeq_mapping_output_new_genome  \n",
       "9   TNSeq_mapping_output_new_genome  \n",
       "10  TNSeq_mapping_output_new_genome  \n",
       "11  TNSeq_mapping_output_new_genome  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#view metafile\n",
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Map transposons in each pool to genome\n",
    "\n",
    "View additional input and output options here: https://github.com/stcoradetti/RBseq/blob/master/README.txt\n",
    "\n",
    "This takes a long time to run and will be stuck on the \"Looking for 20bp sequence barcode\" step for a while. You can see how many barcodes have been processed by switching to the output directory specified above and typing:\n",
    "\n",
    "    wc -l *blastquery.fasta\n",
    "    \n",
    "This number divided by 2 is roughly how many reads have been processed. You can also check the logFile specified below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RBseq_Map_Insertions.py\r\n",
      "2022-05-20 11:55:37 Version: 1.1.4_PBa\r\n",
      "2022-05-20 11:55:37 Release Date: August 8, 2020\r\n",
      "2022-05-20 11:55:37 Options passed:  metafile:TNSeq_mapping_metafile_pools30_to_41_individual.txt  logFile:TNSeq_mapping_pools30_to_41_individual.log  minQual:10  matchBefore:6  matchAfter:6  matchJunction:4  maxFillerSeq:100  barcodeVariation:2  wobbleAllowed:1  scoreDiff:10  minFraction:0.9  filterNeighborhood:10  filterEditDistance:5  minPercentID:95  maxEvalue:0.1  useMappedFiles:False  noInsertHits:False  noBarcodes:False \r\n",
      "2022-05-20 11:55:37 Logging status updates in TNSeq_mapping_pools30_to_41_individual.log\r\n",
      "2022-05-20 11:55:37 Loading TnSeq library metadata from TNSeq_mapping_metafile_pools30_to_41_individual.txt\r\n",
      "2022-05-20 11:55:37 Finding barcodes in fastqs and mapping insertion locations\r\n",
      "2022-05-20 11:55:37   Mapping reads from TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq using insertion model /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt\r\n",
      "2022-05-20 11:55:37   Looking for 20bp sequence barcode flanked by GTCTCT and CGTACG\r\n"
     ]
    }
   ],
   "source": [
    "cmd = 'nice -n 17 python3 /usr2/people/clairedubin/barseq/latest_pipeline/RBseq_Map_Insertions_v1.1.4_PBa_Jskerker.py --metafile {} --logFile {}'.format(metafile_name, mapping_log_file)\n",
    "\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combine pools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a new metafile, this time combining the pools into one larger pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED\n",
    "\n",
    "combined_pool_name = 'POOLS30_to_41_combined'\n",
    "metafile_name = 'TNSeq_mapping_metafile_pools30_to_41_combined.txt'\n",
    "mapping_log_file = 'TNSeq_mapping_pools30_to_41_combined.log'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['Pool', 'ShortName', 'Fastq', 'ReadModel', 'InsertionSequence', 'GenomeSequence', 'OutputDir']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for pool, fastq in short_names_and_fastqs:\n",
    "        \n",
    "        #the pool name is now the combined pool name\n",
    "        to_write = '\\t'.join([combined_pool_name, pool, fastq, ReadModel, InsertionSequence, GenomeSequence, OutputDir])\n",
    "        f.write(to_write+'\\n')\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note how this metafile is different than the one above - the \"Pool\" column is the name of the combined pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pool</th>\n",
       "      <th>ShortName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>ReadModel</th>\n",
       "      <th>InsertionSequence</th>\n",
       "      <th>GenomeSequence</th>\n",
       "      <th>OutputDir</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL30</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL31</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD31_S11_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL32</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD32_S10_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL33</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD33_S9_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL34</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD34_S8_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL35</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD35_S7_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL36</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD36_S6_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL37</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD37_S5_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL38</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD38_S4_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL39</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD39_S3_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL40</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD40_S2_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>POOLS30_to_41_combined</td>\n",
       "      <td>POOL41</td>\n",
       "      <td>TNSeq_merge_FASTQ/REPEAT-RBAD41_S1_L002_merged.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>TNSeq_mapping_output_new_genome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Pool ShortName  \\\n",
       "0   POOLS30_to_41_combined  POOL30     \n",
       "1   POOLS30_to_41_combined  POOL31     \n",
       "2   POOLS30_to_41_combined  POOL32     \n",
       "3   POOLS30_to_41_combined  POOL33     \n",
       "4   POOLS30_to_41_combined  POOL34     \n",
       "5   POOLS30_to_41_combined  POOL35     \n",
       "6   POOLS30_to_41_combined  POOL36     \n",
       "7   POOLS30_to_41_combined  POOL37     \n",
       "8   POOLS30_to_41_combined  POOL38     \n",
       "9   POOLS30_to_41_combined  POOL39     \n",
       "10  POOLS30_to_41_combined  POOL40     \n",
       "11  POOLS30_to_41_combined  POOL41     \n",
       "\n",
       "                                                    Fastq  \\\n",
       "0   TNSeq_merge_FASTQ/REPEAT-RBAD30_S12_L002_merged.fastq   \n",
       "1   TNSeq_merge_FASTQ/REPEAT-RBAD31_S11_L002_merged.fastq   \n",
       "2   TNSeq_merge_FASTQ/REPEAT-RBAD32_S10_L002_merged.fastq   \n",
       "3   TNSeq_merge_FASTQ/REPEAT-RBAD33_S9_L002_merged.fastq    \n",
       "4   TNSeq_merge_FASTQ/REPEAT-RBAD34_S8_L002_merged.fastq    \n",
       "5   TNSeq_merge_FASTQ/REPEAT-RBAD35_S7_L002_merged.fastq    \n",
       "6   TNSeq_merge_FASTQ/REPEAT-RBAD36_S6_L002_merged.fastq    \n",
       "7   TNSeq_merge_FASTQ/REPEAT-RBAD37_S5_L002_merged.fastq    \n",
       "8   TNSeq_merge_FASTQ/REPEAT-RBAD38_S4_L002_merged.fastq    \n",
       "9   TNSeq_merge_FASTQ/REPEAT-RBAD39_S3_L002_merged.fastq    \n",
       "10  TNSeq_merge_FASTQ/REPEAT-RBAD40_S2_L002_merged.fastq    \n",
       "11  TNSeq_merge_FASTQ/REPEAT-RBAD41_S1_L002_merged.fastq    \n",
       "\n",
       "                                                           ReadModel  \\\n",
       "0   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "1   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "2   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "3   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "4   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "5   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "6   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "7   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "8   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "9   /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "10  /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "11  /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "\n",
       "                                                                InsertionSequence  \\\n",
       "0   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "1   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "2   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "3   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "4   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "5   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "6   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "7   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "8   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "9   /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "10  /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "11  /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "\n",
       "                                                              GenomeSequence  \\\n",
       "0   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "1   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "2   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "3   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "4   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "5   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "6   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "7   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "8   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "9   /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "10  /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "11  /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "\n",
       "                          OutputDir  \n",
       "0   TNSeq_mapping_output_new_genome  \n",
       "1   TNSeq_mapping_output_new_genome  \n",
       "2   TNSeq_mapping_output_new_genome  \n",
       "3   TNSeq_mapping_output_new_genome  \n",
       "4   TNSeq_mapping_output_new_genome  \n",
       "5   TNSeq_mapping_output_new_genome  \n",
       "6   TNSeq_mapping_output_new_genome  \n",
       "7   TNSeq_mapping_output_new_genome  \n",
       "8   TNSeq_mapping_output_new_genome  \n",
       "9   TNSeq_mapping_output_new_genome  \n",
       "10  TNSeq_mapping_output_new_genome  \n",
       "11  TNSeq_mapping_output_new_genome  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#view metafile\n",
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This step takes ~ 2 hours for 4 pools."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RBseq_Map_Insertions.py\n",
      "2022-05-23 14:24:08 Version: 1.1.4_PBa\n",
      "2022-05-23 14:24:08 Release Date: August 8, 2020\n",
      "2022-05-23 14:24:08 Options passed:  metafile:TNSeq_mapping_metafile_pools30_to_41_combined.txt  logFile:TNSeq_mapping_pools30_to_41_combined.log  minQual:10  matchBefore:6  matchAfter:6  matchJunction:4  maxFillerSeq:100  barcodeVariation:2  wobbleAllowed:1  scoreDiff:10  minFraction:0.9  filterNeighborhood:10  filterEditDistance:5  minPercentID:95  maxEvalue:0.1  useMappedFiles:True  noInsertHits:False  noBarcodes:False \n",
      "2022-05-23 14:24:08 Logging status updates in TNSeq_mapping_pools30_to_41_combined.log\n",
      "2022-05-23 14:24:08 Loading TnSeq library metadata from TNSeq_mapping_metafile_pools30_to_41_combined.txt\n",
      "2022-05-23 14:24:08 --useMappedFiles option passed. Skipping mapping step and proceeding to pool analysis\n",
      "2022-05-23 14:24:08 Processing mapped reads to characterize mutant pool(s)\n",
      "2022-05-23 14:24:08 Analyzing pool: POOLS30_to_41_combined\n",
      "2022-05-23 14:24:08   Reading entries from: TNSeq_mapping_output_new_genome/POOL30_mapped.txt\n",
      "2022-05-23 14:24:15       1.0 million lines read... \n",
      "2022-05-23 14:24:21       2.0 million lines read... \n",
      "2022-05-23 14:24:27       3.0 million lines read... \n",
      "2022-05-23 14:24:33       4.0 million lines read... \n",
      "2022-05-23 14:24:40       5.0 million lines read... \n",
      "2022-05-23 14:24:44   Reading entries from: TNSeq_mapping_output_new_genome/POOL31_mapped.txt\n",
      "2022-05-23 14:24:51       1.0 million lines read... \n",
      "2022-05-23 14:24:57       2.0 million lines read... \n"
     ]
    }
   ],
   "source": [
    "cmd = 'nice -n 17 python3 /usr2/people/clairedubin/barseq/latest_pipeline/RBseq_Map_Insertions_v1.1.4_PBa_Jskerker.py --useMappedFiles --metafile {} --logFile {}'.format(metafile_name, mapping_log_file)\n",
    "\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "solaenv",
   "language": "python",
   "name": "solaenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
