{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Modified from https://github.com/melanieabrams-pub/RH-seq_with_barcoding/blob/main/annotate_poolfile_for_barseq_v1.2_JSkerker.ipynb and \n",
    "https://github.com/melanieabrams-pub/kluyv_RH-seq_with_barcoding/blob/main/annotate_poolfile_kluyv.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load GFF annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#you don't need to change the path below - this is the most recent annotation file (5/12/22)\n",
    "gff_file = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/modified_gff_file_finalized_08_08_2023'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff = pd.read_csv(gff_file, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1' 'chr2' 'chr3' 'chr4' 'chr5' 'chr6' 'chr7']\n"
     ]
    }
   ],
   "source": [
    "t = gff['scaffold'].unique()\n",
    "print(t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1381.0</td>\n",
       "      <td>2142.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_1</td>\n",
       "      <td>proteinId=2114025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2744.0</td>\n",
       "      <td>3343.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_2</td>\n",
       "      <td>proteinId=2293935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3344.0</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_3</td>\n",
       "      <td>proteinId=2293936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>5330.0</td>\n",
       "      <td>6385.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_4</td>\n",
       "      <td>proteinId=2051335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>15458.0</td>\n",
       "      <td>25533.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_5</td>\n",
       "      <td>proteinId=2121898</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  scaffold source type    start     stop strand       info                 ID\n",
       "0  chr1     JGI    CDS  1381.0   2142.0   +      ID=gene_1  proteinId=2114025\n",
       "1  chr1     JGI    CDS  2744.0   3343.0   -      ID=gene_2  proteinId=2293935\n",
       "2  chr1     JGI    CDS  3344.0   3817.0   -      ID=gene_3  proteinId=2293936\n",
       "3  chr1     JGI    CDS  5330.0   6385.0   -      ID=gene_4  proteinId=2051335\n",
       "4  chr1     JGI    CDS  15458.0  25533.0  +      ID=gene_5  proteinId=2121898"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1381.0</td>\n",
       "      <td>2142.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_1</td>\n",
       "      <td>proteinId=2114025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2744.0</td>\n",
       "      <td>3343.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_2</td>\n",
       "      <td>proteinId=2293935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3344.0</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_3</td>\n",
       "      <td>proteinId=2293936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>5330.0</td>\n",
       "      <td>6385.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_4</td>\n",
       "      <td>proteinId=2051335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>15458.0</td>\n",
       "      <td>25533.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_5</td>\n",
       "      <td>proteinId=2121898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9105</th>\n",
       "      <td>chr7</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4089438.0</td>\n",
       "      <td>4091337.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_10836</td>\n",
       "      <td>proteinId=103797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9106</th>\n",
       "      <td>chr7</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4095568.0</td>\n",
       "      <td>4097284.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_10837</td>\n",
       "      <td>proteinId=2071098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9107</th>\n",
       "      <td>chr7</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4098122.0</td>\n",
       "      <td>4098907.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_10838</td>\n",
       "      <td>proteinId=2114022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9108</th>\n",
       "      <td>chr7</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4100606.0</td>\n",
       "      <td>4101293.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_10840</td>\n",
       "      <td>proteinId=2114023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9109</th>\n",
       "      <td>chr7</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>4101637.0</td>\n",
       "      <td>4103479.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_10841</td>\n",
       "      <td>proteinId=2313233</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9110 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     scaffold source type      start       stop strand           info  \\\n",
       "0     chr1     JGI    CDS  1381.0     2142.0     +      ID=gene_1       \n",
       "1     chr1     JGI    CDS  2744.0     3343.0     -      ID=gene_2       \n",
       "2     chr1     JGI    CDS  3344.0     3817.0     -      ID=gene_3       \n",
       "3     chr1     JGI    CDS  5330.0     6385.0     -      ID=gene_4       \n",
       "4     chr1     JGI    CDS  15458.0    25533.0    +      ID=gene_5       \n",
       "...    ...     ...    ...      ...        ...   ..            ...       \n",
       "9105  chr7     JGI    CDS  4089438.0  4091337.0  +      ID=gene_10836   \n",
       "9106  chr7     JGI    CDS  4095568.0  4097284.0  +      ID=gene_10837   \n",
       "9107  chr7     JGI    CDS  4098122.0  4098907.0  -      ID=gene_10838   \n",
       "9108  chr7     JGI    CDS  4100606.0  4101293.0  -      ID=gene_10840   \n",
       "9109  chr7     JGI    CDS  4101637.0  4103479.0  -      ID=gene_10841   \n",
       "\n",
       "                     ID  \n",
       "0     proteinId=2114025  \n",
       "1     proteinId=2293935  \n",
       "2     proteinId=2293936  \n",
       "3     proteinId=2051335  \n",
       "4     proteinId=2121898  \n",
       "...                 ...  \n",
       "9105  proteinId=103797   \n",
       "9106  proteinId=2071098  \n",
       "9107  proteinId=2114022  \n",
       "9108  proteinId=2114023  \n",
       "9109  proteinId=2313233  \n",
       "\n",
       "[9110 rows x 8 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gff"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Match barcode insert locations to gene annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/auto/sahara/namib/home/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE IF NEEDED: load COMBINED poolfile generated in TNseq mapping step\n",
    "\n",
    "poolfile = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_Apr_26_2024_without_vector_seq_2_fastq_combined_poolfile'\n",
    "\n",
    "#CHANGE IF NEEDED: annotated pool file path to output\n",
    "\n",
    "annotated_poolfile_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>rcbarcode</th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "      <th>All insert mappings</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTCATTCAGGATCTTTGCTC</td>\n",
       "      <td>GAGCAAAGATCCTGAATGAA</td>\n",
       "      <td>154</td>\n",
       "      <td>154</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8764372</td>\n",
       "      <td>Single</td>\n",
       "      <td>144</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:1138:-': 1, 'insert:19:+': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACGTATTCGCAGTCCTCATA</td>\n",
       "      <td>TATGAGGACTGCGAATACGT</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>526973</td>\n",
       "      <td>Single</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr1:526973:-', 21)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCGGCGGCAGCGTTCCTCGC</td>\n",
       "      <td>GCGAGGAACGCTGCCGCCGC</td>\n",
       "      <td>150</td>\n",
       "      <td>150</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>2235514</td>\n",
       "      <td>Single</td>\n",
       "      <td>140</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:365:+': 1, 'insert:1:+': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GAATAGGTAATTAAGCCTGC</td>\n",
       "      <td>GCAGGCTTAATTACCTATTC</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8013504</td>\n",
       "      <td>Single</td>\n",
       "      <td>54</td>\n",
       "      <td>3</td>\n",
       "      <td>[('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:1521:-': 1, 'insert:13:+': 1, 'insert:1:+': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TTCGCGTGGGTGAATTAACT</td>\n",
       "      <td>AGTTAATTCACCCACGCGAA</td>\n",
       "      <td>415</td>\n",
       "      <td>415</td>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>2435148</td>\n",
       "      <td>Single</td>\n",
       "      <td>372</td>\n",
       "      <td>14</td>\n",
       "      <td>[('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:1:+': 5, 'insert:130:+': 1, 'insert:550:+': 1, 'insert:4:+': 1, 'insert:1877:+': 2, 'plasmid:5498:-': 1, 'insert:18:+': 1, 'insert:26:+': 1, 'insert:16:+': 1}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                barcode             rcbarcode  nTot    n scaffold strand  \\\n",
       "0  TTCATTCAGGATCTTTGCTC  GAGCAAAGATCCTGAATGAA  154   154  chr1     +       \n",
       "1  ACGTATTCGCAGTCCTCATA  TATGAGGACTGCGAATACGT  21    21   chr1     -       \n",
       "2  GCGGCGGCAGCGTTCCTCGC  GCGAGGAACGCTGCCGCCGC  150   150  chr1     -       \n",
       "3  GAATAGGTAATTAAGCCTGC  GCAGGCTTAATTACCTATTC  62    62   chr1     +       \n",
       "4  TTCGCGTGGGTGAATTAACT  AGTTAATTCACCCACGCGAA  415   415  chr5     +       \n",
       "\n",
       "       pos    type  nMainLocation  nInsert  \\\n",
       "0  8764372  Single  144            2         \n",
       "1  526973   Single  21             0         \n",
       "2  2235514  Single  140            2         \n",
       "3  8013504  Single  54             3         \n",
       "4  2435148  Single  372            14        \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   All genomic mappings  \\\n",
       "0  [('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "1  [('chr1:526973:-', 21)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "2  [('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "3  [('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "4  [('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]   \n",
       "\n",
       "                                                                                                                                                                      All insert mappings  \n",
       "0  {'Null:0:+': 0, 'insert:1138:-': 1, 'insert:19:+': 1}                                                                                                                                   \n",
       "1  {'Null:0:+': 0}                                                                                                                                                                         \n",
       "2  {'Null:0:+': 0, 'insert:365:+': 1, 'insert:1:+': 1}                                                                                                                                     \n",
       "3  {'Null:0:+': 0, 'insert:1521:-': 1, 'insert:13:+': 1, 'insert:1:+': 1}                                                                                                                  \n",
       "4  {'Null:0:+': 0, 'insert:1:+': 5, 'insert:130:+': 1, 'insert:550:+': 1, 'insert:4:+': 1, 'insert:1877:+': 2, 'plasmid:5498:-': 1, 'insert:18:+': 1, 'insert:26:+': 1, 'insert:16:+': 1}  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(poolfile,sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check that each barcode in the dataframe is unique (the two numbers below should be the same):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of insertions in poolfile: 8521\n",
      "Number of unique barcodes in poolfile: 8521\n"
     ]
    }
   ],
   "source": [
    "print('Number of insertions in poolfile: {}'.format(df.shape[0]))\n",
    "print('Number of unique barcodes in poolfile: {}'.format(len(df['barcode'].unique())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df = df.set_index('transposonLocation') #This is what was there and it is showing error. The transposonlocation was changed to barcode in the next line\n",
    "df = df.set_index('barcode')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['rcbarcode', 'nTot', 'n', 'scaffold', 'strand', 'pos', 'type',\n",
       "       'nMainLocation', 'nInsert', 'All genomic mappings',\n",
       "       'All insert mappings'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rcbarcode</th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "      <th>All insert mappings</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TTCATTCAGGATCTTTGCTC</th>\n",
       "      <td>GAGCAAAGATCCTGAATGAA</td>\n",
       "      <td>154</td>\n",
       "      <td>154</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8764372</td>\n",
       "      <td>Single</td>\n",
       "      <td>144</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:1138:-': 1, 'insert:19:+': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACGTATTCGCAGTCCTCATA</th>\n",
       "      <td>TATGAGGACTGCGAATACGT</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>526973</td>\n",
       "      <td>Single</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr1:526973:-', 21)]</td>\n",
       "      <td>{'Null:0:+': 0}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCGGCGGCAGCGTTCCTCGC</th>\n",
       "      <td>GCGAGGAACGCTGCCGCCGC</td>\n",
       "      <td>150</td>\n",
       "      <td>150</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>2235514</td>\n",
       "      <td>Single</td>\n",
       "      <td>140</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:365:+': 1, 'insert:1:+': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAATAGGTAATTAAGCCTGC</th>\n",
       "      <td>GCAGGCTTAATTACCTATTC</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8013504</td>\n",
       "      <td>Single</td>\n",
       "      <td>54</td>\n",
       "      <td>3</td>\n",
       "      <td>[('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:1521:-': 1, 'insert:13:+': 1, 'insert:1:+': 1}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTCGCGTGGGTGAATTAACT</th>\n",
       "      <td>AGTTAATTCACCCACGCGAA</td>\n",
       "      <td>415</td>\n",
       "      <td>415</td>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>2435148</td>\n",
       "      <td>Single</td>\n",
       "      <td>372</td>\n",
       "      <td>14</td>\n",
       "      <td>[('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]</td>\n",
       "      <td>{'Null:0:+': 0, 'insert:1:+': 5, 'insert:130:+': 1, 'insert:550:+': 1, 'insert:4:+': 1, 'insert:1877:+': 2, 'plasmid:5498:-': 1, 'insert:18:+': 1, 'insert:26:+': 1, 'insert:16:+': 1}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 rcbarcode  nTot    n scaffold strand  \\\n",
       "barcode                                                                 \n",
       "TTCATTCAGGATCTTTGCTC  GAGCAAAGATCCTGAATGAA  154   154  chr1     +       \n",
       "ACGTATTCGCAGTCCTCATA  TATGAGGACTGCGAATACGT  21    21   chr1     -       \n",
       "GCGGCGGCAGCGTTCCTCGC  GCGAGGAACGCTGCCGCCGC  150   150  chr1     -       \n",
       "GAATAGGTAATTAAGCCTGC  GCAGGCTTAATTACCTATTC  62    62   chr1     +       \n",
       "TTCGCGTGGGTGAATTAACT  AGTTAATTCACCCACGCGAA  415   415  chr5     +       \n",
       "\n",
       "                          pos    type  nMainLocation  nInsert  \\\n",
       "barcode                                                         \n",
       "TTCATTCAGGATCTTTGCTC  8764372  Single  144            2         \n",
       "ACGTATTCGCAGTCCTCATA  526973   Single  21             0         \n",
       "GCGGCGGCAGCGTTCCTCGC  2235514  Single  140            2         \n",
       "GAATAGGTAATTAAGCCTGC  8013504  Single  54             3         \n",
       "TTCGCGTGGGTGAATTAACT  2435148  Single  372            14        \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      All genomic mappings  \\\n",
       "barcode                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "TTCATTCAGGATCTTTGCTC  [('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "ACGTATTCGCAGTCCTCATA  [('chr1:526973:-', 21)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "GCGGCGGCAGCGTTCCTCGC  [('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "GAATAGGTAATTAAGCCTGC  [('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "TTCGCGTGGGTGAATTAACT  [('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]   \n",
       "\n",
       "                                                                                                                                                                                         All insert mappings  \n",
       "barcode                                                                                                                                                                                                       \n",
       "TTCATTCAGGATCTTTGCTC  {'Null:0:+': 0, 'insert:1138:-': 1, 'insert:19:+': 1}                                                                                                                                   \n",
       "ACGTATTCGCAGTCCTCATA  {'Null:0:+': 0}                                                                                                                                                                         \n",
       "GCGGCGGCAGCGTTCCTCGC  {'Null:0:+': 0, 'insert:365:+': 1, 'insert:1:+': 1}                                                                                                                                     \n",
       "GAATAGGTAATTAAGCCTGC  {'Null:0:+': 0, 'insert:1521:-': 1, 'insert:13:+': 1, 'insert:1:+': 1}                                                                                                                  \n",
       "TTCGCGTGGGTGAATTAACT  {'Null:0:+': 0, 'insert:1:+': 5, 'insert:130:+': 1, 'insert:550:+': 1, 'insert:4:+': 1, 'insert:1877:+': 2, 'plasmid:5498:-': 1, 'insert:18:+': 1, 'insert:26:+': 1, 'insert:16:+': 1}  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=df.drop(df.columns[-1],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.columns = ['rcbarcode','nTot', 'n', 'scaffold', 'strand', 'pos', 'type',\n",
    "#        'nMainLocation', 'nInsert', 'All genomic mappings',\n",
    "#        'All insert mappings']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rcbarcode</th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TTCATTCAGGATCTTTGCTC</th>\n",
       "      <td>GAGCAAAGATCCTGAATGAA</td>\n",
       "      <td>154</td>\n",
       "      <td>154</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8764372</td>\n",
       "      <td>Single</td>\n",
       "      <td>144</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACGTATTCGCAGTCCTCATA</th>\n",
       "      <td>TATGAGGACTGCGAATACGT</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>526973</td>\n",
       "      <td>Single</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr1:526973:-', 21)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCGGCGGCAGCGTTCCTCGC</th>\n",
       "      <td>GCGAGGAACGCTGCCGCCGC</td>\n",
       "      <td>150</td>\n",
       "      <td>150</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>2235514</td>\n",
       "      <td>Single</td>\n",
       "      <td>140</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAATAGGTAATTAAGCCTGC</th>\n",
       "      <td>GCAGGCTTAATTACCTATTC</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8013504</td>\n",
       "      <td>Single</td>\n",
       "      <td>54</td>\n",
       "      <td>3</td>\n",
       "      <td>[('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTCGCGTGGGTGAATTAACT</th>\n",
       "      <td>AGTTAATTCACCCACGCGAA</td>\n",
       "      <td>415</td>\n",
       "      <td>415</td>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>2435148</td>\n",
       "      <td>Single</td>\n",
       "      <td>372</td>\n",
       "      <td>14</td>\n",
       "      <td>[('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 rcbarcode  nTot    n scaffold strand  \\\n",
       "barcode                                                                 \n",
       "TTCATTCAGGATCTTTGCTC  GAGCAAAGATCCTGAATGAA  154   154  chr1     +       \n",
       "ACGTATTCGCAGTCCTCATA  TATGAGGACTGCGAATACGT  21    21   chr1     -       \n",
       "GCGGCGGCAGCGTTCCTCGC  GCGAGGAACGCTGCCGCCGC  150   150  chr1     -       \n",
       "GAATAGGTAATTAAGCCTGC  GCAGGCTTAATTACCTATTC  62    62   chr1     +       \n",
       "TTCGCGTGGGTGAATTAACT  AGTTAATTCACCCACGCGAA  415   415  chr5     +       \n",
       "\n",
       "                          pos    type  nMainLocation  nInsert  \\\n",
       "barcode                                                         \n",
       "TTCATTCAGGATCTTTGCTC  8764372  Single  144            2         \n",
       "ACGTATTCGCAGTCCTCATA  526973   Single  21             0         \n",
       "GCGGCGGCAGCGTTCCTCGC  2235514  Single  140            2         \n",
       "GAATAGGTAATTAAGCCTGC  8013504  Single  54             3         \n",
       "TTCGCGTGGGTGAATTAACT  2435148  Single  372            14        \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      All genomic mappings  \n",
       "barcode                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "TTCATTCAGGATCTTTGCTC  [('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                         \n",
       "ACGTATTCGCAGTCCTCATA  [('chr1:526973:-', 21)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "GCGGCGGCAGCGTTCCTCGC  [('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                     \n",
       "GAATAGGTAATTAAGCCTGC  [('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  \n",
       "TTCGCGTGGGTGAATTAACT  [('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mapping 2762 barcodes on chr1 to 1356 genes on chr1\n",
      "Mapping 1475 barcodes on chr2 to 715 genes on chr2\n",
      "Mapping 1105 barcodes on chr3 to 605 genes on chr3\n",
      "Mapping 805 barcodes on chr4 to 535 genes on chr4\n",
      "Mapping 596 barcodes on chr5 to 492 genes on chr5\n",
      "Mapping 653 barcodes on chr6 to 396 genes on chr6\n",
      "Mapping 1125 barcodes on chr7 to 455 genes on chr7\n"
     ]
    }
   ],
   "source": [
    "#this takes a few minutes\n",
    "def get_gene_by_bc_position(pos):\n",
    "    \n",
    "    gene_list = [scaffold_gff.loc[interval]['ID'] for interval in idx if pos in interval]\n",
    "    \n",
    "    #barcode position not in any genes\n",
    "    if not gene_list:\n",
    "        return ''\n",
    "    #barcode position is in one gene\n",
    "    if len(gene_list) == 1:\n",
    "        return gene_list[0]\n",
    "    #barcode position is in multiple genes (should be rare)\n",
    "    else:\n",
    "        return ';'.join(gene_list)\n",
    "\n",
    "barcode_to_gene_dict = {}\n",
    "\n",
    "for scaffold in sorted(gff['scaffold'].unique()):\n",
    "    \n",
    "    scaffold_gff = gff[gff['scaffold']==scaffold]\n",
    "    scaffold_barcodes = df[df['scaffold']==scaffold]\n",
    "    \n",
    "    #divide scaffold_gff.shape by 2 here to get number of genes (original shape includes promoters)\n",
    "    print('Mapping {} barcodes on {} to {} genes on {}'.format(scaffold_barcodes.shape[0], scaffold,\n",
    "                                                              round(scaffold_gff.shape[0]/2), scaffold))\n",
    "\n",
    "    #turn the start and stop of each gene into an interval\n",
    "    v = scaffold_gff.loc[:, 'start':'stop'].apply(tuple, 1).tolist()\n",
    "    idx = pd.IntervalIndex.from_tuples(v, closed='both') \n",
    "    scaffold_gff = scaffold_gff.set_index(idx)\n",
    "    \n",
    "    #for each barcode, check if it's in each start/stop interval\n",
    "    barcode_to_gene_dict.update(scaffold_barcodes['pos'].apply(get_gene_by_bc_position).to_dict())\n",
    "    \n",
    "df['gene'] = df.index.map(barcode_to_gene_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rcbarcode</th>\n",
       "      <th>nTot</th>\n",
       "      <th>n</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>strand</th>\n",
       "      <th>pos</th>\n",
       "      <th>type</th>\n",
       "      <th>nMainLocation</th>\n",
       "      <th>nInsert</th>\n",
       "      <th>All genomic mappings</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TTCATTCAGGATCTTTGCTC</th>\n",
       "      <td>GAGCAAAGATCCTGAATGAA</td>\n",
       "      <td>154</td>\n",
       "      <td>154</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8764372</td>\n",
       "      <td>Single</td>\n",
       "      <td>144</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]</td>\n",
       "      <td>proteinId=2298894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACGTATTCGCAGTCCTCATA</th>\n",
       "      <td>TATGAGGACTGCGAATACGT</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>526973</td>\n",
       "      <td>Single</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>[('chr1:526973:-', 21)]</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCGGCGGCAGCGTTCCTCGC</th>\n",
       "      <td>GCGAGGAACGCTGCCGCCGC</td>\n",
       "      <td>150</td>\n",
       "      <td>150</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>2235514</td>\n",
       "      <td>Single</td>\n",
       "      <td>140</td>\n",
       "      <td>2</td>\n",
       "      <td>[('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAATAGGTAATTAAGCCTGC</th>\n",
       "      <td>GCAGGCTTAATTACCTATTC</td>\n",
       "      <td>62</td>\n",
       "      <td>62</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>8013504</td>\n",
       "      <td>Single</td>\n",
       "      <td>54</td>\n",
       "      <td>3</td>\n",
       "      <td>[('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]</td>\n",
       "      <td>proteinId=97899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTCGCGTGGGTGAATTAACT</th>\n",
       "      <td>AGTTAATTCACCCACGCGAA</td>\n",
       "      <td>415</td>\n",
       "      <td>415</td>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>2435148</td>\n",
       "      <td>Single</td>\n",
       "      <td>372</td>\n",
       "      <td>14</td>\n",
       "      <td>[('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 rcbarcode  nTot    n scaffold strand  \\\n",
       "barcode                                                                 \n",
       "TTCATTCAGGATCTTTGCTC  GAGCAAAGATCCTGAATGAA  154   154  chr1     +       \n",
       "ACGTATTCGCAGTCCTCATA  TATGAGGACTGCGAATACGT  21    21   chr1     -       \n",
       "GCGGCGGCAGCGTTCCTCGC  GCGAGGAACGCTGCCGCCGC  150   150  chr1     -       \n",
       "GAATAGGTAATTAAGCCTGC  GCAGGCTTAATTACCTATTC  62    62   chr1     +       \n",
       "TTCGCGTGGGTGAATTAACT  AGTTAATTCACCCACGCGAA  415   415  chr5     +       \n",
       "\n",
       "                          pos    type  nMainLocation  nInsert  \\\n",
       "barcode                                                         \n",
       "TTCATTCAGGATCTTTGCTC  8764372  Single  144            2         \n",
       "ACGTATTCGCAGTCCTCATA  526973   Single  21             0         \n",
       "GCGGCGGCAGCGTTCCTCGC  2235514  Single  140            2         \n",
       "GAATAGGTAATTAAGCCTGC  8013504  Single  54             3         \n",
       "TTCGCGTGGGTGAATTAACT  2435148  Single  372            14        \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      All genomic mappings  \\\n",
       "barcode                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "TTCATTCAGGATCTTTGCTC  [('chr1:8764372:+', 144), ('chr6:477092:-', 1), ('chr7:1928216:+', 1), ('chr4:2631691:-', 1), ('chr3:313312:+', 1), ('chr6:733284:-', 1), ('chr1:4277183:-', 1), ('chr1:1979828:+', 1), ('chr2:991296:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "ACGTATTCGCAGTCCTCATA  [('chr1:526973:-', 21)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                \n",
       "GCGGCGGCAGCGTTCCTCGC  [('chr1:2235514:-', 140), ('chr2:2382780:-', 1), ('chr3:2906544:-', 1), ('chr2:2797042:-', 1), ('chr3:1386264:-', 1), ('chr1:9117601:+', 1), ('chr1:2235513:-', 1), ('chr2:1023360:+', 1), ('chr5:3144776:-', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                      \n",
       "GAATAGGTAATTAAGCCTGC  [('chr1:8013504:+', 54), ('chr5:3325007:-', 2), ('chr4:3047278:-', 1), ('chr2:2715304:-', 1), ('chr1:9310056:+', 1)]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   \n",
       "TTCGCGTGGGTGAATTAACT  [('chr5:2435148:+', 372), ('chr1:9415816:+', 3), ('chr1:7572780:-', 1), ('chr2:1803458:-', 1), ('chr5:2435205:-', 1), ('chr5:2574087:-', 1), ('chr3:2141768:+', 1), ('chr7:2035049:-', 1), ('chr5:3325007:-', 1), ('chr7:2775803:+', 1), ('chr3:2410718:-', 1), ('chr3:2618659:-', 1), ('chr1:6943496:-', 1), ('chr2:4994449:+', 1), ('chr2:4661375:+', 1), ('chr7:1074090:+', 1), ('chr1:9554108:-', 1), ('chr3:2949780:+', 1), ('chr7:1049234:+', 1), ('chr6:3125324:-', 1), ('chr6:201290:+', 1), ('chr2:4512077:+', 1), ('chr5:2435149:+', 1), ('chr2:1751841:-', 1), ('chr1:6903684:-', 1), ('chr7:2406660:-', 1), ('chr1:391550:-', 1), ('chr2:3965159:+', 1)]   \n",
       "\n",
       "                                   gene  \n",
       "barcode                                  \n",
       "TTCATTCAGGATCTTTGCTC  proteinId=2298894  \n",
       "ACGTATTCGCAGTCCTCATA                     \n",
       "GCGGCGGCAGCGTTCCTCGC                     \n",
       "GAATAGGTAATTAAGCCTGC  proteinId=97899    \n",
       "TTCGCGTGGGTGAATTAACT                     "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check that a 'gene' column has been added to df\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### This line is not necessary in the running. Continue to the next line to save the dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print('{} barcodes not in coding sequences'.format(df[df['gene']==''].shape[0]))\n",
    "print()\n",
    "\n",
    "print('{} barcodes in coding sequences'.format(df[df['gene']!=''].shape[0]))\n",
    "print('{} barcodes in Kl coding sequences'.format(df[df['gene'].str.startswith('kl',na=False)].shape[0]))\n",
    "print('{} barcodes in Km coding sequences'.format(df[df['gene'].str.startswith('km',na=False)].shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(annotated_poolfile_path, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "gff[gff['ID']=='kmYHR034C']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "solaenv",
   "language": "python",
   "name": "solaenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
