{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import Seq\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Change the following parameters to your output directory (where you want output files to be written), your metafile, and TNseq output. \n",
    "\n",
    "If these files aren't in the same directory as this notebook, you need use the full path. For example, the full path of 'barseq_output' would be '/usr2/people/clairedubin/barseq/barseq_output'\n",
    "\n",
    "Also change the control and experimental temperatures if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where the BarSeq FASTQs are\n",
    "FASTQ_directory = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Barseq metafile\n",
    "\n",
    "You'll need to create your own metafile - easiest way is edit in Excel. A sample metafile is located at: /usr2/people/clairedubin/barseq/Kluyv_BarSeq_metadata.txt\n",
    "\n",
    "    Fastq: path to barseq FASTQ for each competition\n",
    "    SampleName: unique name for each competition, MUST include ctrl_temp or exp_temp as specified above\n",
    "    DualIndex: index from sequencing (in read name lines of FASTQ and the sequencing info file that Adam sends)\n",
    "    \n",
    "    UsePrecounted: True if you've already run the Barseq counting script on a competition, False otherwise\n",
    "    \n",
    "    poolfile: path to Tnseq output file, should be the same for each competition\n",
    "    output_dir: path to output directory\n",
    "    minRandom: number of bases between start of read and DualIndex, usually in sequence of 1,2,3,4,1,2,3,4 for each sample\n",
    "    maxRandom: same as minRandom\n",
    "    \n",
    "    BeforeBarcode: sequence preceding all barcodes\n",
    "    AfterBarcode: sequence following all barcodes\n",
    "    BarcodeLengths: lengths of barcodes to search for, with expected length of barcode first\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A2_S2_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B1_S3_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B2_S4_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A1_S5_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B1_S7_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B2_S8_L001_R1_001.fastq\r\n"
     ]
    }
   ],
   "source": [
    "!ls {FASTQ_directory}/*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format: sample_name, fastq_path, dual_index, offset\n",
    "#add or delete more lines as needed\n",
    "\n",
    "sample_info = [('50_A2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq', 'CAGATC', '3')\n",
    "               ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@VH00771:102:AAGC27TM5:1:1101:63373:1038 1:N:0:ATTACG\r\n",
      "CNCACTAGTCGACCTGCAGCGTACGGTGCAACAAACAGTATGCGCAGAGACCTCGTGGACATCAGATCGGAAGAGCACATGTTTGGATTCTAGTCACATTACGATTTGGGGGGGGGGGGGGGGGGGGGGGGG\r\n",
      "+\r\n",
      "I#IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII-II9IIIIIIIIII-III99I-9--9I-9----999I99-IIIII\r\n",
      "@VH00771:102:AAGC27TM5:1:1101:63449:1038 1:N:0:ATTACG\r\n",
      "GNCACTAGTCGACCTGCAGCGTACGTAGGCAACCACACGAGCCGTAGAGACCTCGTGGACATCAGATCGGAAGAGCACATGTTTGGACTCTAGTCATATTACGATTTGGGGGGGGGGGGGGGGGGGGGGGGG\r\n",
      "+\r\n",
      "I#IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII-IIIIIIIIIIIIIIIIIIIIIIII9II9IIIII-IIII9II99-9I99-99------9--II-9I9III\r\n",
      "@VH00771:102:AAGC27TM5:1:1101:63525:1038 1:N:0:ATTACG\r\n",
      "TNCACTAGTCGACCTGCAGCGTACGATACGAAATTATGAGCTGTCAGAGACCTCGTGGACATCAGATCGGAAGAGCACATGTTTGGACTCTAGTCACATTACGATTTGGGGGGGGGGGGGGGGGGGGGGGGG\r\n"
     ]
    }
   ],
   "source": [
    "!head /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To convert the annotated file to csv and not to be separated by tab (execute the next three lines only if necessary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "poolfile1 = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_13k_mutants_mapping_output_all_fastqs_available/Tnseq_13k_mutants_all_digestion_time_poolfile_annotated.csv'\n",
    "df=pd.read_csv(poolfile1,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_13k_mutants_mapping_output_all_fastqs_available/Tnseq_13k_mutants_all_digestion_time_poolfile_annotated_1.csv',index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_13k_mutants_mapping_output_all_fastqs_available/Tnseq_13k_mutants_all_digestion_time_poolfile_annotated_1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these variables should be the same for each replicate\n",
    "\n",
    "metafile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping/BarSeq_metafile_13k_mutants_no_mapping.txt'\n",
    "logfile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping/BarSeq_metafile_13k_mutants_no_mapping.log'\n",
    "poolfile = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_13k_mutants_mapping_output_all_fastqs_available/Tnseq_13k_mutants_all_digestion_time_poolfile_annotated_1.csv'\n",
    "output_dir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping'\n",
    "before_barcode = 'GTCGACCTGCAGCGTACG' #bottom strand sequence was used\n",
    "#before_barcode = 'GATGTCCACGAGGTCTCT' #what was there before \n",
    "after_barcode =  'AGAGACCTCGTGGACATC' #bottom strand sequence was used\n",
    "#after_barcode = 'CGTACGCTGCAGGTCGAC'  #what was there before \n",
    "barcode_lengths = '20,19,18,21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if there are any samples where you already counted barcodes, add the sample names here\n",
    "precounted_samples = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['FileIndex','SampleName', 'Fastq', 'DualIndex', 'Poolfile', 'OutputDir', 'minRandom', 'maxRandom',\n",
    "                   'BeforeBarcode', 'AfterBarcode', 'BarcodeLengths', 'UsePrecounted']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for sample_name, fastq, dual_index, offset in sample_info:\n",
    "        \n",
    "        #DUAL INDECES ARE BACKWARDS FROM WHAT THE FLOWCELL OUTPUT SAYS\n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index[::-1], poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'False'])\n",
    "        f.write(to_write+'\\n')\n",
    "        \n",
    "    for precounted_sample_name in precounted_samples:\n",
    "        \n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index, poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'True'])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileIndex</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>DualIndex</th>\n",
       "      <th>Poolfile</th>\n",
       "      <th>OutputDir</th>\n",
       "      <th>minRandom</th>\n",
       "      <th>maxRandom</th>\n",
       "      <th>BeforeBarcode</th>\n",
       "      <th>AfterBarcode</th>\n",
       "      <th>BarcodeLengths</th>\n",
       "      <th>UsePrecounted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>50_A2</td>\n",
       "      <td>50_A2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq</td>\n",
       "      <td>CTAGAC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_13k_mutants_mapping_output_all_fastqs_available/Tnseq_13k_mutants_all_digestion_time_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  FileIndex SampleName  \\\n",
       "0  50_A2     50_A2       \n",
       "\n",
       "                                                                                                                                                                                            Fastq  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq   \n",
       "\n",
       "  DualIndex  \\\n",
       "0  CTAGAC     \n",
       "\n",
       "                                                                                                                                                                                                                                          Poolfile  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_13k_mutants_mapping_output_all_fastqs_available/Tnseq_13k_mutants_all_digestion_time_poolfile_annotated_1.csv   \n",
       "\n",
       "                                                                                                                                                                                                                   OutputDir  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping   \n",
       "\n",
       "   minRandom  maxRandom       BeforeBarcode        AfterBarcode  \\\n",
       "0  3          3          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "\n",
       "  BarcodeLengths  UsePrecounted  \n",
       "0  20,19,18,21    False          "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Barseq counting\n",
    "\n",
    "Might take a few hours!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['TTCG', 'CTCG', 'GTCG', 'AACG', 'ACCG', 'AGCG', 'ATAG', 'ATTG', 'ATGG', 'ATCA', 'ATCT', 'ATCC']\n",
      "  Counting barcodes in /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq\n",
      "  Estimated sequencing error rate for barcodes: 0.5%\n",
      "  Barcodes seen once: 54058189, twice: 1175829, three times or more: 1721224\n",
      "  Chao estimate of population size: 1242649993.3067312\n",
      "Saving summary statistics for fastqs to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping/fastqSummaryStats.txt\n"
     ]
    }
   ],
   "source": [
    "!python3 /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/no_mapping/RBseq_Count_BarCodes_without_tnseq_poolfile.py -m {metafile_name} -l {logfile_name}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB0029_37A1'\n",
    "y_column = 'KFRB0029_37B1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB0029_37A Abundance', fontsize=11)\n",
    "plt.ylabel('KFRB0029_37B Abundance', fontsize=11)\n",
    "plt.title('Plot of TN_insertion abundances at 37 oC (Bio reps)', fontsize=13, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_01_03_2024_37oC.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB0029_37A1'\n",
    "y_column = 'KFRB0029_37B1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB0029_37A Abundance', fontsize=11)\n",
    "plt.ylabel('KFRB0029_37B Abundance', fontsize=11)\n",
    "plt.title('Plot of TN_insertion abundances at 37 oC (Bio reps)', fontsize=13, color='purple')\n",
    "\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 3000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 3000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_37oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## to normalize the Barseq pool count file based on the number of reads obtained"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### this is the normalization that worked with the right result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_new_37•C_with_older_50•C_1.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_37•C (A)'] = df['37•C (A)'].sum()\n",
    "df['Total_Count_37•C (B)'] = df['37•C (B)'].sum()\n",
    "df['Total_Count_45•C (A)'] = df['45•C (A)'].sum()\n",
    "df['Total_Count_45•C (B)'] = df['45•C (B)'].sum()\n",
    "df['Total_Count_45•C (C)'] = df['45•C (C)'].sum()\n",
    "df['Total_Count_45•C (D)'] = df['45•C (D)'].sum()\n",
    "df['Total_Count_50•C (A)'] = df['50•C (A)'].sum()\n",
    "df['Total_Count_50•C (B)'] = df['50•C (B)'].sum()\n",
    "df['Total_Count_50•C (C)'] = df['50•C (C)'].sum()\n",
    "df['Total_Count_50•C (D)'] = df['50•C (D)'].sum()\n",
    "\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['normalized_37•C (A)'] = df['37•C (A)'] / df['Total_Count_37•C (A)']\n",
    "df['normalized_37•C (B)'] = df['37•C (B)'] / df['Total_Count_37•C (B)']\n",
    "df['normalized_45•C (A)'] = df['45•C (A)'] / df['Total_Count_45•C (A)']\n",
    "df['normalized_45•C (B)'] = df['45•C (B)'] / df['Total_Count_45•C (B)']\n",
    "df['normalized_45•C (C)'] = df['45•C (C)'] / df['Total_Count_45•C (C)']\n",
    "df['normalized_45•C (D)'] = df['45•C (D)'] / df['Total_Count_45•C (D)']\n",
    "df['normalized_50•C (A)'] = df['50•C (A)'] / df['Total_Count_50•C (A)']\n",
    "df['normalized_50•C (B)'] = df['50•C (B)'] / df['Total_Count_50•C (B)']\n",
    "df['normalized_50•C (C)'] = df['50•C (C)'] / df['Total_Count_50•C (C)']\n",
    "df['normalized_50•C (D)'] = df['50•C (D)'] / df['Total_Count_50•C (D)']\n",
    "\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_normalized.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To make the fitness abundance agaian using the normalized values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the CSV file\n",
    "csv_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_normalized.csv'\n",
    "df = pd.read_csv(csv_file_path)\n",
    "\n",
    "# Save the DataFrame to a text file (txt)\n",
    "txt_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_normalized.txt'\n",
    "df.to_csv(txt_file_path, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_normalized.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'normalized_37•C (A)'\n",
    "y_column = 'normalized_37•C (B)'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('normalized_37•C (A) Abundance', fontsize=11)\n",
    "plt.ylabel('Normalized_37•C (B) Abundance', fontsize=11)\n",
    "plt.title('Plot of TN_insertion abundances_37 oC_normalized (A)', fontsize=13, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.04)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.04)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_normalized.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'normalized_37•C (A)'\n",
    "y_column = 'normalized_37•C (B)'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('normalized_37•C (A) Abundance', fontsize=11)\n",
    "plt.ylabel('Normalized_37•C (B) Abundance', fontsize=11)\n",
    "plt.title('Plot of TN_insertion abundances_37 oC_normalized (A)', fontsize=13, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.015)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.015)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_06_12_2024/normalized_combined_fastqs_barseq_pool_counts_06_12_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB021_1'\n",
    "y_column = 'KFRB022_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB021_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB022_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC_normalized (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_06_12_2024/normalized_combined_fastqs_barseq_pool_counts_06_12_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB023_1'\n",
    "y_column = 'KFRB024_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB023_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB024_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC_normalized (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_06_12_2024/normalized_combined_fastqs_barseq_pool_counts_06_12_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB025_1'\n",
    "y_column = 'KFRB026_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB025_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB026_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_50 oC_normalized (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_06_12_2024/normalized_combined_fastqs_barseq_pool_counts_06_12_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB027_1'\n",
    "y_column = 'KFRB028_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB027_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB028_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_50 oC_normalized (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# To do statistical t-test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To handle rows where some values are zero and convert them to NaN before performing the Mann-Whitney U test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import mannwhitneyu\n",
    "\n",
    "# Load the dataset\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_normalized.csv')\n",
    "\n",
    "# Convert all columns to numeric, coercing errors to NaN\n",
    "for col in df.columns:\n",
    "    df[col] = pd.to_numeric(df[col], errors='coerce')\n",
    "\n",
    "# Replace zeros with NaN\n",
    "df.replace(0, np.nan, inplace=True)\n",
    "\n",
    "# Define the data columns for the existing and new conditions\n",
    "data_37C = df[['37•C (A)', '37•C (B)']]\n",
    "data_45C = df[['45•C (A)', '45•C (B)', '45•C (C)', '45•C (D)']]\n",
    "data_50C = df[['50•C (A)', '50•C (B)', '50•C (C)', '50•C (D)']]\n",
    "            \n",
    "\n",
    "\n",
    "\n",
    "# List to store p-values for existing conditions\n",
    "p_values_37_vs_45 = []\n",
    "p_values_37_vs_50 = []\n",
    "\n",
    "# Iterate over rows and perform Mann-Whitney U test for each\n",
    "for index, row in df.iterrows():\n",
    "    # Drop NaN values from the datasets\n",
    "    data_37C_values = row[data_37C.columns].dropna()\n",
    "    data_45C_values = row[data_45C.columns].dropna()\n",
    "    \n",
    "    # Perform the Mann-Whitney U test for 37C vs. 45C\n",
    "    if len(data_37C_values) > 0 and len(data_45C_values) > 0:\n",
    "        statistic, p_value = mannwhitneyu(data_37C_values, data_45C_values, alternative='two-sided')\n",
    "        p_values_37_vs_45.append(p_value)\n",
    "    else:\n",
    "        p_values_37_vs_45.append(np.nan)\n",
    "\n",
    "    # Perform the Mann-Whitney U test for 37C vs. 50C\n",
    "    #data_37C_values = row[data_25C.columns].dropna()\n",
    "    data_50C_values = row[data_50C.columns].dropna()\n",
    "\n",
    "    if len(data_37C_values) > 0 and len(data_50C_values) > 0:\n",
    "        statistic, p_value = mannwhitneyu(data_37C_values, data_50C_values, alternative='two-sided')\n",
    "        p_values_37_vs_50.append(p_value)\n",
    "    else:\n",
    "        p_values_37_vs_50.append(np.nan)\n",
    "\n",
    "# Add the p-values to the DataFrame\n",
    "df['MannWhitneyU_p_value_37_vs_45'] = p_values_37_vs_45\n",
    "df['MannWhitneyU_p_value_37_vs_50'] = p_values_37_vs_50\n",
    "\n",
    "# Save the DataFrame with the calculated p-values to a new CSV file\n",
    "output_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_biorep_all_temp_output_with_mannwhitneyu_p_values_zero_values_removed_combined.csv'\n",
    "df.to_csv(output_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
