{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Note: the mapping script takes days to run! You should be running this notebook using screen, so if you close this page the script will keep running.\n",
    "\n",
    "If you close the notebook while using screen, the log output within the notebook might stop being printed. But the logfile specified in the line where you call the script will keep being written.\n",
    "\n",
    "How to run a Jupyter notebook with screen: https://docs.google.com/document/d/1lLcl3HhDBrn87M4-IhkpDLOmvmmf09-KeLSgn3U21hE/edit?usp=sharing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make metafile for TNseq mapping script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Info that will be the same for all pools\n",
    "\n",
    "metafile_name: Name of file where we're going to write the metadata\n",
    "\n",
    "combined_pool_name: What we're going to call the combination of individual pools\n",
    "\n",
    "ReadModel: Full or relative path to a text file describing the expected structure of TnSeq reads.  The file consists of 4 lines.  Line 1 denotes the number of initial variable bases to ignore as 'N's.  Line 2 is the sequence preceding the barcode.  Line 3 is the expected length of sequence barcode denoted as 'N's.  Line 4 is the sequence between the barcode and the junction with the genome.  Example:\n",
    "\n",
    "    nnnnnn\n",
    "    GATGTCCACGAGGTCTCT\n",
    "    NNNNNNNNNNNNNNNNNNNN\n",
    "    CGTACGCTGCAGGTCGACAATGATCCAAACTATCAGTGTTTGA\t\n",
    "\n",
    "InsertionSequence: Full or relative path to a text file describing giving the full sequence of each insertion.  This should be a fasta format file with two entries labeled 'insert' and 'plasmid' giving the full sequence of the expected insertion into the genome and and the remaining source plasmid sequence, if appropriate. Example:\n",
    "\n",
    "    >insert\n",
    "    agtcac...\n",
    "    >plasmid\n",
    "    actgact... \n",
    "      \n",
    "GenomeSequence: Full or relative path to a .fasta file of genome sequences for all species in pool.\n",
    "\n",
    "OutputDir: Full or relative path to the desired directory to save output files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_metafile_13k_mutants_Mar_22_2024.txt\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_13k_mutants_Mar_22_2024.log\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/Readmodel_file\n"
     ]
    }
   ],
   "source": [
    "#CHANGE THESE AS NEEDED\n",
    "\n",
    "metafile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_metafile_13k_mutants_Mar_22_2024.txt'\n",
    "mapping_log_file = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_13k_mutants_Mar_22_2024.log'\n",
    "\n",
    "ReadModel = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/Readmodel_file'\n",
    "InsertionSequence = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/insertion_seq_file'\n",
    "GenomeSequence = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/TnSeq_Results_may_2023/TnSeq_M003593/analysed_files/Myceliopthora_genome_with_vector_seq.txt'\n",
    "OutputDir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence'\n",
    "print(metafile_name)\n",
    "print(mapping_log_file)\n",
    "print(ReadModel)\n",
    "#print(metafile_name, mapping_log_file, ReadModel)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pool-specific information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where you saved the merged FASTQs in TNseq_prepare_files.ipynb\n",
    "FASTQ_directory = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/TNSeq_merge_FASTQ/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/TNSeq_merge_FASTQ//KFRB010_S1_L001_merged.fastq\r\n"
     ]
    }
   ],
   "source": [
    "#get the list of fastqs we generated in the BBmerge step above\n",
    "#copy and paste these into the cell below this one\n",
    "\n",
    "!ls {FASTQ_directory}/*_merged.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each pool, come up with a UNIQUE short name for the pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED BUT KEEP THE [('NAME', 'PATH/TO/FASTQ.FASTQ')] FORMAT\n",
    "\n",
    "short_names_and_fastqs = [('Tnseq_13k_mutants_Mar_22_2024', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/TNSeq_merge_FASTQ/KFRB010_S1_L001_merged.fastq')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Writing the above info into a metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['Pool', 'ShortName', 'Fastq', 'ReadModel', 'InsertionSequence', 'GenomeSequence', 'OutputDir']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for pool, fastq in short_names_and_fastqs:\n",
    "        \n",
    "        #we write the pool name twice because we're first processing each pool separately then combining them\n",
    "        #if we were combining them all at once then the first pool variable would be the name of the combined pool\n",
    "        to_write = '\\t'.join([pool, pool, fastq, ReadModel, InsertionSequence, GenomeSequence, OutputDir])\n",
    "        f.write(to_write+'\\n')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pool</th>\n",
       "      <th>ShortName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>ReadModel</th>\n",
       "      <th>InsertionSequence</th>\n",
       "      <th>GenomeSequence</th>\n",
       "      <th>OutputDir</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tnseq_13k_mutants_Mar_22_2024</td>\n",
       "      <td>Tnseq_13k_mutants_Mar_22_2024</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/TNSeq_merge_FASTQ/KFRB010_S1_L001_merged.fastq</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/Readmodel_file</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/insertion_seq_file</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/TnSeq_Results_may_2023/TnSeq_M003593/analysed_files/Myceliopthora_genome_with_vector_seq.txt</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_genome</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            Pool                      ShortName  \\\n",
       "0  Tnseq_13k_mutants_Mar_22_2024  Tnseq_13k_mutants_Mar_22_2024   \n",
       "\n",
       "                                                                                                                                      Fastq  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/TNSeq_merge_FASTQ/KFRB010_S1_L001_merged.fastq   \n",
       "\n",
       "                                                                                                              ReadModel  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/Readmodel_file   \n",
       "\n",
       "                                                                                                          InsertionSequence  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/insertion_seq_file   \n",
       "\n",
       "                                                                                                                                           GenomeSequence  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/TnSeq_Results_may_2023/TnSeq_M003593/analysed_files/Myceliopthora_genome_with_vector_seq.txt   \n",
       "\n",
       "                                                                                                                                                    OutputDir  \n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_genome  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#view metafile\n",
    "pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_metafile_13k_mutants_Mar_22_2024.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Map transposons in each pool to genome\n",
    "\n",
    "View additional input and output options here: https://github.com/stcoradetti/RBseq/blob/master/README.txt\n",
    "\n",
    "This takes a long time to run and will be stuck on the \"Looking for 20bp sequence barcode\" step for a while. You can see how many barcodes have been processed by switching to the output directory specified above and typing:\n",
    "\n",
    "    wc -l *blastquery.fasta\n",
    "    \n",
    "This number divided by 2 is roughly how many reads have been processed. You can also check the logFile specified below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "env: PATH=$PATH/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages/:/usr2/people/shollyt22/anaconda3/bin:/usr/lib64/qt-3.3/bin:/usr/condabin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr2/people/kayleeec/bin/ncbi-blast-2.13.0+/bin:/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages:/usr2/people/kayleeec/bin/ncbi-blast-2.13.0+/bin:/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages:/usr2/people/kayleeec/bin/ncbi-blast-2.13.0+/bin:/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages/\n"
     ]
    }
   ],
   "source": [
    "#always run this since the server couldnt read directory with the blast package from the local \n",
    "%env PATH=$PATH/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages/:/usr2/people/shollyt22/anaconda3/bin:/usr/lib64/qt-3.3/bin:/usr/condabin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/usr2/people/kayleeec/bin/ncbi-blast-2.13.0+/bin:/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages:/usr2/people/kayleeec/bin/ncbi-blast-2.13.0+/bin:/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages:/usr2/people/kayleeec/bin/ncbi-blast-2.13.0+/bin:/auto/sahara/namib/home/shollyt22/anaconda3/lib/python3.9/site-packages/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RBseq_Map_Insertions.py\n",
      "2024-03-22 18:30:38 Version: 1.1.4_PBa\n",
      "2024-03-22 18:30:38 Release Date: August 8, 2020\n",
      "2024-03-22 18:30:38 Options passed:  metafile:/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_metafile_13k_mutants_Mar_22_2024.txt  logFile:/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_13k_mutants_Mar_22_2024.log  minQual:10  matchBefore:6  matchAfter:6  matchJunction:4  maxFillerSeq:100  barcodeVariation:2  wobbleAllowed:1  scoreDiff:10  minFraction:0.9  filterNeighborhood:10  filterEditDistance:5  minPercentID:95  maxEvalue:0.1  useMappedFiles:False  noInsertHits:False  noBarcodes:False \n",
      "2024-03-22 18:30:38 Logging status updates in /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_13k_mutants_Mar_22_2024.log\n",
      "2024-03-22 18:30:38 Loading TnSeq library metadata from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_mapping_metafile_13k_mutants_Mar_22_2024.txt\n",
      "2024-03-22 18:30:38 Finding barcodes in fastqs and mapping insertion locations\n",
      "2024-03-22 18:30:38   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/TNSeq_merge_FASTQ/KFRB010_S1_L001_merged.fastq using insertion model /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/Readmodel_file\n",
      "2024-03-22 18:30:38   Looking for 20bp sequence barcode flanked by GTCTCT and CGTACG\n",
      "2024-03-22 18:31:08     196191 reads without recognizable, compliant barcodes written to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_noBarcodeFound.txt\n",
      "2024-03-22 18:31:08     332899 query sequences with compliant barcodes written to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_blastquery.fasta\n",
      "2024-03-22 18:31:08   BLASTING reads against /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/TnSeq_Results_may_2023/TnSeq_M003593/analysed_files/Myceliopthora_genome_with_vector_seq.txt\n",
      "2024-03-22 18:31:32     76566 Reads mapped to genome.\n",
      "2024-03-22 18:31:32   BLASTING reads against full insertion sequence in /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/insertion_seq_file\n",
      "2024-03-22 18:31:43     3 Reads mapped to insertion/plasmid sequence.\n",
      "2024-03-22 18:31:43 Processing mapped reads to characterize mutant pool(s)\n",
      "2024-03-22 18:31:43 Analyzing pool: Tnseq_13k_mutants_Mar_22_2024\n",
      "2024-03-22 18:31:43   Reading entries from: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_mapped.txt\n",
      "2024-03-22 18:31:44     76569 lines read from 1 files\n",
      "2024-03-22 18:31:44   Masking out off-by-one barcodes. (Most likely sequencing errors)\n",
      "2024-03-22 18:31:44     Wrote 297 off-by-one barcodes (397 reads) to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile_offByOne.txt\n",
      "2024-03-22 18:31:44   Processing barcodes with mapped genomic locations\n",
      "2024-03-22 18:31:44     Wrote 8561 barcodes (39023 reads) to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile\n",
      "2024-03-22 18:31:44     Wrote 2022 barcodes (36025 reads) to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile_multiLocus.txt\n",
      "2024-03-22 18:31:44     Wrote 130 barcodes (1122 reads) to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile_ambiguous.txt\n",
      "2024-03-22 18:31:44   Processing hits to insert only\n",
      "2024-03-22 18:31:44     Wrote 2 barcodes (2 reads) to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile_insertHitsOnly.txt\n",
      "2024-03-22 18:31:44   Finding similar barcodes that mapped to the same location (off-by-two, indels, etc).\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_1\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_2\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_3\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_4\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_5\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_6\n",
      "2024-03-22 18:31:44     Checking scaffold chromosome_7\n",
      "2024-03-22 18:31:45     Checking scaffold vector_seq\n",
      "2024-03-22 18:34:23     Moved 372 barcodes with likely sequencing errors to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile_filteredLocal\n",
      "2024-03-22 18:34:23     Wrote 8189 barcodes (39023 reads) to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output_with_vector_sequence/Tnseq_13k_mutants_Mar_22_2024_poolfile\n"
     ]
    }
   ],
   "source": [
    "cmd = 'nice -n 17 python3 /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/RBseq_Map_Insertions_v1.1.4_PBa_Jskerker.py --metafile {} --logFile {}'.format(metafile_name, mapping_log_file)\n",
    "\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Combine pools"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a new metafile, this time combining the pools into one larger pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED\n",
    "\n",
    "combined_pool_name = 'POOLS30_to_41_combined'\n",
    "metafile_name = 'TNSeq_mapping_metafile_pools30_to_41_combined.txt'\n",
    "mapping_log_file = 'TNSeq_mapping_pools30_to_41_combined.log'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['Pool', 'ShortName', 'Fastq', 'ReadModel', 'InsertionSequence', 'GenomeSequence', 'OutputDir']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for pool, fastq in short_names_and_fastqs:\n",
    "        \n",
    "        #the pool name is now the combined pool name\n",
    "        to_write = '\\t'.join([combined_pool_name, pool, fastq, ReadModel, InsertionSequence, GenomeSequence, OutputDir])\n",
    "        f.write(to_write+'\\n')\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note how this metafile is different than the one above - the \"Pool\" column is the name of the combined pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#view metafile\n",
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This step takes ~ 2 hours for 4 pools."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "cmd = 'nice -n 17 python3 /usr2/people/clairedubin/barseq/latest_pipeline/RBseq_Map_Insertions_v1.1.4_PBa_Jskerker.py --useMappedFiles --metafile {} --logFile {}'.format(metafile_name, mapping_log_file)\n",
    "\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##The next command below was executed for reformatting the insertionSequence file required for mapping transposon in previous step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for reformatting the insertionSequence file required for mapping transposon\n",
    "!makeblastdb -in {InsertionSequence} -dbtype nucl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "solaenv",
   "language": "python",
   "name": "solaenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
