{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import Seq\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Change the following parameters to your output directory (where you want output files to be written), your metafile, and TNseq output. \n",
    "\n",
    "If these files aren't in the same directory as this notebook, you need use the full path. For example, the full path of 'barseq_output' would be '/usr2/people/clairedubin/barseq/barseq_output'\n",
    "\n",
    "Also change the control and experimental temperatures if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where the BarSeq FASTQs are\n",
    "FASTQ_directory = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Barseq metafile\n",
    "\n",
    "You'll need to create your own metafile - easiest way is edit in Excel. A sample metafile is located at: /usr2/people/clairedubin/barseq/Kluyv_BarSeq_metadata.txt\n",
    "\n",
    "    Fastq: path to barseq FASTQ for each competition\n",
    "    SampleName: unique name for each competition, MUST include ctrl_temp or exp_temp as specified above\n",
    "    DualIndex: index from sequencing (in read name lines of FASTQ and the sequencing info file that Adam sends)\n",
    "    \n",
    "    UsePrecounted: True if you've already run the Barseq counting script on a competition, False otherwise\n",
    "    \n",
    "    poolfile: path to Tnseq output file, should be the same for each competition\n",
    "    output_dir: path to output directory\n",
    "    minRandom: number of bases between start of read and DualIndex, usually in sequence of 1,2,3,4,1,2,3,4 for each sample\n",
    "    maxRandom: same as minRandom\n",
    "    \n",
    "    BeforeBarcode: sequence preceding all barcodes\n",
    "    AfterBarcode: sequence following all barcodes\n",
    "    BarcodeLengths: lengths of barcodes to search for, with expected length of barcode first\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB007_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB008_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB019_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB020_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB021_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB022_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB023_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB024_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB025_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB026_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB027_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB028_S1_L001_R1_001.fastq\r\n"
     ]
    }
   ],
   "source": [
    "!ls {FASTQ_directory}/*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format: sample_name, fastq_path, dual_index, offset\n",
    "#add or delete more lines as needed\n",
    "\n",
    "sample_info = [('KFRB007_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB007_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB008_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB008_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('KFRB019_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB019_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB020_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB020_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('KFRB021_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB021_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB022_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB022_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('KFRB023_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB023_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB024_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB024_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('KFRB025_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB025_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB026_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB026_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('KFRB027_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB027_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB028_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB028_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@M07716:226:000000000-DLCF9:1:1101:15739:1331 1:N:0:1\r\n",
      "CATGTAGCGTCGACCTGCAGCGTACGTGAATGAGGGTAAGTGAGAAAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "BBBCCFFFDDCCGGGGGGGGGGGGGGHHHGHHHGHGGHHHHHGHHHHHHHHHGHHGHHGGHHHHHHHHHHGGGGHHHHHHHGHHHGHHHHHHHHHHHHHGGGGHHHHHHHGGHHHHGGGHGHHHHHHHHHHHHGGGGG><<\r\n",
      "@M07716:226:000000000-DLCF9:1:1101:15765:1331 1:N:0:1\r\n",
      "TGTGTAGCGTCGACCTGCAGCGTACGACCACGCAGCTTCCAAGACCAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "CCCCCFFFCCCCGGGGGGGGGGGGGGGGGGHGGGGGHHHHHHHHHHHHHHHHGHHGHHGGHHGHHHHHHHGGGGHHHHHHHGHHHGHHHHHHHHHHHHHGEGGHHHHHHHGGHHHHGGGHGHHHHHHHGHHHHGGGGG<:;\r\n",
      "@M07716:226:000000000-DLCF9:1:1101:15802:1332 1:N:0:1\r\n",
      "AATGTAGCGTCGACCTGCAGCGTACGTCCGCTGAGCCCTCAACGGGAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n"
     ]
    }
   ],
   "source": [
    "!head /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB020_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To convert the annotated file to csv and not to be separated by tab (execute the next three lines only if necessary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "poolfile1 = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated.csv'\n",
    "df=pd.read_csv(poolfile1,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv',index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these variables should be the same for each replicate\n",
    "\n",
    "metafile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/BarSeq_metafile_13k_mutants_06_09_2024.txt'\n",
    "logfile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/BarSeq_counting_13k_mutants_06_09_2024.log'\n",
    "poolfile = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv'\n",
    "output_dir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification'\n",
    "before_barcode = 'GTCGACCTGCAGCGTACG' #bottom strand sequence was used\n",
    "#before_barcode = 'GATGTCCACGAGGTCTCT' what was there before \n",
    "after_barcode =  'AGAGACCTCGTGGACATC' #bottom strand sequence was used\n",
    "#after_barcode = 'CGTACGCTGCAGGTCGAC'  what was there before \n",
    "barcode_lengths = '20,19,18,21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if there are any samples where you already counted barcodes, add the sample names here\n",
    "precounted_samples = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['FileIndex','SampleName', 'Fastq', 'DualIndex', 'Poolfile', 'OutputDir', 'minRandom', 'maxRandom',\n",
    "                   'BeforeBarcode', 'AfterBarcode', 'BarcodeLengths', 'UsePrecounted']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for sample_name, fastq, dual_index, offset in sample_info:\n",
    "        \n",
    "        #DUAL INDECES ARE BACKWARDS FROM WHAT THE FLOWCELL OUTPUT SAYS\n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index[::-1], poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'False'])\n",
    "        f.write(to_write+'\\n')\n",
    "        \n",
    "    for precounted_sample_name in precounted_samples:\n",
    "        \n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index, poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'True'])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileIndex</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>DualIndex</th>\n",
       "      <th>Poolfile</th>\n",
       "      <th>OutputDir</th>\n",
       "      <th>minRandom</th>\n",
       "      <th>maxRandom</th>\n",
       "      <th>BeforeBarcode</th>\n",
       "      <th>AfterBarcode</th>\n",
       "      <th>BarcodeLengths</th>\n",
       "      <th>UsePrecounted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KFRB007_1</td>\n",
       "      <td>KFRB007_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB007_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KFRB008_1</td>\n",
       "      <td>KFRB008_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB008_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KFRB019_1</td>\n",
       "      <td>KFRB019_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB019_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KFRB020_1</td>\n",
       "      <td>KFRB020_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB020_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KFRB021_1</td>\n",
       "      <td>KFRB021_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB021_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KFRB022_1</td>\n",
       "      <td>KFRB022_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB022_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KFRB023_1</td>\n",
       "      <td>KFRB023_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB023_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KFRB024_1</td>\n",
       "      <td>KFRB024_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB024_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KFRB025_1</td>\n",
       "      <td>KFRB025_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB025_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KFRB026_1</td>\n",
       "      <td>KFRB026_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB026_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>KFRB027_1</td>\n",
       "      <td>KFRB027_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB027_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>KFRB028_1</td>\n",
       "      <td>KFRB028_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB028_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    FileIndex SampleName  \\\n",
       "0   KFRB007_1  KFRB007_1   \n",
       "1   KFRB008_1  KFRB008_1   \n",
       "2   KFRB019_1  KFRB019_1   \n",
       "3   KFRB020_1  KFRB020_1   \n",
       "4   KFRB021_1  KFRB021_1   \n",
       "5   KFRB022_1  KFRB022_1   \n",
       "6   KFRB023_1  KFRB023_1   \n",
       "7   KFRB024_1  KFRB024_1   \n",
       "8   KFRB025_1  KFRB025_1   \n",
       "9   KFRB026_1  KFRB026_1   \n",
       "10  KFRB027_1  KFRB027_1   \n",
       "11  KFRB028_1  KFRB028_1   \n",
       "\n",
       "                                                                                                                                                          Fastq  \\\n",
       "0   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB007_S1_L001_R1_001.fastq   \n",
       "1   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB008_S1_L001_R1_001.fastq   \n",
       "2   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB019_S1_L001_R1_001.fastq   \n",
       "3   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB020_S1_L001_R1_001.fastq   \n",
       "4   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB021_S1_L001_R1_001.fastq   \n",
       "5   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB022_S1_L001_R1_001.fastq   \n",
       "6   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB023_S1_L001_R1_001.fastq   \n",
       "7   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB024_S1_L001_R1_001.fastq   \n",
       "8   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB025_S1_L001_R1_001.fastq   \n",
       "9   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB026_S1_L001_R1_001.fastq   \n",
       "10  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB027_S1_L001_R1_001.fastq   \n",
       "11  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB028_S1_L001_R1_001.fastq   \n",
       "\n",
       "   DualIndex  \\\n",
       "0   GCACTA     \n",
       "1   TGTAGC     \n",
       "2   GCACTA     \n",
       "3   TGTAGC     \n",
       "4   GCACTA     \n",
       "5   TGTAGC     \n",
       "6   GCACTA     \n",
       "7   TGTAGC     \n",
       "8   GCACTA     \n",
       "9   TGTAGC     \n",
       "10  GCACTA     \n",
       "11  TGTAGC     \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                      Poolfile  \\\n",
       "0   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "1   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "2   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "3   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "4   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "5   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "6   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "7   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "8   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "9   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "10  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "11  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv   \n",
       "\n",
       "                                                                                                                                                                                                                             OutputDir  \\\n",
       "0   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "1   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "2   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "3   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "4   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "5   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "6   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "7   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "8   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "9   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "10  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "11  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification   \n",
       "\n",
       "    minRandom  maxRandom       BeforeBarcode        AfterBarcode  \\\n",
       "0   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "1   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "2   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "3   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "4   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "5   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "6   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "7   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "8   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "9   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "10  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "11  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "\n",
       "   BarcodeLengths  UsePrecounted  \n",
       "0   20,19,18,21    False          \n",
       "1   20,19,18,21    False          \n",
       "2   20,19,18,21    False          \n",
       "3   20,19,18,21    False          \n",
       "4   20,19,18,21    False          \n",
       "5   20,19,18,21    False          \n",
       "6   20,19,18,21    False          \n",
       "7   20,19,18,21    False          \n",
       "8   20,19,18,21    False          \n",
       "9   20,19,18,21    False          \n",
       "10  20,19,18,21    False          \n",
       "11  20,19,18,21    False          "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Barseq counting\n",
    "\n",
    "Might take a few hours!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-06-09 17:34:08 RBseq_Count_BarCodes.py\n",
      "2024-06-09 17:34:08 Version: 1.1.4\n",
      "2024-06-09 17:34:08 Release Date: July 16, 2020\n",
      "2024-06-09 17:34:08 Options passed:  metafile:/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/BarSeq_metafile_13k_mutants_06_09_2024.txt  logFile:/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/BarSeq_counting_13k_mutants_06_09_2024.log  minQual:10  matchBefore:6  matchAfter:6  quietMode:False \n",
      "2024-06-09 17:34:08 Logging status updates in /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/BarSeq_counting_13k_mutants_06_09_2024.log\n",
      "2024-06-09 17:34:08 Loading TnSeq library metadata from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/BarSeq_metafile_13k_mutants_06_09_2024.txt\n",
      "2024-06-09 17:34:08 Setting output directory as: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification (from first line of metadata file)\n",
      "2024-06-09 17:34:08 Loading mapped barcodes in mutant pool from: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv (from first line of metadata file)\n",
      "2024-06-09 17:34:08 Read 8521 barcodes from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/Batch_2_sequencing_Apr_26_2024/TNSeq_13k_mutants_mapping_output_without_vector_sequence_jumpstart/TNSeq_13k_mutants_mapping_output_jumpstart_polymerase_Apr_30_2024_2_fastq_combined/Tnseq_13k_mutants_04_30_2024_combined_poolfile_annotated_1.csv\n",
      "2024-06-09 17:34:08 Finding barcodes in fastqs and counting occurrences\n",
      "2024-06-09 17:34:08 ---------------------\n",
      "2024-06-09 17:34:08   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB007_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:08   Filtering reads on second index: GCACTA\n",
      "2024-06-09 17:34:08   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:34:16     1199451 reads processed.\n",
      "2024-06-09 17:34:16     1142096 reads with compliant barcodes.\n",
      "2024-06-09 17:34:16     57355 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:16       26379 reads without expected Dual Index.\n",
      "2024-06-09 17:34:16       24661 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:16       6315 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:16       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:16       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:16   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB007_1.counts\n",
      "2024-06-09 17:34:17   Total barcodes seen (incudes sequencing errors): 104374\n",
      "2024-06-09 17:34:17   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:17     Number of barcodes from poolfile seen: 5710\n",
      "2024-06-09 17:34:17     Reads with barcodes from poolfile: 424355\n",
      "2024-06-09 17:34:17   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 106211 times.\n",
      "2024-06-09 17:34:17   Number of reads that differ from this barcode by one base pair (likely sequencing errors):5274\n",
      "2024-06-09 17:34:17   Estimated sequencing error rate for barcodes: 4.7%\n",
      "2024-06-09 17:34:17   Barcodes seen once (highly inflated by sequencing errors): 77250\n",
      "2024-06-09 17:34:17   Barcodes seen twice (slightly inflated by sequencing errors): 9120\n",
      "2024-06-09 17:34:17   Barcodes seen three times or more: 18004\n",
      "2024-06-09 17:34:17   Chao estimate of population size (ones^2/2*twos): 330000\n",
      "2024-06-09 17:34:17 ---------------------\n",
      "2024-06-09 17:34:17   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB008_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:17   Filtering reads on second index: TGTAGC\n",
      "2024-06-09 17:34:17   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:34:25     1201972 reads processed.\n",
      "2024-06-09 17:34:25     1149363 reads with compliant barcodes.\n",
      "2024-06-09 17:34:25     52609 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:25       19003 reads without expected Dual Index.\n",
      "2024-06-09 17:34:25       25407 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:25       8199 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:25       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:25       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:25   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB008_1.counts\n",
      "2024-06-09 17:34:25   Total barcodes seen (incudes sequencing errors): 108458\n",
      "2024-06-09 17:34:25   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:25     Number of barcodes from poolfile seen: 5799\n",
      "2024-06-09 17:34:25     Reads with barcodes from poolfile: 421067\n",
      "2024-06-09 17:34:25   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 99532 times.\n",
      "2024-06-09 17:34:25   Number of reads that differ from this barcode by one base pair (likely sequencing errors):5905\n",
      "2024-06-09 17:34:25   Estimated sequencing error rate for barcodes: 5.6%\n",
      "2024-06-09 17:34:25   Barcodes seen once (highly inflated by sequencing errors): 79443\n",
      "2024-06-09 17:34:25   Barcodes seen twice (slightly inflated by sequencing errors): 9940\n",
      "2024-06-09 17:34:25   Barcodes seen three times or more: 19075\n",
      "2024-06-09 17:34:25   Chao estimate of population size (ones^2/2*twos): 320000\n",
      "2024-06-09 17:34:25 ---------------------\n",
      "2024-06-09 17:34:25   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB019_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:25   Filtering reads on second index: GCACTA\n",
      "2024-06-09 17:34:25   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:34:29     554288 reads processed.\n",
      "2024-06-09 17:34:29     531401 reads with compliant barcodes.\n",
      "2024-06-09 17:34:29     22887 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:29       12152 reads without expected Dual Index.\n",
      "2024-06-09 17:34:29       7902 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:29       2833 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:29       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:29       17 reads with quality scores less than 10\n",
      "2024-06-09 17:34:29   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB019_1.counts\n",
      "2024-06-09 17:34:29   Total barcodes seen (incudes sequencing errors): 61762\n",
      "2024-06-09 17:34:29   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:29     Number of barcodes from poolfile seen: 4928\n",
      "2024-06-09 17:34:29     Reads with barcodes from poolfile: 147241\n",
      "2024-06-09 17:34:29   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 87577 times.\n",
      "2024-06-09 17:34:29   Number of reads that differ from this barcode by one base pair (likely sequencing errors):2744\n",
      "2024-06-09 17:34:29   Estimated sequencing error rate for barcodes: 3.0%\n",
      "2024-06-09 17:34:29   Barcodes seen once (highly inflated by sequencing errors): 43948\n",
      "2024-06-09 17:34:29   Barcodes seen twice (slightly inflated by sequencing errors): 5429\n",
      "2024-06-09 17:34:29   Barcodes seen three times or more: 12385\n",
      "2024-06-09 17:34:29   Chao estimate of population size (ones^2/2*twos): 180000\n",
      "2024-06-09 17:34:29 ---------------------\n",
      "2024-06-09 17:34:29   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB020_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:29   Filtering reads on second index: TGTAGC\n",
      "2024-06-09 17:34:29   Looking for sequence barcode preceeded by CGTACG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-06-09 17:34:32     445364 reads processed.\n",
      "2024-06-09 17:34:32     430510 reads with compliant barcodes.\n",
      "2024-06-09 17:34:32     14854 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:32       7330 reads without expected Dual Index.\n",
      "2024-06-09 17:34:32       6212 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:32       1312 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:32       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:32       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:32   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB020_1.counts\n",
      "2024-06-09 17:34:32   Total barcodes seen (incudes sequencing errors): 28314\n",
      "2024-06-09 17:34:32   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:32     Number of barcodes from poolfile seen: 4841\n",
      "2024-06-09 17:34:32     Reads with barcodes from poolfile: 132067\n",
      "2024-06-09 17:34:32   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 74165 times.\n",
      "2024-06-09 17:34:32   Number of reads that differ from this barcode by one base pair (likely sequencing errors):753\n",
      "2024-06-09 17:34:32   Estimated sequencing error rate for barcodes: 1.0%\n",
      "2024-06-09 17:34:32   Barcodes seen once (highly inflated by sequencing errors): 14990\n",
      "2024-06-09 17:34:32   Barcodes seen twice (slightly inflated by sequencing errors): 3492\n",
      "2024-06-09 17:34:32   Barcodes seen three times or more: 9832\n",
      "2024-06-09 17:34:32   Chao estimate of population size (ones^2/2*twos): 32000\n",
      "2024-06-09 17:34:32 ---------------------\n",
      "2024-06-09 17:34:32   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB021_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:32   Filtering reads on second index: GCACTA\n",
      "2024-06-09 17:34:32   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:34:36     690993 reads processed.\n",
      "2024-06-09 17:34:36     664233 reads with compliant barcodes.\n",
      "2024-06-09 17:34:36     26760 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:36       11800 reads without expected Dual Index.\n",
      "2024-06-09 17:34:36       10719 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:36       4241 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:36       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:36       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:36   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB021_1.counts\n",
      "2024-06-09 17:34:36   Total barcodes seen (incudes sequencing errors): 58447\n",
      "2024-06-09 17:34:36   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:36     Number of barcodes from poolfile seen: 5244\n",
      "2024-06-09 17:34:36     Reads with barcodes from poolfile: 197008\n",
      "2024-06-09 17:34:36   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 94066 times.\n",
      "2024-06-09 17:34:36   Number of reads that differ from this barcode by one base pair (likely sequencing errors):2389\n",
      "2024-06-09 17:34:36   Estimated sequencing error rate for barcodes: 2.5%\n",
      "2024-06-09 17:34:36   Barcodes seen once (highly inflated by sequencing errors): 36222\n",
      "2024-06-09 17:34:36   Barcodes seen twice (slightly inflated by sequencing errors): 6673\n",
      "2024-06-09 17:34:36   Barcodes seen three times or more: 15552\n",
      "2024-06-09 17:34:36   Chao estimate of population size (ones^2/2*twos): 98000\n",
      "2024-06-09 17:34:36 ---------------------\n",
      "2024-06-09 17:34:36   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB022_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:36   Filtering reads on second index: TGTAGC\n",
      "2024-06-09 17:34:36   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:34:41     740202 reads processed.\n",
      "2024-06-09 17:34:41     710135 reads with compliant barcodes.\n",
      "2024-06-09 17:34:41     30067 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:41       14257 reads without expected Dual Index.\n",
      "2024-06-09 17:34:41       13436 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:41       2374 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:41       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:41       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:41   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB022_1.counts\n",
      "2024-06-09 17:34:41   Total barcodes seen (incudes sequencing errors): 43618\n",
      "2024-06-09 17:34:41   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:41     Number of barcodes from poolfile seen: 5634\n",
      "2024-06-09 17:34:41     Reads with barcodes from poolfile: 221396\n",
      "2024-06-09 17:34:41   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 99376 times.\n",
      "2024-06-09 17:34:41   Number of reads that differ from this barcode by one base pair (likely sequencing errors):1308\n",
      "2024-06-09 17:34:41   Estimated sequencing error rate for barcodes: 1.3%\n",
      "2024-06-09 17:34:41   Barcodes seen once (highly inflated by sequencing errors): 22958\n",
      "2024-06-09 17:34:41   Barcodes seen twice (slightly inflated by sequencing errors): 5061\n",
      "2024-06-09 17:34:41   Barcodes seen three times or more: 15599\n",
      "2024-06-09 17:34:41   Chao estimate of population size (ones^2/2*twos): 52000\n",
      "2024-06-09 17:34:41 ---------------------\n",
      "2024-06-09 17:34:41   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB023_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:41   Filtering reads on second index: GCACTA\n",
      "2024-06-09 17:34:41   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:34:48     1034509 reads processed.\n",
      "2024-06-09 17:34:48     995994 reads with compliant barcodes.\n",
      "2024-06-09 17:34:48     38515 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:48       18269 reads without expected Dual Index.\n",
      "2024-06-09 17:34:48       15583 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:48       4663 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:48       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:48       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:48   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB023_1.counts\n",
      "2024-06-09 17:34:49   Total barcodes seen (incudes sequencing errors): 110508\n",
      "2024-06-09 17:34:49   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:49     Number of barcodes from poolfile seen: 5754\n",
      "2024-06-09 17:34:49     Reads with barcodes from poolfile: 356839\n",
      "2024-06-09 17:34:49   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 78176 times.\n",
      "2024-06-09 17:34:49   Number of reads that differ from this barcode by one base pair (likely sequencing errors):2992\n",
      "2024-06-09 17:34:49   Estimated sequencing error rate for barcodes: 3.7%\n",
      "2024-06-09 17:34:49   Barcodes seen once (highly inflated by sequencing errors): 83213\n",
      "2024-06-09 17:34:49   Barcodes seen twice (slightly inflated by sequencing errors): 8588\n",
      "2024-06-09 17:34:49   Barcodes seen three times or more: 18707\n",
      "2024-06-09 17:34:49   Chao estimate of population size (ones^2/2*twos): 400000\n",
      "2024-06-09 17:34:49 ---------------------\n",
      "2024-06-09 17:34:49   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB024_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:49   Filtering reads on second index: TGTAGC\n",
      "2024-06-09 17:34:49   Looking for sequence barcode preceeded by CGTACG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-06-09 17:34:55     952960 reads processed.\n",
      "2024-06-09 17:34:55     923185 reads with compliant barcodes.\n",
      "2024-06-09 17:34:55     29775 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:34:55       13731 reads without expected Dual Index.\n",
      "2024-06-09 17:34:55       13453 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:34:55       2591 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:34:55       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:34:55       0 reads with quality scores less than 10\n",
      "2024-06-09 17:34:55   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB024_1.counts\n",
      "2024-06-09 17:34:55   Total barcodes seen (incudes sequencing errors): 51152\n",
      "2024-06-09 17:34:55   Matching barcodes to poolfile\n",
      "2024-06-09 17:34:55     Number of barcodes from poolfile seen: 5987\n",
      "2024-06-09 17:34:55     Reads with barcodes from poolfile: 348257\n",
      "2024-06-09 17:34:55   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 63400 times.\n",
      "2024-06-09 17:34:55   Number of reads that differ from this barcode by one base pair (likely sequencing errors):1084\n",
      "2024-06-09 17:34:55   Estimated sequencing error rate for barcodes: 1.7%\n",
      "2024-06-09 17:34:55   Barcodes seen once (highly inflated by sequencing errors): 28016\n",
      "2024-06-09 17:34:55   Barcodes seen twice (slightly inflated by sequencing errors): 5611\n",
      "2024-06-09 17:34:55   Barcodes seen three times or more: 17525\n",
      "2024-06-09 17:34:55   Chao estimate of population size (ones^2/2*twos): 70000\n",
      "2024-06-09 17:34:55 ---------------------\n",
      "2024-06-09 17:34:55   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB025_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:34:55   Filtering reads on second index: GCACTA\n",
      "2024-06-09 17:34:55   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:35:02     1013555 reads processed.\n",
      "2024-06-09 17:35:02     977658 reads with compliant barcodes.\n",
      "2024-06-09 17:35:02     35897 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:35:02       14979 reads without expected Dual Index.\n",
      "2024-06-09 17:35:02       15927 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:35:02       4991 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:35:02       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:35:02       0 reads with quality scores less than 10\n",
      "2024-06-09 17:35:02   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB025_1.counts\n",
      "2024-06-09 17:35:02   Total barcodes seen (incudes sequencing errors): 107387\n",
      "2024-06-09 17:35:02   Matching barcodes to poolfile\n",
      "2024-06-09 17:35:02     Number of barcodes from poolfile seen: 5824\n",
      "2024-06-09 17:35:02     Reads with barcodes from poolfile: 320794\n",
      "2024-06-09 17:35:02   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 56875 times.\n",
      "2024-06-09 17:35:02   Number of reads that differ from this barcode by one base pair (likely sequencing errors):1952\n",
      "2024-06-09 17:35:02   Estimated sequencing error rate for barcodes: 3.3%\n",
      "2024-06-09 17:35:02   Barcodes seen once (highly inflated by sequencing errors): 78370\n",
      "2024-06-09 17:35:02   Barcodes seen twice (slightly inflated by sequencing errors): 8857\n",
      "2024-06-09 17:35:02   Barcodes seen three times or more: 20160\n",
      "2024-06-09 17:35:02   Chao estimate of population size (ones^2/2*twos): 350000\n",
      "2024-06-09 17:35:02 ---------------------\n",
      "2024-06-09 17:35:02   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB026_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:35:02   Filtering reads on second index: TGTAGC\n",
      "2024-06-09 17:35:02   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:35:08     945922 reads processed.\n",
      "2024-06-09 17:35:08     916266 reads with compliant barcodes.\n",
      "2024-06-09 17:35:08     29656 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:35:08       13161 reads without expected Dual Index.\n",
      "2024-06-09 17:35:08       13731 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:35:08       2764 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:35:08       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:35:08       0 reads with quality scores less than 10\n",
      "2024-06-09 17:35:08   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB026_1.counts\n",
      "2024-06-09 17:35:09   Total barcodes seen (incudes sequencing errors): 54088\n",
      "2024-06-09 17:35:09   Matching barcodes to poolfile\n",
      "2024-06-09 17:35:09     Number of barcodes from poolfile seen: 6000\n",
      "2024-06-09 17:35:09     Reads with barcodes from poolfile: 327902\n",
      "2024-06-09 17:35:09   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 55491 times.\n",
      "2024-06-09 17:35:09   Number of reads that differ from this barcode by one base pair (likely sequencing errors):778\n",
      "2024-06-09 17:35:09   Estimated sequencing error rate for barcodes: 1.4%\n",
      "2024-06-09 17:35:09   Barcodes seen once (highly inflated by sequencing errors): 29818\n",
      "2024-06-09 17:35:09   Barcodes seen twice (slightly inflated by sequencing errors): 5726\n",
      "2024-06-09 17:35:09   Barcodes seen three times or more: 18544\n",
      "2024-06-09 17:35:09   Chao estimate of population size (ones^2/2*twos): 78000\n",
      "2024-06-09 17:35:09 ---------------------\n",
      "2024-06-09 17:35:09   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB027_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:35:09   Filtering reads on second index: GCACTA\n",
      "2024-06-09 17:35:09   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-06-09 17:35:17     1192898 reads processed.\n",
      "2024-06-09 17:35:17     1146514 reads with compliant barcodes.\n",
      "2024-06-09 17:35:17     46384 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:35:17       22373 reads without expected Dual Index.\n",
      "2024-06-09 17:35:17       17747 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:35:17       6263 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:35:17       34 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:35:17       129 reads with quality scores less than 10\n",
      "2024-06-09 17:35:17   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB027_1.counts\n",
      "2024-06-09 17:35:17   Total barcodes seen (incudes sequencing errors): 139434\n",
      "2024-06-09 17:35:17   Matching barcodes to poolfile\n",
      "2024-06-09 17:35:17     Number of barcodes from poolfile seen: 6026\n",
      "2024-06-09 17:35:17     Reads with barcodes from poolfile: 385192\n",
      "2024-06-09 17:35:17   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 59033 times.\n",
      "2024-06-09 17:35:17   Number of reads that differ from this barcode by one base pair (likely sequencing errors):2373\n",
      "2024-06-09 17:35:17   Estimated sequencing error rate for barcodes: 3.9%\n",
      "2024-06-09 17:35:17   Barcodes seen once (highly inflated by sequencing errors): 108345\n",
      "2024-06-09 17:35:17   Barcodes seen twice (slightly inflated by sequencing errors): 9365\n",
      "2024-06-09 17:35:17   Barcodes seen three times or more: 21724\n",
      "2024-06-09 17:35:17   Chao estimate of population size (ones^2/2*twos): 630000\n",
      "2024-06-09 17:35:17 ---------------------\n",
      "2024-06-09 17:35:17   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB028_S1_L001_R1_001.fastq\n",
      "2024-06-09 17:35:17   Filtering reads on second index: TGTAGC\n",
      "2024-06-09 17:35:17   Looking for sequence barcode preceeded by CGTACG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-06-09 17:35:24     1016107 reads processed.\n",
      "2024-06-09 17:35:24     984615 reads with compliant barcodes.\n",
      "2024-06-09 17:35:24     31492 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-06-09 17:35:24       14316 reads without expected Dual Index.\n",
      "2024-06-09 17:35:24       13213 reads without expected sequence before the barcode region.\n",
      "2024-06-09 17:35:24       3963 reads without expected sequence after the barcode region.\n",
      "2024-06-09 17:35:24       0 reads with noncompliant barcodes. (Contains Ns, etc).\n",
      "2024-06-09 17:35:24       0 reads with quality scores less than 10\n",
      "2024-06-09 17:35:24   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/countsFiles/KFRB028_1.counts\n",
      "2024-06-09 17:35:24   Total barcodes seen (incudes sequencing errors): 81358\n",
      "2024-06-09 17:35:24   Matching barcodes to poolfile\n",
      "2024-06-09 17:35:24     Number of barcodes from poolfile seen: 5946\n",
      "2024-06-09 17:35:24     Reads with barcodes from poolfile: 349920\n",
      "2024-06-09 17:35:24   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 54155 times.\n",
      "2024-06-09 17:35:24   Number of reads that differ from this barcode by one base pair (likely sequencing errors):1549\n",
      "2024-06-09 17:35:24   Estimated sequencing error rate for barcodes: 2.8%\n",
      "2024-06-09 17:35:24   Barcodes seen once (highly inflated by sequencing errors): 54341\n",
      "2024-06-09 17:35:24   Barcodes seen twice (slightly inflated by sequencing errors): 7539\n",
      "2024-06-09 17:35:24   Barcodes seen three times or more: 19478\n",
      "2024-06-09 17:35:24   Chao estimate of population size (ones^2/2*twos): 200000\n",
      "2024-06-09 17:35:24 ---------------------\n",
      "2024-06-09 17:35:24 Saving summary statistics for fastqs to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/fastqSummaryStats.txt\n",
      "2024-06-09 17:35:24 Processed 12 sequence files corresponding to 12 biological samples.\n",
      "2024-06-09 17:35:24 Counted 3632038 total reads for 7188 mapped barcodes from the mutant pool.\n",
      "2024-06-09 17:35:24 Saving poolcount file to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/poolCount.txt\n",
      "2024-06-09 17:35:26 Barcodes found in BarSeq but not in TnSeq poolfile written to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/script_modification/barcodes_not_in_poolfile.txt\n"
     ]
    }
   ],
   "source": [
    "!python3 /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/RBseq_Count_BarCodes_v2_edited.py -m {metafile_name} -l {logfile_name}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB007_1'\n",
    "y_column = 'KFRB008_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB007_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB008_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances at 37 oC', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_01_03_2024_37oC.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB007_1'\n",
    "y_column = 'KFRB008_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB007_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB008_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances at 37 oC (A)', fontsize=16, color='purple')\n",
    "\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 12500)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 10000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_37oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB019_1'\n",
    "y_column = 'KFRB020_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB019_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB020_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances at 37 oC', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_01_03_2024_37oC.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB019_1'\n",
    "y_column = 'KFRB020_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB019_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB020_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_37 oC (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 8000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 8000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB021_1'\n",
    "y_column = 'KFRB022_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB021_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB022_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 10000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 10000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB023_1'\n",
    "y_column = 'KFRB024_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB021_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB022_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 6000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 6000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB025_1'\n",
    "y_column = 'KFRB026_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB025_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB026_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_50 oC (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 8000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 8000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB027_1'\n",
    "y_column = 'KFRB028_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('KFRB027_1 Abundance', fontsize=14)\n",
    "plt.ylabel('KFRB028_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_50 oC (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 8000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 8000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## to normalize the Barseq pool count file based on the number of reads obtained"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### normalization based on individual sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_KFRB007_1'] = df['KFRB007_1'].sum()\n",
    "df['Total_Count_KFRB008_1'] = df['KFRB008_1'].sum()\n",
    "df['Total_Count_KFRB019_1'] = df['KFRB019_1'].sum()\n",
    "df['Total_Count_KFRB020_1'] = df['KFRB020_1'].sum()\n",
    "df['Total_Count_KFRB021_1'] = df['KFRB021_1'].sum()\n",
    "df['Total_Count_KFRB022_1'] = df['KFRB022_1'].sum()\n",
    "df['Total_Count_KFRB023_1'] = df['KFRB023_1'].sum()\n",
    "df['Total_Count_KFRB024_1'] = df['KFRB024_1'].sum()\n",
    "df['Total_Count_KFRB025_1'] = df['KFRB025_1'].sum()\n",
    "df['Total_Count_KFRB026_1'] = df['KFRB026_1'].sum()\n",
    "df['Total_Count_KFRB027_1'] = df['KFRB027_1'].sum()\n",
    "df['Total_Count_KFRB028_1'] = df['KFRB028_1'].sum()\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['KFRB007_1'] = df['KFRB007_1'] / df['Total_Count_KFRB007_1']\n",
    "df['KFRB008_1'] = df['KFRB008_1'] / df['Total_Count_KFRB008_1']\n",
    "df['KFRB019_1'] = df['KFRB019_1'] / df['Total_Count_KFRB019_1']\n",
    "df['KFRB020_1'] = df['KFRB020_1'] / df['Total_Count_KFRB020_1']\n",
    "df['KFRB021_1'] = df['KFRB021_1'] / df['Total_Count_KFRB021_1']\n",
    "df['KFRB022_1'] = df['KFRB022_1'] / df['Total_Count_KFRB022_1']\n",
    "df['KFRB023_1'] = df['KFRB023_1'] / df['Total_Count_KFRB023_1']\n",
    "df['KFRB024_1'] = df['KFRB024_1'] / df['Total_Count_KFRB024_1']\n",
    "df['KFRB025_1'] = df['KFRB025_1'] / df['Total_Count_KFRB025_1']\n",
    "df['KFRB026_1'] = df['KFRB026_1'] / df['Total_Count_KFRB026_1']\n",
    "df['KFRB027_1'] = df['KFRB027_1'] / df['Total_Count_KFRB027_1']\n",
    "df['KFRB028_1'] = df['KFRB028_1'] / df['Total_Count_KFRB028_1']\n",
    "\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### this is the normalization that worked with the right result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/poolCount.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_KFRB007_1'] = df['KFRB007_1'].sum()\n",
    "df['Total_Count_KFRB008_1'] = df['KFRB008_1'].sum()\n",
    "df['Total_Count_KFRB019_1'] = df['KFRB019_1'].sum()\n",
    "df['Total_Count_KFRB020_1'] = df['KFRB020_1'].sum()\n",
    "df['Total_Count_KFRB021_1'] = df['KFRB021_1'].sum()\n",
    "df['Total_Count_KFRB022_1'] = df['KFRB022_1'].sum()\n",
    "df['Total_Count_KFRB023_1'] = df['KFRB023_1'].sum()\n",
    "df['Total_Count_KFRB024_1'] = df['KFRB024_1'].sum()\n",
    "df['Total_Count_KFRB025_1'] = df['KFRB025_1'].sum()\n",
    "df['Total_Count_KFRB026_1'] = df['KFRB026_1'].sum()\n",
    "df['Total_Count_KFRB027_1'] = df['KFRB027_1'].sum()\n",
    "df['Total_Count_KFRB028_1'] = df['KFRB028_1'].sum()\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['KFRB007_1'] = df['KFRB007_1'] / df['Total_Count_KFRB007_1']\n",
    "df['KFRB008_1'] = df['KFRB008_1'] / df['Total_Count_KFRB008_1']\n",
    "df['KFRB019_1'] = df['KFRB019_1'] / df['Total_Count_KFRB019_1']\n",
    "df['KFRB020_1'] = df['KFRB020_1'] / df['Total_Count_KFRB020_1']\n",
    "df['KFRB021_1'] = df['KFRB021_1'] / df['Total_Count_KFRB021_1']\n",
    "df['KFRB022_1'] = df['KFRB022_1'] / df['Total_Count_KFRB022_1']\n",
    "df['KFRB023_1'] = df['KFRB023_1'] / df['Total_Count_KFRB023_1']\n",
    "df['KFRB024_1'] = df['KFRB024_1'] / df['Total_Count_KFRB024_1']\n",
    "df['KFRB025_1'] = df['KFRB025_1'] / df['Total_Count_KFRB025_1']\n",
    "df['KFRB026_1'] = df['KFRB026_1'] / df['Total_Count_KFRB026_1']\n",
    "df['KFRB027_1'] = df['KFRB027_1'] / df['Total_Count_KFRB027_1']\n",
    "df['KFRB028_1'] = df['KFRB028_1'] / df['Total_Count_KFRB028_1']\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To make the fitness abundance agaian using the normalized values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the CSV file\n",
    "csv_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.csv'\n",
    "df = pd.read_csv(csv_file_path)\n",
    "\n",
    "# Save the DataFrame to a text file (txt)\n",
    "txt_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt'\n",
    "df.to_csv(txt_file_path, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB007_1'\n",
    "y_column = 'KFRB008_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB007_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB008_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_37 oC_normalized (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.04)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.04)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB019_1'\n",
    "y_column = 'KFRB020_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB019_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB020_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_37 oC_normalized (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB021_1'\n",
    "y_column = 'KFRB022_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB021_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB022_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC_normalized (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB023_1'\n",
    "y_column = 'KFRB024_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB023_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB024_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC_normalized (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB025_1'\n",
    "y_column = 'KFRB026_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB025_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB026_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_50 oC_normalized (A)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/TNSeq_13k_mutants_mapping_output/TNSeq_barseq_outputs_using_combined_tnseq_poolfile_all_temp_05_23_2024/normalized_barseq_pool_counts_05_28_2024.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'KFRB027_1'\n",
    "y_column = 'KFRB028_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_KFRB027_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_KFRB028_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC_normalized (B)', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.02)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.02)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
