{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Note: the mapping script takes days to run! You should be running this notebook using screen, so if you close this page the script will keep running.\n",
    "\n",
    "If you close the notebook while using screen, the log output within the notebook might stop being printed. But the logfile specified in the line where you call the script will keep being written.\n",
    "\n",
    "How to run a Jupyter notebook with screen: https://docs.google.com/document/d/1lLcl3HhDBrn87M4-IhkpDLOmvmmf09-KeLSgn3U21hE/edit?usp=sharing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: .bashrc: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!source .bashrc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pandas in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (1.5.3)\n",
      "Requirement already satisfied: numpy>=1.21.0 in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (from pandas) (1.24.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (from pandas) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (from pandas) (2023.3)\n",
      "Requirement already satisfied: six>=1.5 in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (from python-dateutil>=2.8.1->pandas) (1.16.0)\n"
     ]
    }
   ],
   "source": [
    "!pip install pandas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make metafile for TNseq mapping script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Info that will be the same for all pools\n",
    "\n",
    "metafile_name: Name of file where we're going to write the metadata\n",
    "\n",
    "combined_pool_name: What we're going to call the combination of individual pools\n",
    "\n",
    "ReadModel: Full or relative path to a text file describing the expected structure of TnSeq reads.  The file consists of 4 lines.  Line 1 denotes the number of initial variable bases to ignore as 'N's.  Line 2 is the sequence preceding the barcode.  Line 3 is the expected length of sequence barcode denoted as 'N's.  Line 4 is the sequence between the barcode and the junction with the genome.  Example:\n",
    "\n",
    "    nnnnnn\n",
    "    GATGTCCACGAGGTCTCT\n",
    "    NNNNNNNNNNNNNNNNNNNN\n",
    "    CGTACGCTGCAGGTCGACAATGATCCAAACTATCAGTGTTTGA\t\n",
    "\n",
    "InsertionSequence: Full or relative path to a text file describing giving the full sequence of each insertion.  This should be a fasta format file with two entries labeled 'insert' and 'plasmid' giving the full sequence of the expected insertion into the genome and and the remaining source plasmid sequence, if appropriate. Example:\n",
    "\n",
    "    >insert\n",
    "    agtcac...\n",
    "    >plasmid\n",
    "    actgact... \n",
    "      \n",
    "GenomeSequence: Full or relative path to a .fasta file of genome sequences for all species in pool.\n",
    "\n",
    "OutputDir: Full or relative path to the desired directory to save output files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED\n",
    "\n",
    "metafile_name = '../20032023_novaseq/TNSeq_mapping_metafile.txt'\n",
    "mapping_log_file = '../20032023_novaseq/TNSeq_mapping.log'\n",
    "\n",
    "ReadModel = '/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt'\n",
    "InsertionSequence = '/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt'\n",
    "GenomeSequence = '/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt'\n",
    "OutputDir = '../20032023_novaseq/FullRun_Output'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pool-specific information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where you saved the merged FASTQs in TNseq_prepare_files.ipynb\n",
    "FASTQ_directory = '../20032023_novaseq/TNSeq_merge_FASTQ'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../20032023_novaseq/TNSeq_merge_FASTQ/ADRB501_deep_S1_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/ADRB502_deep_S2_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/ADRB503_deep_S3_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/ADRB504_deep_S4_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/ADRB505_deep_S5_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/ADRB506_deep_S6_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/KCRB001_deep_S7_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/KCRB002_deep_S8_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/KCRB003_deep_S9_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/KCRB004_deep_S10_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/KCRB005_deep_S11_L002_merged.fastq\r\n",
      "../20032023_novaseq/TNSeq_merge_FASTQ/Undetermined_S0_L002_merged.fastq\r\n"
     ]
    }
   ],
   "source": [
    "#get the list of fastqs we generated in the BBmerge step above\n",
    "#copy and paste these into the cell below this one\n",
    "\n",
    "!ls {FASTQ_directory}/*_merged.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each pool, come up with a UNIQUE short name for the pool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED BUT KEEP THE [('NAME', 'PATH/TO/FASTQ.FASTQ')] FORMAT\n",
    "\n",
    "short_names_and_fastqs = [('28C_ADRB501', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB501_deep_S1_L002_merged.fastq'),\n",
    "                          ('28C_ADRB502', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB502_deep_S2_L002_merged.fastq'),\n",
    "                         ('28C_ADRB503', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB503_deep_S3_L002_merged.fastq'),\n",
    "                         ('42C_ADRB504', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB504_deep_S4_L002_merged.fastq'),\n",
    "                         ('42C_ADRB505', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB505_deep_S5_L002_merged.fastq'),\n",
    "                         ('42C_ADRB506', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB506_deep_S6_L002_merged.fastq'),\n",
    "                         ('28C_KCRB001', '../20032023_novaseq/TNSeq_merge_FASTQ/KCRB001_deep_S7_L002_merged.fastq'),\n",
    "                         ('28C_KCRB002', '../20032023_novaseq/TNSeq_merge_FASTQ/KCRB002_deep_S8_L002_merged.fastq'),\n",
    "                         ('28C_KCRB003', '../20032023_novaseq/TNSeq_merge_FASTQ/KCRB003_deep_S9_L002_merged.fastq'),\n",
    "                         ('42C_KCRB004', '../20032023_novaseq/TNSeq_merge_FASTQ/KCRB004_deep_S10_L002_merged.fastq'),\n",
    "                         ('42C_KCRB005', '../20032023_novaseq/TNSeq_merge_FASTQ/KCRB005_deep_S11_L002_merged.fastq'),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE THESE AS NEEDED BUT KEEP THE [('NAME', 'PATH/TO/FASTQ.FASTQ')] FORMAT\n",
    "\n",
    "short_names_and_fastqs = [('28C_ADRB501_test', '../20032023_novaseq/TNSeq_merge_FASTQ/ADRB501_deep_S1_L002_merged_subsample.fastq'),\n",
    "                          ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Writing the above info into a metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['Pool', 'ShortName', 'Fastq', 'ReadModel', 'InsertionSequence', 'GenomeSequence', 'OutputDir']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for pool, fastq in short_names_and_fastqs:\n",
    "        \n",
    "        #we write the pool name twice because we're first processing each pool separately then combining them\n",
    "        #if we were combining them all at once then the first pool variable would be the name of the combined pool\n",
    "        to_write = '\\t'.join([pool, pool, fastq, ReadModel, InsertionSequence, GenomeSequence, OutputDir])\n",
    "        f.write(to_write+'\\n')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pool</th>\n",
       "      <th>ShortName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>ReadModel</th>\n",
       "      <th>InsertionSequence</th>\n",
       "      <th>GenomeSequence</th>\n",
       "      <th>OutputDir</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>28C_ADRB501_test</td>\n",
       "      <td>28C_ADRB501_test</td>\n",
       "      <td>../20032023_novaseq/TNSeq_merge_FASTQ/ADRB501_deep_S1_L002_merged_subsample.fastq</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt</td>\n",
       "      <td>/usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt</td>\n",
       "      <td>../20032023_novaseq/FullRun_Output</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Pool         ShortName  \\\n",
       "0  28C_ADRB501_test  28C_ADRB501_test   \n",
       "\n",
       "                                                                               Fastq  \\\n",
       "0  ../20032023_novaseq/TNSeq_merge_FASTQ/ADRB501_deep_S1_L002_merged_subsample.fastq   \n",
       "\n",
       "                                                          ReadModel  \\\n",
       "0  /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt   \n",
       "\n",
       "                                                               InsertionSequence  \\\n",
       "0  /usr2/people/clairedubin/barseq/latest_pipeline/kluyv_insertion_corrected.txt   \n",
       "\n",
       "                                                             GenomeSequence  \\\n",
       "0  /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt   \n",
       "\n",
       "                            OutputDir  \n",
       "0  ../20032023_novaseq/FullRun_Output  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#view metafile\n",
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Map transposons in each pool to genome\n",
    "\n",
    "View additional input and output options here: https://github.com/stcoradetti/RBseq/blob/master/README.txt\n",
    "\n",
    "This takes a long time to run and will be stuck on the \"Looking for 20bp sequence barcode\" step for a while. You can see how many insertions have been processed by switching to the output directory specified above and typing:\n",
    "\n",
    "    wc -l *blastquery.fasta\n",
    "    \n",
    "This number divided by 2 is roughly how many reads have been processed. You can also check the logFile specified below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RBseq_Map_Insertions.py\n",
      "2023-03-29 19:52:09 Version: 1.1.4_PBa\n",
      "2023-03-29 19:52:09 Release Date: August 8, 2020\n",
      "2023-03-29 19:52:09 Options passed:  metafile:../20032023_novaseq/TNSeq_mapping_metafile.txt  logFile:../20032023_novaseq/TNSeq_mapping.log  minQual:10  matchBefore:6  matchAfter:6  matchJunction:4  maxFillerSeq:100  barcodeVariation:2  wobbleAllowed:1  scoreDiff:10  minFraction:0.9  filterNeighborhood:10  filterEditDistance:5  minPercentID:95  maxEvalue:0.1  useMappedFiles:False  noInsertHits:False  noBarcodes:False \n",
      "2023-03-29 19:52:09 Logging status updates in ../20032023_novaseq/TNSeq_mapping.log\n",
      "2023-03-29 19:52:09 Loading TnSeq library metadata from ../20032023_novaseq/TNSeq_mapping_metafile.txt\n",
      "2023-03-29 19:52:09 Finding barcodes in fastqs and mapping insertion locations\n",
      "2023-03-29 19:52:09   Mapping reads from ../20032023_novaseq/TNSeq_merge_FASTQ/ADRB501_deep_S1_L002_merged_subsample.fastq using insertion model /usr2/people/clairedubin/barseq/latest_pipeline/Model_pJC9BC.txt\n",
      "2023-03-29 19:52:09   Looking for 20bp sequence barcode flanked by GTCTCT and CGTACG\n",
      "2023-03-29 19:52:10     0 reads without recognizable, compliant barcodes written to ../20032023_novaseq/FullRun_Output/28C_ADRB501_test_noBarcodeFound.txt\n",
      "2023-03-29 19:52:10     0 query sequences with compliant barcodes written to ../20032023_novaseq/FullRun_Output/28C_ADRB501_test_blastquery.fasta\n",
      "2023-03-29 19:52:10   BLASTING reads against /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr2/people/kayleeec/rhseq/RHSeq_without_barcodes/RHseq_Map_Insertions_v1-1.1.4_PBa_Jskerker.py\", line 912, in <module>\n",
      "    main(sys.argv[1:])\n",
      "  File \"/usr2/people/kayleeec/rhseq/RHSeq_without_barcodes/RHseq_Map_Insertions_v1-1.1.4_PBa_Jskerker.py\", line 424, in main\n",
      "    stdout, stderr = blastGenome()\n",
      "  File \"/usr2/people/kayleeec/anaconda3/lib/python3.10/site-packages/Bio/Application/__init__.py\", line 574, in __call__\n",
      "    raise ApplicationError(return_code, str(self), stdout_str, stderr_str)\n",
      "Bio.Application.ApplicationError: Non-zero return code 127 from 'blastn -out ../20032023_novaseq/FullRun_Output/28C_ADRB501_test_blastGenome.txt -outfmt 6 -query ../20032023_novaseq/FullRun_Output/28C_ADRB501_test_blastquery.fasta -db /usr2/people/clairedubin/barseq/genome_and_annotations/Klac_Kmar.fsa.txt -evalue 0.1 -perc_identity 95', message '/bin/sh: blastn: command not found'\n"
     ]
    }
   ],
   "source": [
    "cmd = 'nice -n 17 python3 /usr2/people/kayleeec/rhseq/RHSeq_without_barcodes/RHseq_Map_Insertions_v1-1.1.4_PBa_Jskerker.py --metafile {} --logFile {}'.format(metafile_name, mapping_log_file)\n",
    "\n",
    "!{cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting blastpy3\n",
      "  Using cached blastpy3-0.3.0.tar.gz (9.6 kB)\n",
      "  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting pyfaidx>=0.5.8\n",
      "  Downloading pyfaidx-0.7.2.1-py3-none-any.whl (28 kB)\n",
      "Requirement already satisfied: six in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (from pyfaidx>=0.5.8->blastpy3) (1.16.0)\n",
      "Requirement already satisfied: setuptools in /auto/sahara/namib/home/kayleeec/anaconda3/lib/python3.10/site-packages (from pyfaidx>=0.5.8->blastpy3) (65.6.3)\n",
      "Building wheels for collected packages: blastpy3\n",
      "  Building wheel for blastpy3 (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for blastpy3: filename=blastpy3-0.3.0-py3-none-any.whl size=10722 sha256=52358464c3e0c38da3e7a707c970d37aab035b9afb886bf8728e3f4d21e3713d\n",
      "  Stored in directory: /auto/sahara/namib/home/kayleeec/.cache/pip/wheels/b6/51/c1/3a73571decb662a4388e7e8965f1ba50575c0a49f0df8ae53a\n",
      "Successfully built blastpy3\n",
      "Installing collected packages: pyfaidx, blastpy3\n",
      "Successfully installed blastpy3-0.3.0 pyfaidx-0.7.2.1\n"
     ]
    }
   ],
   "source": [
    "!pip install blastpy3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
