{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "99374c91",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "def merge_fasta(files, output_file):\n",
    "    with open(output_file, 'w') as outfile:\n",
    "        for file in files:\n",
    "            for record in SeqIO.parse(file, 'fasta'):\n",
    "                SeqIO.write(record, outfile, 'fasta')\n",
    "\n",
    "# Example usage\n",
    "files_to_merge = ['/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_10_mins/TNSeq_merge_FASTQ/OORBO28_S1_L001_merged.fastq', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_15_mins/TNSeq_merge_FASTQ/OORBO29_S1_L001_merged.fastq', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_20_mins/TNSeq_merge_FASTQ/OORBO30_S1_L001_merged.fastq']\n",
    "merge_fasta(files_to_merge, '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/TNSeq_merge_FASTQ/all_digestion_time_merged.fastq')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee7aafc2",
   "metadata": {},
   "source": [
    "## to count total number of reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b04ab12d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 1279040\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_10_mins/TNSeq_merge_FASTQ/OORBO28_S1_L001_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f6a6055e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 1254821\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_15_mins/TNSeq_merge_FASTQ/OORBO29_S1_L001_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "18c42993",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 1314803\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_20_mins/TNSeq_merge_FASTQ/OORBO30_S1_L001_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3981fb6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "\n",
    "def merge_fastq(files, output_file):\n",
    "    with open(output_file, 'w') as outfile:\n",
    "        for file in files:\n",
    "            for record in SeqIO.parse(file, 'fastq'):\n",
    "                SeqIO.write(record, outfile, 'fastq')\n",
    "\n",
    "# Example usage\n",
    "files_to_merge = ['/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_10_mins/TNSeq_merge_FASTQ/OORBO28_S1_L001_merged.fastq', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_15_mins/TNSeq_merge_FASTQ/OORBO29_S1_L001_merged.fastq', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/digestion_time_20_mins/TNSeq_merge_FASTQ/OORBO30_S1_L001_merged.fastq']\n",
    "merge_fastq(files_to_merge, '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/TNSeq_merge_FASTQ/all_digestion_time_merged.fastq')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "0450a655",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 3848664\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/TNSeq_merge_FASTQ/all_digestion_time_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "df69938e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
