{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import Seq\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Change the following parameters to your output directory (where you want output files to be written), your metafile, and TNseq output. \n",
    "\n",
    "If these files aren't in the same directory as this notebook, you need use the full path. For example, the full path of 'barseq_output' would be '/usr2/people/clairedubin/barseq/barseq_output'\n",
    "\n",
    "Also change the control and experimental temperatures if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where the BarSeq FASTQs are\n",
    "FASTQ_directory = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Barseq metafile\n",
    "\n",
    "You'll need to create your own metafile - easiest way is edit in Excel. A sample metafile is located at: /usr2/people/clairedubin/barseq/Kluyv_BarSeq_metadata.txt\n",
    "\n",
    "    Fastq: path to barseq FASTQ for each competition\n",
    "    SampleName: unique name for each competition, MUST include ctrl_temp or exp_temp as specified above\n",
    "    DualIndex: index from sequencing (in read name lines of FASTQ and the sequencing info file that Adam sends)\n",
    "    \n",
    "    UsePrecounted: True if you've already run the Barseq counting script on a competition, False otherwise\n",
    "    \n",
    "    poolfile: path to Tnseq output file, should be the same for each competition\n",
    "    output_dir: path to output directory\n",
    "    minRandom: number of bases between start of read and DualIndex, usually in sequence of 1,2,3,4,1,2,3,4 for each sample\n",
    "    maxRandom: same as minRandom\n",
    "    \n",
    "    BeforeBarcode: sequence preceding all barcodes\n",
    "    AfterBarcode: sequence following all barcodes\n",
    "    BarcodeLengths: lengths of barcodes to search for, with expected length of barcode first\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB006_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB007_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB008_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB009_S1_L001_R1_001.fastq\r\n"
     ]
    }
   ],
   "source": [
    "!ls {FASTQ_directory}/*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format: sample_name, fastq_path, dual_index, offset\n",
    "#add or delete more lines as needed\n",
    "\n",
    "sample_info = [('OORB006_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB006_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('OORB007_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB007_S1_L001_R1_001.fastq', 'CGATGT', '2'), \n",
    "               ('OORB008_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB008_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('OORB009_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB009_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@M02248:478:000000000-DMCTD:1:1101:15779:1330 1:N:0:NTCACG\r\n",
      "AGCACTAGTCGACCTGCAGCGTACGTTCTACTTCTTAGCCAACTAAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "CCCDDFFFFFCCGGGGGGGGGGGGGHHHGHHHHHHHHHHHGHGHHHHHHHHGHHGHHGGHHHHHHHHHHGGGGHHHGHHHGHHHGHHHHHHHHHHHHHHHHHGGHFGHHGGHHHHGGGHGHHHHHHHHHFHHGGGCD>--\r\n",
      "@M02248:478:000000000-DMCTD:1:1101:15674:1330 1:N:0:NTCACG\r\n",
      "CGCACTAGTCGACCTGCAGCGTACGGTGCTCATAACAGACGGGGTAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "BCCCDDFFFFDCGGGGGGGGGGGGGGGHGHHHHHHHHHHHGGGGGGGHHHHGHHGGHGGHHHHHHHHHHGGGGHHHHHHHGHHHGHHHHHHHHHHGGHHHHHGGHAGHHGGHHHHGGGHGHHHHHHHHGHGHGG@D@-<-\r\n",
      "@M02248:478:000000000-DMCTD:1:1101:15877:1332 1:N:0:NTCACG\r\n",
      "CGCACTAGTCGACCTGCAGCGTACGTAATTTGGCACACATACCACAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n"
     ]
    }
   ],
   "source": [
    "!head /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB006_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if there are any samples where you already counted barcodes, add the sample names here\n",
    "precounted_samples = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these variables should be the same for each replicate\n",
    "\n",
    "metafile_name = 'BarSeq_metafile_12_04_2023.txt'\n",
    "logfile_name = 'BarSeq_counting_12_04_2023.log'\n",
    "poolfile = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv'\n",
    "output_dir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023'\n",
    "before_barcode = 'GTCGACCTGCAGCGTACG' #bottom strand sequence was used\n",
    "#before_barcode = 'GATGTCCACGAGGTCTCT' what was there before \n",
    "after_barcode =  'AGAGACCTCGTGGACATC' #bottom strand sequence was used\n",
    "#after_barcode = 'CGTACGCTGCAGGTCGAC'  what was there before \n",
    "barcode_lengths = '20,19,18,21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['FileIndex','SampleName', 'Fastq', 'DualIndex', 'Poolfile', 'OutputDir', 'minRandom', 'maxRandom',\n",
    "                   'BeforeBarcode', 'AfterBarcode', 'BarcodeLengths', 'UsePrecounted']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for sample_name, fastq, dual_index, offset in sample_info:\n",
    "        \n",
    "        #DUAL INDECES ARE BACKWARDS FROM WHAT THE FLOWCELL OUTPUT SAYS\n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index[::-1], poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'False'])\n",
    "        f.write(to_write+'\\n')\n",
    "        \n",
    "    for precounted_sample_name in precounted_samples:\n",
    "        \n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index, poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'True'])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileIndex</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>DualIndex</th>\n",
       "      <th>Poolfile</th>\n",
       "      <th>OutputDir</th>\n",
       "      <th>minRandom</th>\n",
       "      <th>maxRandom</th>\n",
       "      <th>BeforeBarcode</th>\n",
       "      <th>AfterBarcode</th>\n",
       "      <th>BarcodeLengths</th>\n",
       "      <th>UsePrecounted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OORB006_1</td>\n",
       "      <td>OORB006_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB006_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OORB007_1</td>\n",
       "      <td>OORB007_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB007_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OORB008_1</td>\n",
       "      <td>OORB008_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB008_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OORB009_1</td>\n",
       "      <td>OORB009_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB009_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FileIndex SampleName  \\\n",
       "0  OORB006_1  OORB006_1   \n",
       "1  OORB007_1  OORB007_1   \n",
       "2  OORB008_1  OORB008_1   \n",
       "3  OORB009_1  OORB009_1   \n",
       "\n",
       "                                                                                                                               Fastq  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB006_S1_L001_R1_001.fastq   \n",
       "1  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB007_S1_L001_R1_001.fastq   \n",
       "2  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB008_S1_L001_R1_001.fastq   \n",
       "3  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB009_S1_L001_R1_001.fastq   \n",
       "\n",
       "  DualIndex  \\\n",
       "0  GCACTA     \n",
       "1  TGTAGC     \n",
       "2  GCACTA     \n",
       "3  TGTAGC     \n",
       "\n",
       "                                                                                                                                        Poolfile  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv   \n",
       "1  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv   \n",
       "2  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv   \n",
       "3  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv   \n",
       "\n",
       "                                                                                                                                                                                            OutputDir  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023   \n",
       "1  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023   \n",
       "2  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023   \n",
       "3  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023   \n",
       "\n",
       "   minRandom  maxRandom       BeforeBarcode        AfterBarcode  \\\n",
       "0  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "1  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "2  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "3  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "\n",
       "  BarcodeLengths  UsePrecounted  \n",
       "0  20,19,18,21    False          \n",
       "1  20,19,18,21    False          \n",
       "2  20,19,18,21    False          \n",
       "3  20,19,18,21    False          "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Barseq counting\n",
    "\n",
    "Might take a few hours!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-12-04 12:01:30 RBseq_Count_BarCodes.py\n",
      "2023-12-04 12:01:30 Version: 1.1.4\n",
      "2023-12-04 12:01:30 Release Date: July 16, 2020\n",
      "2023-12-04 12:01:30 Options passed:  metafile:BarSeq_metafile_12_04_2023.txt  logFile:BarSeq_counting_12_04_2023.log  minQual:10  matchBefore:6  matchAfter:6  quietMode:False \n",
      "2023-12-04 12:01:30 Logging status updates in BarSeq_counting_12_04_2023.log\n",
      "2023-12-04 12:01:30 Loading TnSeq library metadata from BarSeq_metafile_12_04_2023.txt\n",
      "2023-12-04 12:01:30 Setting output directory as: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023 (from first line of metadata file)\n",
      "2023-12-04 12:01:30 Loading mapped barcodes in mutant pool from: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv (from first line of metadata file)\n",
      "2023-12-04 12:01:30 Read 2443 barcodes from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNseq_single_position_nMainLocation_annotated_file.csv\n",
      "2023-12-04 12:01:30 Finding barcodes in fastqs and counting occurances\n",
      "2023-12-04 12:01:30 ---------------------\n",
      "2023-12-04 12:01:30   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB006_S1_L001_R1_001.fastq\n",
      "2023-12-04 12:01:30   Filtering reads on second index: GCACTA\n",
      "2023-12-04 12:01:30   Looking for sequence barcode preceeded by CGTACG\n",
      "2023-12-04 12:01:37     1244895 reads processed.\n",
      "2023-12-04 12:01:37     1183585 reads with compliant barcodes.\n",
      "2023-12-04 12:01:37     61310 reads without recognizable, compliant barcodes. Of those:\n",
      "2023-12-04 12:01:37       31422 reads without expected Dual Index.\n",
      "2023-12-04 12:01:37       25392 reads without expected sequence before the barcode region.\n",
      "2023-12-04 12:01:37       4496 reads without expeced sequence after the barcode region.\n",
      "2023-12-04 12:01:37       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2023-12-04 12:01:37       0 reads with quality scores less than 10\n",
      "2023-12-04 12:01:37   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023/countsFiles/OORB006_1.counts\n",
      "2023-12-04 12:01:37   Total barcodes seen (incudes sequencing errors): 38331\n",
      "2023-12-04 12:01:37   Matching barcodes to poolfile\n",
      "2023-12-04 12:01:37     Number of barcodes from poolfile seen: 82\n",
      "2023-12-04 12:01:37     Reads with barcodes from poolfile: 276721\n",
      "2023-12-04 12:01:37   Most abundant barcode: GTGCTCATAACAGACGGGGT seen 305737 times.\n",
      "2023-12-04 12:01:37   Number of reads that differ from this barcode by one base pair (likely sequencing errors):12495\n",
      "2023-12-04 12:01:37   Estimated sequencing error rate for barcodes: 3.9%\n",
      "2023-12-04 12:01:37   Barcodes seen once (highly inflated by sequencing errors): 29843\n",
      "2023-12-04 12:01:37   Barcodes seen twice (slightly inflated by sequencing errors): 3844\n",
      "2023-12-04 12:01:37   Barcodes seen three times or more: 4644\n",
      "2023-12-04 12:01:37   Chao estimate of population size (ones^2/2*twos): 120000\n",
      "2023-12-04 12:01:37 ---------------------\n",
      "2023-12-04 12:01:37   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB007_S1_L001_R1_001.fastq\n",
      "2023-12-04 12:01:37   Filtering reads on second index: TGTAGC\n",
      "2023-12-04 12:01:37   Looking for sequence barcode preceeded by CGTACG\n",
      "2023-12-04 12:01:45     1445474 reads processed.\n",
      "2023-12-04 12:01:45     1367650 reads with compliant barcodes.\n",
      "2023-12-04 12:01:45     77824 reads without recognizable, compliant barcodes. Of those:\n",
      "2023-12-04 12:01:45       28259 reads without expected Dual Index.\n",
      "2023-12-04 12:01:45       22211 reads without expected sequence before the barcode region.\n",
      "2023-12-04 12:01:45       27354 reads without expeced sequence after the barcode region.\n",
      "2023-12-04 12:01:45       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2023-12-04 12:01:45       0 reads with quality scores less than 10\n",
      "2023-12-04 12:01:45   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023/countsFiles/OORB007_1.counts\n",
      "2023-12-04 12:01:45   Total barcodes seen (incudes sequencing errors): 67531\n",
      "2023-12-04 12:01:45   Matching barcodes to poolfile\n",
      "2023-12-04 12:01:45     Number of barcodes from poolfile seen: 98\n",
      "2023-12-04 12:01:45     Reads with barcodes from poolfile: 295838\n",
      "2023-12-04 12:01:45   Most abundant barcode: GTGCTCATAACAGACGGGGT seen 326036 times.\n",
      "2023-12-04 12:01:45   Number of reads that differ from this barcode by one base pair (likely sequencing errors):21191\n",
      "2023-12-04 12:01:45   Estimated sequencing error rate for barcodes: 6.1%\n",
      "2023-12-04 12:01:45   Barcodes seen once (highly inflated by sequencing errors): 54420\n",
      "2023-12-04 12:01:45   Barcodes seen twice (slightly inflated by sequencing errors): 5931\n",
      "2023-12-04 12:01:45   Barcodes seen three times or more: 7180\n",
      "2023-12-04 12:01:45   Chao estimate of population size (ones^2/2*twos): 250000\n",
      "2023-12-04 12:01:45 ---------------------\n",
      "2023-12-04 12:01:45   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB008_S1_L001_R1_001.fastq\n",
      "2023-12-04 12:01:45   Filtering reads on second index: GCACTA\n",
      "2023-12-04 12:01:45   Looking for sequence barcode preceeded by CGTACG\n",
      "2023-12-04 12:01:52     1191976 reads processed.\n",
      "2023-12-04 12:01:52     1138558 reads with compliant barcodes.\n",
      "2023-12-04 12:01:52     53418 reads without recognizable, compliant barcodes. Of those:\n",
      "2023-12-04 12:01:52       27450 reads without expected Dual Index.\n",
      "2023-12-04 12:01:52       22182 reads without expected sequence before the barcode region.\n",
      "2023-12-04 12:01:52       3786 reads without expeced sequence after the barcode region.\n",
      "2023-12-04 12:01:52       18 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2023-12-04 12:01:52       18 reads with quality scores less than 10\n",
      "2023-12-04 12:01:52   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023/countsFiles/OORB008_1.counts\n",
      "2023-12-04 12:01:52   Total barcodes seen (incudes sequencing errors): 40616\n",
      "2023-12-04 12:01:52   Matching barcodes to poolfile\n",
      "2023-12-04 12:01:52     Number of barcodes from poolfile seen: 95\n",
      "2023-12-04 12:01:52     Reads with barcodes from poolfile: 220439\n",
      "2023-12-04 12:01:52   Most abundant barcode: GTGCTCATAACAGACGGGGT seen 263361 times.\n",
      "2023-12-04 12:01:52   Number of reads that differ from this barcode by one base pair (likely sequencing errors):9377\n",
      "2023-12-04 12:01:52   Estimated sequencing error rate for barcodes: 3.4%\n",
      "2023-12-04 12:01:52   Barcodes seen once (highly inflated by sequencing errors): 32827\n",
      "2023-12-04 12:01:52   Barcodes seen twice (slightly inflated by sequencing errors): 3688\n",
      "2023-12-04 12:01:52   Barcodes seen three times or more: 4101\n",
      "2023-12-04 12:01:52   Chao estimate of population size (ones^2/2*twos): 150000\n",
      "2023-12-04 12:01:52 ---------------------\n",
      "2023-12-04 12:01:52   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs/OORB009_S1_L001_R1_001.fastq\n",
      "2023-12-04 12:01:52   Filtering reads on second index: TGTAGC\n",
      "2023-12-04 12:01:52   Looking for sequence barcode preceeded by CGTACG\n",
      "2023-12-04 12:01:55     553585 reads processed.\n",
      "2023-12-04 12:01:55     534493 reads with compliant barcodes.\n",
      "2023-12-04 12:01:55     19092 reads without recognizable, compliant barcodes. Of those:\n",
      "2023-12-04 12:01:55       9178 reads without expected Dual Index.\n",
      "2023-12-04 12:01:55       8504 reads without expected sequence before the barcode region.\n",
      "2023-12-04 12:01:55       1410 reads without expeced sequence after the barcode region.\n",
      "2023-12-04 12:01:55       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2023-12-04 12:01:55       0 reads with quality scores less than 10\n",
      "2023-12-04 12:01:55   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023/countsFiles/OORB009_1.counts\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2023-12-04 12:01:55   Total barcodes seen (incudes sequencing errors): 4657\n",
      "2023-12-04 12:01:55   Matching barcodes to poolfile\n",
      "2023-12-04 12:01:55     Number of barcodes from poolfile seen: 82\n",
      "2023-12-04 12:01:55     Reads with barcodes from poolfile: 122514\n",
      "2023-12-04 12:01:55   Most abundant barcode: GGTATCGGAATTGAAGATCT seen 129357 times.\n",
      "2023-12-04 12:01:55   Number of reads that differ from this barcode by one base pair (likely sequencing errors):2434\n",
      "2023-12-04 12:01:55   Estimated sequencing error rate for barcodes: 1.8%\n",
      "2023-12-04 12:01:55   Barcodes seen once (highly inflated by sequencing errors): 3430\n",
      "2023-12-04 12:01:55   Barcodes seen twice (slightly inflated by sequencing errors): 347\n",
      "2023-12-04 12:01:55   Barcodes seen three times or more: 880\n",
      "2023-12-04 12:01:55   Chao estimate of population size (ones^2/2*twos): 17000\n",
      "2023-12-04 12:01:55 ---------------------\n",
      "2023-12-04 12:01:55 Saving summary statistics for fastqs to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023/fastqSummaryStats.txt\n",
      "2023-12-04 12:01:55 Processed 4 sequence files corresponding to 4 biological samples.\n",
      "2023-12-04 12:01:55 Counted 915512 total reads for 144 mapped barcodes from the mutant pool.\n",
      "2023-12-04 12:01:55 Saving poolcount file to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_nMainLocation_12_04_2023/poolCount.txt\n"
     ]
    }
   ],
   "source": [
    "!python3 /usr2/people/clairedubin/barseq/latest_pipeline/RBseq_Count_BarCodes_v2.py -m {metafile_name} -l {logfile_name}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To convert the annotated file to csv and not to be separated by tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(poolfile,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('/usr2/people/shollyt22/shollyt22//TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv',index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_11_06_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB006_1'\n",
    "y_column = 'OORB007_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB006_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB007_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"fitness_of_TN_abundances_09_01_2023.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_11_06_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB006_1'\n",
    "y_column = 'OORB007_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB006_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB007_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances', fontsize=16, color='purple')\n",
    "\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 60)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 60)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "\n",
    "plt.savefig(\"fitness_of_TN_abundances_09_01_2023.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_11_06_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB008_1'\n",
    "y_column = 'OORB009_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB008_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB009_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"fitness_of_TN_abundances_09_01_2023.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_11_06_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB006_1'\n",
    "y_column = 'OORB008_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB006_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB008_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"fitness_of_TN_abundances_09_01_2023.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
