{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE AS NEEDED\n",
    "\n",
    "#Set this to the same as it was in BarSeq_counting.ipynb\n",
    "output_dir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencingsBarseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/TNSeq_barseq_outputs_11_06_2023'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### RHseq parameters\n",
    "\n",
    "\n",
    "    ctrl_temp and exp_temp: Control and experimental temperatures, as written in the sample names for the metafile from BarSeq_counting.ipynb\n",
    "\n",
    "    sp1, sp2: Species prefixes as in the TNseq annotation file (e.g. K. lactis is Kl) - must be the same length for both species\n",
    "\n",
    "    include_promoters: If False, only analyze barcodes which landed in the CDS of a gene. If True, analyze barcodes which landed in the promoter and CDS.\n",
    "\n",
    "    collapsed: \n",
    "    \n",
    "        If collapsed = False, fitness ratios will be calculated separately for each experimental replicate: \n",
    "    \n",
    "$$log_{2}(\\frac{experimental\\;replicate\\;1\\;normalized\\;read\\;count}{control\\;average\\;normalized\\;read\\;count});\\; log_{2}(\\frac{experimental\\;replicate\\;2\\;normalized\\;read\\;count}{control\\;average\\;normalized\\;read\\;count}),\\; etc.$$\n",
    "\n",
    "\n",
    "        If collapsed = True, the fitness ratio will be caluclated from the average of experimental replicates: \n",
    "        \n",
    "$$log_{2}\\frac{experimental\\;replicates\\;AVERAGE\\;normalized\\;read\\;count}{control\\;average\\;normalized\\;read\\;count}$$\n",
    "\n",
    "\n",
    "\n",
    "    filter_strategy:\n",
    "\n",
    "        If filter_strategy = 'coeff':\n",
    "        Barcode will be kept if in at least one temperature, the coefficient of variation is LESS THAN than coeff_var_cutoff.\n",
    "\n",
    "        If filter_strategy = 'reads':\n",
    "        Barcode will be kept if in at least one temperature, the average normalized read count is GREATER THAN  norm_reads_cutoff.\n",
    "\n",
    "        If filter_strategy = 'both':\n",
    "        Barcode will be kept if it passes BOTH read count and coefficient of variation filters as described above.\n",
    "\n",
    "    max_coeff_var_per_barcode: Maximum coefficient of variation of a barcode in a given temperature \n",
    "    \n",
    "    min_norm_reads_per_barcode: Minimum average normalized read count for a barcode in a given temperature\n",
    "    \n",
    "    min_inserts_per_allele: Minimum number of barcodes which were inserted into each allele of a gene\n",
    "    \n",
    "    max_coeffvar_per_allele: Maximum coefficient of variation of fitness ratios for each allele of a gene\n",
    "\n",
    "For reference, the following parameter set was used in the Saccharomyces Barseq paper:\n",
    "\n",
    "    collapsed = False\n",
    "    include_promoters = False\n",
    "\n",
    "    filter_strategy = 'both'\n",
    "    max_coeff_var_per_barcode = 2\n",
    "    min_norm_reads_per_barcode = 1.1\n",
    "\n",
    "    min_inserts_per_allele = 3\n",
    "    max_coeffvar_per_allele = 10\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CHANGE AS NEEDED\n",
    "\n",
    "ctrl_temp = 'OORB004_1'\n",
    "exp_temp = 'OORB005_1'\n",
    "\n",
    "sp1, sp2 = 'OORB004_1', 'OORB005_1'\n",
    "prefix_length = len(sp1)\n",
    "\n",
    "include_promoters = False\n",
    "collapsed = False\n",
    "\n",
    "filter_strategy = 'both'\n",
    "max_coeff_var_per_barcode = 2\n",
    "min_norm_reads_per_barcode = 1.1\n",
    "\n",
    "min_inserts_per_allele = 3\n",
    "max_coeffvar_per_allele = 10"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load Barseq count data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene</th>\n",
       "      <th>OORB004_1</th>\n",
       "      <th>OORB005_1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTAACAGGTACTTTGAGCCT</td>\n",
       "      <td>chr4</td>\n",
       "      <td>1861183</td>\n",
       "      <td>proteinId=2306381</td>\n",
       "      <td>107</td>\n",
       "      <td>129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGTGTGTTCTTCCAACCTAG</td>\n",
       "      <td>chr2</td>\n",
       "      <td>1265125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7413</td>\n",
       "      <td>7083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATTAAAAGAGAATTAGGCTA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6115762</td>\n",
       "      <td>proteinId=2297294</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCATCATGGACCCGTGTAAT</td>\n",
       "      <td>chr2</td>\n",
       "      <td>4661375</td>\n",
       "      <td>proteinId=2302542</td>\n",
       "      <td>1791</td>\n",
       "      <td>2385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TGTCGGCCCGCAAAAACTCA</td>\n",
       "      <td>chr7</td>\n",
       "      <td>2325983</td>\n",
       "      <td>NaN</td>\n",
       "      <td>113</td>\n",
       "      <td>120</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                barcode scaffold      pos               gene  OORB004_1  \\\n",
       "0  TTAACAGGTACTTTGAGCCT  chr4     1861183  proteinId=2306381  107         \n",
       "1  AGTGTGTTCTTCCAACCTAG  chr2     1265125  NaN                7413        \n",
       "2  ATTAAAAGAGAATTAGGCTA  chr1     6115762  proteinId=2297294  32          \n",
       "3  TCATCATGGACCCGTGTAAT  chr2     4661375  proteinId=2302542  1791        \n",
       "4  TGTCGGCCCGCAAAAACTCA  chr7     2325983  NaN                113         \n",
       "\n",
       "   OORB005_1  \n",
       "0  129        \n",
       "1  7083       \n",
       "2  31         \n",
       "3  2385       \n",
       "4  120        "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('{}/poolCount.txt'.format(output_dir), sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "ctrl_cols = [col for col in df.columns if ctrl_temp in col]\n",
    "exp_cols = [col for col in df.columns if exp_temp in col]\n",
    "competition_cols = ctrl_cols + exp_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Control columns: ['OORB004_1']\n",
      "Experimental columns: ['OORB005_1']\n"
     ]
    }
   ],
   "source": [
    "#check that these look correct\n",
    "print('Control columns: {}'.format(ctrl_cols))\n",
    "print('Experimental columns: {}'.format(exp_cols))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 1: Normalize read counts\n",
    "\n",
    "Normalized read count = (raw read count * average total mapped reads across replicates) / total mapped reads for replicate "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_mapped_reads = {}\n",
    "\n",
    "for col in competition_cols:\n",
    "    total_mapped_reads[col] = sum(df[col])\n",
    "    \n",
    "avg_total_mapped_reads = np.mean(list(total_mapped_reads.values()))\n",
    "\n",
    "for col in competition_cols:\n",
    "    df[col+'_normalized_read_count'] = avg_total_mapped_reads*df[col]/total_mapped_reads[col]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene</th>\n",
       "      <th>OORB004_1</th>\n",
       "      <th>OORB005_1</th>\n",
       "      <th>OORB004_1_normalized_read_count</th>\n",
       "      <th>OORB005_1_normalized_read_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTAACAGGTACTTTGAGCCT</td>\n",
       "      <td>chr4</td>\n",
       "      <td>1861183</td>\n",
       "      <td>proteinId=2306381</td>\n",
       "      <td>107</td>\n",
       "      <td>129</td>\n",
       "      <td>114.541364</td>\n",
       "      <td>121.031338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AGTGTGTTCTTCCAACCTAG</td>\n",
       "      <td>chr2</td>\n",
       "      <td>1265125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7413</td>\n",
       "      <td>7083</td>\n",
       "      <td>7935.468510</td>\n",
       "      <td>6645.464884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATTAAAAGAGAATTAGGCTA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6115762</td>\n",
       "      <td>proteinId=2297294</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "      <td>34.255361</td>\n",
       "      <td>29.085050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCATCATGGACCCGTGTAAT</td>\n",
       "      <td>chr2</td>\n",
       "      <td>4661375</td>\n",
       "      <td>proteinId=2302542</td>\n",
       "      <td>1791</td>\n",
       "      <td>2385</td>\n",
       "      <td>1917.229745</td>\n",
       "      <td>2237.672420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TGTCGGCCCGCAAAAACTCA</td>\n",
       "      <td>chr7</td>\n",
       "      <td>2325983</td>\n",
       "      <td>NaN</td>\n",
       "      <td>113</td>\n",
       "      <td>120</td>\n",
       "      <td>120.964244</td>\n",
       "      <td>112.587292</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                barcode scaffold      pos               gene  OORB004_1  \\\n",
       "0  TTAACAGGTACTTTGAGCCT  chr4     1861183  proteinId=2306381  107         \n",
       "1  AGTGTGTTCTTCCAACCTAG  chr2     1265125  NaN                7413        \n",
       "2  ATTAAAAGAGAATTAGGCTA  chr1     6115762  proteinId=2297294  32          \n",
       "3  TCATCATGGACCCGTGTAAT  chr2     4661375  proteinId=2302542  1791        \n",
       "4  TGTCGGCCCGCAAAAACTCA  chr7     2325983  NaN                113         \n",
       "\n",
       "   OORB005_1  OORB004_1_normalized_read_count  OORB005_1_normalized_read_count  \n",
       "0  129        114.541364                       121.031338                       \n",
       "1  7083       7935.468510                      6645.464884                      \n",
       "2  31         34.255361                        29.085050                        \n",
       "3  2385       1917.229745                      2237.672420                      \n",
       "4  120        120.964244                       112.587292                       "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 2: Filter barcodes\n",
    "\n",
    "Drop barcodes with no reads in any Barseq competitions, barcodes in noncoding regions, and barcodes mapping to the plasmid."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Beginning with 4986 barcodes seen in the TNseq pool\n",
      "\n",
      "Dropping barcodes which were seen in the TNseq pool, but not in any Barseq competitions.\n",
      "3707 barcodes remain for Barseq analysis.\n",
      "\n",
      "Dropping barcodes with only 1 read across competitions.\n",
      "3519 barcodes remain for Barseq analysis.\n",
      "\n",
      "Dropping barcodes in noncoding regions.\n",
      "1370 barcodes remain for Barseq analysis.\n",
      "\n",
      "Dropping barcodes mapping to the plasmid.\n",
      "1370 barcodes remain for Barseq analysis.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print('Beginning with {} barcodes seen in the TNseq pool'.format(df.shape[0]))\n",
    "print()\n",
    "\n",
    "#drop barcodes with no reads in any competition\n",
    "total_reads = df[competition_cols].sum(axis=1)\n",
    "df = df[total_reads >= 1]\n",
    "print('Dropping barcodes which were seen in the TNseq pool, but not in any Barseq competitions.')\n",
    "print('{} barcodes remain for Barseq analysis.'.format(df.shape[0]))\n",
    "print()\n",
    "\n",
    "#drop barcodes with only 1 read across competitions\n",
    "total_reads = df[competition_cols].sum(axis=1)\n",
    "df = df[total_reads > 1]\n",
    "print('Dropping barcodes with only 1 read across competitions.')\n",
    "print('{} barcodes remain for Barseq analysis.'.format(df.shape[0]))\n",
    "print()\n",
    "\n",
    "#drop noncoding barcodes\n",
    "df = df[~df['gene'].isna()]\n",
    "print('Dropping barcodes in noncoding regions.')\n",
    "print('{} barcodes remain for Barseq analysis.'.format(df.shape[0]))\n",
    "print()\n",
    "\n",
    "\n",
    "#drop barcodes mapping to plasmid\n",
    "df = df[df['gene']!='plasmid']\n",
    "print('Dropping barcodes mapping to the plasmid.')\n",
    "print('{} barcodes remain for Barseq analysis.'.format(df.shape[0]))\n",
    "print()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 3: Calculate average of normalized read counts and coefficient of variation across replicates for each temperature."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#replace 0 with NaN so 0 values are not counted in calculating means\n",
    "\n",
    "ctrl_norm_cols = [col + '_normalized_read_count' for col in ctrl_cols]\n",
    "exp_norm_cols = [col + '_normalized_read_count' for col in exp_cols]\n",
    "\n",
    "df[ctrl_norm_cols+exp_norm_cols] = df[ctrl_norm_cols+exp_norm_cols].replace(0.0, np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mean coeff var for OORB004_1 = nan.\n",
      "Mean coeff var for OORB005_1 = nan.\n"
     ]
    }
   ],
   "source": [
    "df[ctrl_temp+'_avg_normalized_read_count'] = df[ctrl_norm_cols].mean(axis=1)\n",
    "df[ctrl_temp+'_coeffvar'] = df[ctrl_norm_cols].std(axis=1)/df[ctrl_temp+'_avg_normalized_read_count']\n",
    "\n",
    "df[exp_temp+'_avg_normalized_read_count'] = df[exp_norm_cols].mean(axis=1)\n",
    "df[exp_temp+'_coeffvar'] = df[exp_norm_cols].std(axis=1)/df[exp_temp+'_avg_normalized_read_count']\n",
    "\n",
    "\n",
    "print('Mean coeff var for {} = {}.'.format(ctrl_temp, df[ctrl_temp+'_coeffvar'].mean()))\n",
    "print('Mean coeff var for {} = {}.'.format(exp_temp, df[exp_temp+'_coeffvar'].mean()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene</th>\n",
       "      <th>OORB004_1</th>\n",
       "      <th>OORB005_1</th>\n",
       "      <th>OORB004_1_normalized_read_count</th>\n",
       "      <th>OORB005_1_normalized_read_count</th>\n",
       "      <th>OORB004_1_avg_normalized_read_count</th>\n",
       "      <th>OORB004_1_coeffvar</th>\n",
       "      <th>OORB005_1_avg_normalized_read_count</th>\n",
       "      <th>OORB005_1_coeffvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTAACAGGTACTTTGAGCCT</td>\n",
       "      <td>chr4</td>\n",
       "      <td>1861183</td>\n",
       "      <td>proteinId=2306381</td>\n",
       "      <td>107</td>\n",
       "      <td>129</td>\n",
       "      <td>114.541364</td>\n",
       "      <td>121.031338</td>\n",
       "      <td>114.541364</td>\n",
       "      <td>NaN</td>\n",
       "      <td>121.031338</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATTAAAAGAGAATTAGGCTA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6115762</td>\n",
       "      <td>proteinId=2297294</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "      <td>34.255361</td>\n",
       "      <td>29.085050</td>\n",
       "      <td>34.255361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.085050</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCATCATGGACCCGTGTAAT</td>\n",
       "      <td>chr2</td>\n",
       "      <td>4661375</td>\n",
       "      <td>proteinId=2302542</td>\n",
       "      <td>1791</td>\n",
       "      <td>2385</td>\n",
       "      <td>1917.229745</td>\n",
       "      <td>2237.672420</td>\n",
       "      <td>1917.229745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2237.672420</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CGAGTTCAAAGATGTTCAGT</td>\n",
       "      <td>chr3</td>\n",
       "      <td>4987853</td>\n",
       "      <td>proteinId=2305679</td>\n",
       "      <td>240</td>\n",
       "      <td>303</td>\n",
       "      <td>256.915209</td>\n",
       "      <td>284.282911</td>\n",
       "      <td>256.915209</td>\n",
       "      <td>NaN</td>\n",
       "      <td>284.282911</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TTATCAGGCACGGACGAATT</td>\n",
       "      <td>chr1</td>\n",
       "      <td>547608</td>\n",
       "      <td>proteinId=2294147</td>\n",
       "      <td>395</td>\n",
       "      <td>454</td>\n",
       "      <td>422.839614</td>\n",
       "      <td>425.955253</td>\n",
       "      <td>422.839614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>425.955253</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                barcode scaffold      pos               gene  OORB004_1  \\\n",
       "0  TTAACAGGTACTTTGAGCCT  chr4     1861183  proteinId=2306381  107         \n",
       "2  ATTAAAAGAGAATTAGGCTA  chr1     6115762  proteinId=2297294  32          \n",
       "3  TCATCATGGACCCGTGTAAT  chr2     4661375  proteinId=2302542  1791        \n",
       "5  CGAGTTCAAAGATGTTCAGT  chr3     4987853  proteinId=2305679  240         \n",
       "6  TTATCAGGCACGGACGAATT  chr1     547608   proteinId=2294147  395         \n",
       "\n",
       "   OORB005_1  OORB004_1_normalized_read_count  \\\n",
       "0  129        114.541364                        \n",
       "2  31         34.255361                         \n",
       "3  2385       1917.229745                       \n",
       "5  303        256.915209                        \n",
       "6  454        422.839614                        \n",
       "\n",
       "   OORB005_1_normalized_read_count  OORB004_1_avg_normalized_read_count  \\\n",
       "0  121.031338                       114.541364                            \n",
       "2  29.085050                        34.255361                             \n",
       "3  2237.672420                      1917.229745                           \n",
       "5  284.282911                       256.915209                            \n",
       "6  425.955253                       422.839614                            \n",
       "\n",
       "   OORB004_1_coeffvar  OORB005_1_avg_normalized_read_count  OORB005_1_coeffvar  \n",
       "0 NaN                  121.031338                          NaN                  \n",
       "2 NaN                  29.085050                           NaN                  \n",
       "3 NaN                  2237.672420                         NaN                  \n",
       "5 NaN                  284.282911                          NaN                  \n",
       "6 NaN                  425.955253                          NaN                  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 4: Calculate fitness ratios\n",
    "\n",
    "\n",
    "For EACH experimental replicate: fitness ratio = log2( experimental rep normalized read count / control average normalized read count)\n",
    "\n",
    "Drop barcodes where the average fitness ratio is NaN."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_norm_cols = [col for col in df.columns if exp_temp in col and 'normalized_read_count' in col]\n",
    "\n",
    "for col in exp_norm_cols:\n",
    "    new_col = col.replace('normalized_read_count','fitness_ratio')\n",
    "    df.loc[:, new_col] = np.log2(df[col]/df[ctrl_temp+'_avg_normalized_read_count'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropping 70 barcodes where log2(exp_avg/ctrl_avg) is NaN.\n",
      "1300 barcodes remain for Barseq analysis.\n"
     ]
    }
   ],
   "source": [
    "#drop barcodes where log2(exp_avg/ctrl_avg) is NaN\n",
    "print('Dropping {} barcodes where log2(exp_avg/ctrl_avg) is NaN.'.format(df[df['{}_avg_fitness_ratio'.format(exp_temp)].isna()].shape[0]))\n",
    "df = df[~df['{}_avg_fitness_ratio'.format(exp_temp)].isna()]\n",
    "print('{} barcodes remain for Barseq analysis.'.format(df.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>pos</th>\n",
       "      <th>gene</th>\n",
       "      <th>OORB004_1</th>\n",
       "      <th>OORB005_1</th>\n",
       "      <th>OORB004_1_normalized_read_count</th>\n",
       "      <th>OORB005_1_normalized_read_count</th>\n",
       "      <th>OORB004_1_avg_normalized_read_count</th>\n",
       "      <th>OORB004_1_coeffvar</th>\n",
       "      <th>OORB005_1_avg_normalized_read_count</th>\n",
       "      <th>OORB005_1_coeffvar</th>\n",
       "      <th>OORB005_1_fitness_ratio</th>\n",
       "      <th>OORB005_1_avg_fitness_ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TTAACAGGTACTTTGAGCCT</td>\n",
       "      <td>chr4</td>\n",
       "      <td>1861183</td>\n",
       "      <td>proteinId=2306381</td>\n",
       "      <td>107</td>\n",
       "      <td>129</td>\n",
       "      <td>114.541364</td>\n",
       "      <td>121.031338</td>\n",
       "      <td>114.541364</td>\n",
       "      <td>NaN</td>\n",
       "      <td>121.031338</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.079512</td>\n",
       "      <td>0.079512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATTAAAAGAGAATTAGGCTA</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6115762</td>\n",
       "      <td>proteinId=2297294</td>\n",
       "      <td>32</td>\n",
       "      <td>31</td>\n",
       "      <td>34.255361</td>\n",
       "      <td>29.085050</td>\n",
       "      <td>34.255361</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29.085050</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.236052</td>\n",
       "      <td>-0.236052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCATCATGGACCCGTGTAAT</td>\n",
       "      <td>chr2</td>\n",
       "      <td>4661375</td>\n",
       "      <td>proteinId=2302542</td>\n",
       "      <td>1791</td>\n",
       "      <td>2385</td>\n",
       "      <td>1917.229745</td>\n",
       "      <td>2237.672420</td>\n",
       "      <td>1917.229745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2237.672420</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.222976</td>\n",
       "      <td>0.222976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CGAGTTCAAAGATGTTCAGT</td>\n",
       "      <td>chr3</td>\n",
       "      <td>4987853</td>\n",
       "      <td>proteinId=2305679</td>\n",
       "      <td>240</td>\n",
       "      <td>303</td>\n",
       "      <td>256.915209</td>\n",
       "      <td>284.282911</td>\n",
       "      <td>256.915209</td>\n",
       "      <td>NaN</td>\n",
       "      <td>284.282911</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.146035</td>\n",
       "      <td>0.146035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TTATCAGGCACGGACGAATT</td>\n",
       "      <td>chr1</td>\n",
       "      <td>547608</td>\n",
       "      <td>proteinId=2294147</td>\n",
       "      <td>395</td>\n",
       "      <td>454</td>\n",
       "      <td>422.839614</td>\n",
       "      <td>425.955253</td>\n",
       "      <td>422.839614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>425.955253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010591</td>\n",
       "      <td>0.010591</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                barcode scaffold      pos               gene  OORB004_1  \\\n",
       "0  TTAACAGGTACTTTGAGCCT  chr4     1861183  proteinId=2306381  107         \n",
       "2  ATTAAAAGAGAATTAGGCTA  chr1     6115762  proteinId=2297294  32          \n",
       "3  TCATCATGGACCCGTGTAAT  chr2     4661375  proteinId=2302542  1791        \n",
       "5  CGAGTTCAAAGATGTTCAGT  chr3     4987853  proteinId=2305679  240         \n",
       "6  TTATCAGGCACGGACGAATT  chr1     547608   proteinId=2294147  395         \n",
       "\n",
       "   OORB005_1  OORB004_1_normalized_read_count  \\\n",
       "0  129        114.541364                        \n",
       "2  31         34.255361                         \n",
       "3  2385       1917.229745                       \n",
       "5  303        256.915209                        \n",
       "6  454        422.839614                        \n",
       "\n",
       "   OORB005_1_normalized_read_count  OORB004_1_avg_normalized_read_count  \\\n",
       "0  121.031338                       114.541364                            \n",
       "2  29.085050                        34.255361                             \n",
       "3  2237.672420                      1917.229745                           \n",
       "5  284.282911                       256.915209                            \n",
       "6  425.955253                       422.839614                            \n",
       "\n",
       "   OORB004_1_coeffvar  OORB005_1_avg_normalized_read_count  \\\n",
       "0 NaN                  121.031338                            \n",
       "2 NaN                  29.085050                             \n",
       "3 NaN                  2237.672420                           \n",
       "5 NaN                  284.282911                            \n",
       "6 NaN                  425.955253                            \n",
       "\n",
       "   OORB005_1_coeffvar  OORB005_1_fitness_ratio  OORB005_1_avg_fitness_ratio  \n",
       "0 NaN                  0.079512                 0.079512                     \n",
       "2 NaN                 -0.236052                -0.236052                     \n",
       "3 NaN                  0.222976                 0.222976                     \n",
       "5 NaN                  0.146035                 0.146035                     \n",
       "6 NaN                  0.010591                 0.010591                     "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save to csv\n",
    "df.to_csv('{}/barcode_counts_and_fitness_ratios_UNFILTERED.csv'.format(output_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 4: Filter barcodes by coefficient of variation and/or average of normalized read counts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropping 1300 barcodes with coeff >= 2 and/or average normalized read count <= 1.1 in at least one temperature.\n",
      "0 barcodes remain for Barseq analysis.\n"
     ]
    }
   ],
   "source": [
    "coeff_var_cols = [exp_temp+'_coeffvar', ctrl_temp+'_coeffvar']\n",
    "avg_norm_reads_cols = [exp_temp+'_avg_normalized_read_count', ctrl_temp+'_avg_normalized_read_count']\n",
    "\n",
    "#make a copy for filtering so we don't mess up the original dataframe\n",
    "df_copy = df.copy()\n",
    "\n",
    "if filter_strategy == 'coeff':    \n",
    "    filtered_df = df_copy[df_copy[coeff_var_cols].apply(lambda x: x <= max_coeff_var_per_barcode, axis=1).sum(axis=1) >= 1]\n",
    "    print('Dropping {} barcodes with coeff >= {} in at least one temperature.'.format(df.shape[0]-filtered_df.shape[0], max_coeff_var_per_barcode))\n",
    "    \n",
    "elif filter_strategy == 'reads':\n",
    "    filtered_df = df_copy[df_copy[avg_norm_reads_cols].apply(lambda x: x >= min_norm_reads_per_barcode, axis=1).sum(axis=1) >= 1]\n",
    "    print('Dropping {} barcodes with average normalized read count <= {} in at least one temperature.'.format(df.shape[0]-filtered_df.shape[0], min_norm_reads_per_barcode))\n",
    "\n",
    "elif filter_strategy == 'both':\n",
    "    coeff_filt = df[df[coeff_var_cols].apply(lambda x: x <= max_coeff_var_per_barcode, axis=1).sum(axis=1) >= 1]\n",
    "    read_filt = df[df[avg_norm_reads_cols].apply(lambda x: x >= min_norm_reads_per_barcode, axis=1).sum(axis=1) >= 1]\n",
    "    filtered_df = df[(df.index.isin(coeff_filt.index)) & (df.index.isin(read_filt.index))]\n",
    "    print('Dropping {} barcodes with coeff >= {} and/or average normalized read count <= {} in at least one temperature.'.format(df.shape[0]-filtered_df.shape[0], max_coeff_var_per_barcode, min_norm_reads_per_barcode))\n",
    "\n",
    "print('{} barcodes remain for Barseq analysis.'.format(filtered_df.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_file = '{}/barcode_counts_and_fitness_ratios_FILTERED.csv'.format(output_dir)\n",
    "\n",
    "with open(output_file, 'w') as f:\n",
    "    #write filter parameters to file\n",
    "    f.write('# FILTER PARAMS: filter_strategy={}; max_coeff_var_per_barcode={}; min_norm_reads_per_barcode={}'.format(filter_strategy,max_coeff_var_per_barcode,min_norm_reads_per_barcode))\n",
    "    #write remaining barcodes to file\n",
    "    filtered_df.to_csv(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 6: Filter by number of inserts in each gene/allele and coefficient of variation of inserts in each gene/allele.\n",
    "\n",
    "Number of inserts means number of barcodes mapping to a given gene and allele.\n",
    "\n",
    "Coefficient of variation of inserts means the coefficient of variation of "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Defaults from Melanie's scripts, change if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "#split annotation column into gene and allele columns\n",
    "filtered_df['allele'] = filtered_df['gene'].str[:prefix_length]\n",
    "filtered_df['gene_name'] = filtered_df['gene'].str[prefix_length:]\n",
    "\n",
    "#gene names must be at least 3 characters long\n",
    "filtered_df = filtered_df[filtered_df['gene_name'].str.len() >= 3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "#group barcodes by gene/allele and count number of inserts per allele\n",
    "insert_counts_per_allele = filtered_df.groupby('gene_name')['allele'].value_counts().unstack(level=1)\n",
    "insert_counts_per_allele.columns = [col+'_insert_counts' for col in insert_counts_per_allele.columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#group barcodes by gene/allele and calculate coefficient of variation of experimental/control average normalized read counts\n",
    "def coeffvar(col):\n",
    "    return np.std(col)/np.mean(col)\n",
    "\n",
    "exp_ctrl_ratio = '{}/{}_avg_normalized_read_count'.format(exp_temp, ctrl_temp)\n",
    "filtered_df[exp_ctrl_ratio] = filtered_df[exp_temp+'_avg_normalized_read_count']/filtered_df[ctrl_temp+'_avg_normalized_read_count']\n",
    "coeff_var_by_allele = filtered_df.groupby(['gene_name', 'allele'])[exp_ctrl_ratio].apply(coeffvar).unstack(level=1)\n",
    "coeff_var_by_allele.columns = [col+'_coeff_var' for col in coeff_var_by_allele.columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge inserts per allele and coeffvar data into one frame\n",
    "inserts_per_allele = insert_counts_per_allele.merge(coeff_var_by_allele, left_index=True, right_index=True)\n",
    "inserts_per_allele.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Beginning with 0 genes with at least one insertion in either species.\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'OORB004_1_insert_counts'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py:3621\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3620\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m-> 3621\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_engine\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcasted_key\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3622\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx:136\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/_libs/index.pyx:163\u001b[0m, in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:5198\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "File \u001b[0;32mpandas/_libs/hashtable_class_helper.pxi:5206\u001b[0m, in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'OORB004_1_insert_counts'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[25], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mBeginning with \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m genes with at least one insertion in either species.\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mformat(inserts_per_allele\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m]))\n\u001b[0;32m----> 3\u001b[0m inserts_per_allele_filtered_count \u001b[38;5;241m=\u001b[39m inserts_per_allele[(\u001b[43minserts_per_allele\u001b[49m\u001b[43m[\u001b[49m\u001b[43msp1\u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m_insert_counts\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m min_inserts_per_allele)\n\u001b[1;32m      4\u001b[0m                                                  \u001b[38;5;241m&\u001b[39m (inserts_per_allele[sp2\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_insert_counts\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m min_inserts_per_allele)]\n\u001b[1;32m      6\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mDropping \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m genes with less than \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m insertions per allele in one or both species.\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mformat(inserts_per_allele\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m]\u001b[38;5;241m-\u001b[39minserts_per_allele_filtered_count\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m], min_inserts_per_allele))\n\u001b[1;32m      8\u001b[0m inserts_per_allele_filtered \u001b[38;5;241m=\u001b[39m inserts_per_allele_filtered_count[(inserts_per_allele_filtered_count[sp1\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_coeff_var\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m<\u001b[39m\u001b[38;5;241m=\u001b[39m max_coeffvar_per_allele) \n\u001b[1;32m      9\u001b[0m                                                  \u001b[38;5;241m&\u001b[39m (inserts_per_allele_filtered_count[sp2\u001b[38;5;241m+\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m_coeff_var\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m<\u001b[39m\u001b[38;5;241m=\u001b[39m max_coeffvar_per_allele)]\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/frame.py:3505\u001b[0m, in \u001b[0;36mDataFrame.__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3503\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcolumns\u001b[38;5;241m.\u001b[39mnlevels \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m:\n\u001b[1;32m   3504\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_getitem_multilevel(key)\n\u001b[0;32m-> 3505\u001b[0m indexer \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcolumns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_loc\u001b[49m\u001b[43m(\u001b[49m\u001b[43mkey\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3506\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m is_integer(indexer):\n\u001b[1;32m   3507\u001b[0m     indexer \u001b[38;5;241m=\u001b[39m [indexer]\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py:3623\u001b[0m, in \u001b[0;36mIndex.get_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3621\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine\u001b[38;5;241m.\u001b[39mget_loc(casted_key)\n\u001b[1;32m   3622\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m-> 3623\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(key) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01merr\u001b[39;00m\n\u001b[1;32m   3624\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m:\n\u001b[1;32m   3625\u001b[0m     \u001b[38;5;66;03m# If we have a listlike key, _check_indexing_error will raise\u001b[39;00m\n\u001b[1;32m   3626\u001b[0m     \u001b[38;5;66;03m#  InvalidIndexError. Otherwise we fall through and re-raise\u001b[39;00m\n\u001b[1;32m   3627\u001b[0m     \u001b[38;5;66;03m#  the TypeError.\u001b[39;00m\n\u001b[1;32m   3628\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_check_indexing_error(key)\n",
      "\u001b[0;31mKeyError\u001b[0m: 'OORB004_1_insert_counts'"
     ]
    }
   ],
   "source": [
    "print('Beginning with {} genes with at least one insertion in either species.'.format(inserts_per_allele.shape[0]))\n",
    "\n",
    "inserts_per_allele_filtered_count = inserts_per_allele[(inserts_per_allele[sp1+'_insert_counts'] >= min_inserts_per_allele)\n",
    "                                                 & (inserts_per_allele[sp2+'_insert_counts'] >= min_inserts_per_allele)]\n",
    "\n",
    "print('Dropping {} genes with less than {} insertions per allele in one or both species.'.format(inserts_per_allele.shape[0]-inserts_per_allele_filtered_count.shape[0], min_inserts_per_allele))\n",
    "\n",
    "inserts_per_allele_filtered = inserts_per_allele_filtered_count[(inserts_per_allele_filtered_count[sp1+'_coeff_var'] <= max_coeffvar_per_allele) \n",
    "                                                 & (inserts_per_allele_filtered_count[sp2+'_coeff_var'] <= max_coeffvar_per_allele)]\n",
    "\n",
    "print('Dropping {} genes with coefficient of variation greater than {} in one or both species.'.format(inserts_per_allele_filtered_count.shape[0]-inserts_per_allele_filtered.shape[0], max_coeffvar_per_allele))\n",
    "\n",
    "\n",
    "print('{} genes remain for analysis.'.format(inserts_per_allele_filtered.shape[0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'inserts_per_allele_filtered' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[26], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;66;03m#drop barcodes mapping to genes that didn't pass above filter\u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m filtered_df \u001b[38;5;241m=\u001b[39m filtered_df[filtered_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mgene_name\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39misin(\u001b[43minserts_per_allele_filtered\u001b[49m\u001b[38;5;241m.\u001b[39mindex)]\n",
      "\u001b[0;31mNameError\u001b[0m: name 'inserts_per_allele_filtered' is not defined"
     ]
    }
   ],
   "source": [
    "#drop barcodes mapping to genes that didn't pass above filter\n",
    "filtered_df = filtered_df[filtered_df['gene_name'].isin(inserts_per_allele_filtered.index)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Step 7: Calculate effect size, Mann Whitney U test and multiple hypothesis testing correction\n",
    "\n",
    "Effect size = difference of fitness ratio means"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def multiple_columns_to_list(df):\n",
    "    \n",
    "    ###take values from multiple columns (eg. fitness ratios from different replicates) and \n",
    "    ###combine into a single list\n",
    "    \n",
    "    val_list = []\n",
    "    for col in df.columns:\n",
    "        val_list += df[col].dropna().tolist()\n",
    "    return val_list\n",
    "\n",
    "fitness_ratio_columns = [col for col in filtered_df.columns if 'fitness_ratio' in col and 'avg' not in col]\n",
    "fitness_ratios_by_allele = filtered_df.groupby(['gene_name', 'allele'])[fitness_ratio_columns].apply(multiple_columns_to_list).to_frame()\n",
    "fitness_ratios_by_allele = fitness_ratios_by_allele.unstack(level=1)\n",
    "fitness_ratios_by_allele_columns = [col + '_fitness_ratios' for col in fitness_ratios_by_allele.columns.get_level_values(1)]\n",
    "fitness_ratios_by_allele.columns = fitness_ratios_by_allele_columns\n",
    "fitness_ratios_by_allele.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Effect size\n",
    "\n",
    "fitness_ratios_by_allele[sp1+'_mean'] = fitness_ratios_by_allele[fitness_ratios_by_allele_columns[0]].apply(np.mean, axis=0)\n",
    "fitness_ratios_by_allele[sp2+'_mean'] = fitness_ratios_by_allele[fitness_ratios_by_allele_columns[1]].apply(np.mean, axis=0)\n",
    "fitness_ratios_by_allele['effect_size'] = fitness_ratios_by_allele[sp1+'_mean'] - fitness_ratios_by_allele[sp2+'_mean']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Mann Whitney U statistic\n",
    "fitness_ratios_by_allele['MannWhitney_U'] = fitness_ratios_by_allele.apply(lambda x: \n",
    "                                     stats.mannwhitneyu(x[fitness_ratios_by_allele_columns[0]],\n",
    "                                                        x[fitness_ratios_by_allele_columns[1]],\n",
    "                                                       alternative='two-sided')[0], axis=1)\n",
    "                                    \n",
    "##Mann Whitney p value\n",
    "fitness_ratios_by_allele['MannWhitney_p'] = fitness_ratios_by_allele.apply(lambda x: \n",
    "                                     stats.mannwhitneyu(x[fitness_ratios_by_allele_columns[0]],\n",
    "                                                        x[fitness_ratios_by_allele_columns[1]],\n",
    "                                                       alternative='two-sided')[1], axis=1)                            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Correction for multiple hypothesis testing\n",
    "fitness_ratios_by_allele['MannWhitney_p_adj'] = smm.multipletests(fitness_ratios_by_allele['MannWhitney_p'].tolist(),\n",
    "                                                                  method='fdr_bh')[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "##merge with insert count and coefficient of variation data above\n",
    "\n",
    "fitness_ratios_by_allele = fitness_ratios_by_allele.merge(inserts_per_allele_filtered, left_index=True, right_index=True)\n",
    "fitness_ratios_by_allele = fitness_ratios_by_allele.sort_values('MannWhitney_p_adj')\n",
    "fitness_ratios_by_allele.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save to csv\n",
    "\n",
    "output_file = '{}/allele_fitness_ratios_and_p_vals.csv'.format(output_dir)\n",
    "\n",
    "with open(output_file, 'w') as f:\n",
    "    #write filter parameters to file\n",
    "    f.write('# FILTER PARAMS: filter_strategy={}; max_coeff_var_per_barcode={}; min_norm_reads_per_barcode={}; min_inserts_per_allele={}; max_coeffvar_per_allele={}'.format(filter_strategy,max_coeff_var_per_barcode,min_norm_reads_per_barcode,min_inserts_per_allele, max_coeffvar_per_allele))\n",
    "    #write remaining barcodes to file\n",
    "    fitness_ratios_by_allele.to_csv(f)\n",
    "\n",
    "fitness_ratios_by_allele.to_csv('{}/allele_fitness_ratios_and_p_vals.csv'.format(output_dir))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Anything with p <= 0.05?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fitness_ratios_by_allele[fitness_ratios_by_allele['MannWhitney_p_adj'] <= 0.05]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plotting "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def plot_fitness_ratios_by_gene(gene):\n",
    "    \n",
    "    fitness_ratio_cols = [col for col in fitness_ratios_by_allele if '_fitness_ratios' in col]\n",
    "\n",
    "    sns.kdeplot(fitness_ratios_by_allele.loc[gene][fitness_ratio_cols[0]], label=fitness_ratio_cols[0].split('_')[0])\n",
    "    sns.kdeplot(fitness_ratios_by_allele.loc[gene][fitness_ratio_cols[1]], label=fitness_ratio_cols[1].split('_')[0])\n",
    "    plt.legend()\n",
    "    plt.title(gene)\n",
    "    plt.xlabel('log2 {}/{}'.format(exp_temp, ctrl_temp))\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "plot_fitness_ratios_by_gene('YER022W')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
