{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import Seq\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Change the following parameters to your output directory (where you want output files to be written), your metafile, and TNseq output. \n",
    "\n",
    "If these files aren't in the same directory as this notebook, you need use the full path. For example, the full path of 'barseq_output' would be '/usr2/people/clairedubin/barseq/barseq_output'\n",
    "\n",
    "Also change the control and experimental temperatures if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where the BarSeq FASTQs are\n",
    "FASTQ_directory = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Barseq metafile\n",
    "\n",
    "You'll need to create your own metafile - easiest way is edit in Excel. A sample metafile is located at: /usr2/people/clairedubin/barseq/Kluyv_BarSeq_metadata.txt\n",
    "\n",
    "    Fastq: path to barseq FASTQ for each competition\n",
    "    SampleName: unique name for each competition, MUST include ctrl_temp or exp_temp as specified above\n",
    "    DualIndex: index from sequencing (in read name lines of FASTQ and the sequencing info file that Adam sends)\n",
    "    \n",
    "    UsePrecounted: True if you've already run the Barseq counting script on a competition, False otherwise\n",
    "    \n",
    "    poolfile: path to Tnseq output file, should be the same for each competition\n",
    "    output_dir: path to output directory\n",
    "    minRandom: number of bases between start of read and DualIndex, usually in sequence of 1,2,3,4,1,2,3,4 for each sample\n",
    "    maxRandom: same as minRandom\n",
    "    \n",
    "    BeforeBarcode: sequence preceding all barcodes\n",
    "    AfterBarcode: sequence following all barcodes\n",
    "    BarcodeLengths: lengths of barcodes to search for, with expected length of barcode first\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB001_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB002_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB003_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB004_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB005_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB006_S1_L001_R1_001.fastq\r\n"
     ]
    }
   ],
   "source": [
    "!ls {FASTQ_directory}/*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format: sample_name, fastq_path, dual_index, offset\n",
    "#add or delete more lines as needed\n",
    "\n",
    "sample_info = [('KFRB001_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB001_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB002_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB002_S1_L001_R1_001.fastq', 'CGATGT', '2'), \n",
    "               ('KFRB003_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB003_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB004_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB004_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('KFRB005_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB005_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('KFRB006_1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB006_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@M02248:14:000000000-DNH2T:1:1102:15643:1331 1:N:0:NTCACG\r\n",
      "AGCACTAGTCGACCTGCAGCGTACGTATATGCGGTCCATGTTAGTAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "CCCCCFFFFFCBGGGGGGGGGEEFEGHHHHHHGGGGGHHHHHHHHHHHHHHGHHGHHGGHHHHGHHHHHGEGGGHHGHHHGHHHGFHHHHHHHHHGHHHHHGFEHFHHHGGHHHHGGGHGHHHHHHHCFGHHGGGG@A--\r\n",
      "@M02248:14:000000000-DNH2T:1:1102:16005:1334 1:N:0:NTCACG\r\n",
      "TTATTTCCTAGACAAATTAGAGCCAATACCATCAGCTTTACCGTCTTTCCAGAAATTGTTCCAAGTATCGGCAACAGCTTTATCAATACCATGAAAAATATCAACCACACCAGAAGCAGCATCAGTGACGACACTTGGAA\r\n",
      "+\r\n",
      "AAAAAFFFDFFFGGGG1BFGF1FFCGHHFFGFHHBGDHHGDGC0BAFDDGHFFG1FFFADEGHHHAEFFE/E/EFF0FFGGFHHHH1GFGHGFBGGHHHHDHFFBGCFFAEAGHHFFH0FGHHHBGFGH/FGGCFHGHFG\r\n",
      "@M02248:14:000000000-DNH2T:1:1102:15502:1334 1:N:0:NTCACG\r\n",
      "GGCACTAGTCGACCTGCAGCGTACGACGGTGCTCCAAGTGTGGGCAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAATAT\r\n"
     ]
    }
   ],
   "source": [
    "!head /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB001_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if there are any samples where you already counted barcodes, add the sample names here\n",
    "precounted_samples = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these variables should be the same for each replicate\n",
    "\n",
    "metafile_name = 'BarSeq_metafile_02_29_2024.txt'\n",
    "logfile_name = 'BarSeq_counting_02_29_2024.log'\n",
    "poolfile = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv'\n",
    "output_dir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024'\n",
    "before_barcode = 'GTCGACCTGCAGCGTACG' #bottom strand sequence was used\n",
    "#before_barcode = 'GATGTCCACGAGGTCTCT' what was there before \n",
    "after_barcode =  'AGAGACCTCGTGGACATC' #bottom strand sequence was used\n",
    "#after_barcode = 'CGTACGCTGCAGGTCGAC'  what was there before \n",
    "barcode_lengths = '20,19,18,21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['FileIndex','SampleName', 'Fastq', 'DualIndex', 'Poolfile', 'OutputDir', 'minRandom', 'maxRandom',\n",
    "                   'BeforeBarcode', 'AfterBarcode', 'BarcodeLengths', 'UsePrecounted']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for sample_name, fastq, dual_index, offset in sample_info:\n",
    "        \n",
    "        #DUAL INDECES ARE BACKWARDS FROM WHAT THE FLOWCELL OUTPUT SAYS\n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index[::-1], poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'False'])\n",
    "        f.write(to_write+'\\n')\n",
    "        \n",
    "    for precounted_sample_name in precounted_samples:\n",
    "        \n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index, poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'True'])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileIndex</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>DualIndex</th>\n",
       "      <th>Poolfile</th>\n",
       "      <th>OutputDir</th>\n",
       "      <th>minRandom</th>\n",
       "      <th>maxRandom</th>\n",
       "      <th>BeforeBarcode</th>\n",
       "      <th>AfterBarcode</th>\n",
       "      <th>BarcodeLengths</th>\n",
       "      <th>UsePrecounted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KFRB001_1</td>\n",
       "      <td>KFRB001_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB001_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KFRB002_1</td>\n",
       "      <td>KFRB002_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB002_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KFRB003_1</td>\n",
       "      <td>KFRB003_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB003_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KFRB004_1</td>\n",
       "      <td>KFRB004_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB004_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KFRB005_1</td>\n",
       "      <td>KFRB005_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB005_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KFRB006_1</td>\n",
       "      <td>KFRB006_1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB006_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   FileIndex SampleName  \\\n",
       "0  KFRB001_1  KFRB001_1   \n",
       "1  KFRB002_1  KFRB002_1   \n",
       "2  KFRB003_1  KFRB003_1   \n",
       "3  KFRB004_1  KFRB004_1   \n",
       "4  KFRB005_1  KFRB005_1   \n",
       "5  KFRB006_1  KFRB006_1   \n",
       "\n",
       "                                                                                                                                        Fastq  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB001_S1_L001_R1_001.fastq   \n",
       "1  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB002_S1_L001_R1_001.fastq   \n",
       "2  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB003_S1_L001_R1_001.fastq   \n",
       "3  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB004_S1_L001_R1_001.fastq   \n",
       "4  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB005_S1_L001_R1_001.fastq   \n",
       "5  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB006_S1_L001_R1_001.fastq   \n",
       "\n",
       "  DualIndex  \\\n",
       "0  GCACTA     \n",
       "1  TGTAGC     \n",
       "2  GCACTA     \n",
       "3  TGTAGC     \n",
       "4  GCACTA     \n",
       "5  TGTAGC     \n",
       "\n",
       "                                                                                                                                                                                   Poolfile  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv   \n",
       "1  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv   \n",
       "2  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv   \n",
       "3  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv   \n",
       "4  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv   \n",
       "5  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv   \n",
       "\n",
       "                                                                                                                            OutputDir  \\\n",
       "0  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024   \n",
       "1  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024   \n",
       "2  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024   \n",
       "3  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024   \n",
       "4  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024   \n",
       "5  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024   \n",
       "\n",
       "   minRandom  maxRandom       BeforeBarcode        AfterBarcode  \\\n",
       "0  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "1  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "2  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "3  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "4  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "5  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "\n",
       "  BarcodeLengths  UsePrecounted  \n",
       "0  20,19,18,21    False          \n",
       "1  20,19,18,21    False          \n",
       "2  20,19,18,21    False          \n",
       "3  20,19,18,21    False          \n",
       "4  20,19,18,21    False          \n",
       "5  20,19,18,21    False          "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Barseq counting\n",
    "\n",
    "Might take a few hours!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-02-29 18:34:56 RBseq_Count_BarCodes.py\n",
      "2024-02-29 18:34:56 Version: 1.1.4\n",
      "2024-02-29 18:34:56 Release Date: July 16, 2020\n",
      "2024-02-29 18:34:56 Options passed:  metafile:BarSeq_metafile_02_29_2024.txt  logFile:BarSeq_counting_02_29_2024.log  minQual:10  matchBefore:6  matchAfter:6  quietMode:False \n",
      "2024-02-29 18:34:56 Logging status updates in BarSeq_counting_02_29_2024.log\n",
      "2024-02-29 18:34:56 Loading TnSeq library metadata from BarSeq_metafile_02_29_2024.txt\n",
      "2024-02-29 18:34:56 Setting output directory as: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024 (from first line of metadata file)\n",
      "2024-02-29 18:34:56 Loading mapped barcodes in mutant pool from: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv (from first line of metadata file)\n",
      "2024-02-29 18:34:56 Read 4986 barcodes from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_mapping_output_new_genome_without_vector_seq_11_04_2023/Tnseq_11_04_2023_annotated_mod1.csv\n",
      "2024-02-29 18:34:56 Finding barcodes in fastqs and counting occurances\n",
      "2024-02-29 18:34:56 ---------------------\n",
      "2024-02-29 18:34:56   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB001_S1_L001_R1_001.fastq\n",
      "2024-02-29 18:34:56   Filtering reads on second index: GCACTA\n",
      "2024-02-29 18:34:56   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-02-29 18:35:02     1192746 reads processed.\n",
      "2024-02-29 18:35:02     1138095 reads with compliant barcodes.\n",
      "2024-02-29 18:35:02     54651 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-02-29 18:35:02       26286 reads without expected Dual Index.\n",
      "2024-02-29 18:35:02       22098 reads without expected sequence before the barcode region.\n",
      "2024-02-29 18:35:02       6267 reads without expeced sequence after the barcode region.\n",
      "2024-02-29 18:35:02       2 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-02-29 18:35:02       2 reads with quality scores less than 10\n",
      "2024-02-29 18:35:02   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/countsFiles/KFRB001_1.counts\n",
      "2024-02-29 18:35:02   Total barcodes seen (incudes sequencing errors): 95859\n",
      "2024-02-29 18:35:02   Matching barcodes to poolfile\n",
      "2024-02-29 18:35:02     Number of barcodes from poolfile seen: 2803\n",
      "2024-02-29 18:35:02     Reads with barcodes from poolfile: 513690\n",
      "2024-02-29 18:35:02   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 110463 times.\n",
      "2024-02-29 18:35:02   Number of reads that differ from this barcode by one base pair (likely sequencing errors):6921\n",
      "2024-02-29 18:35:02   Estimated sequencing error rate for barcodes: 5.9%\n",
      "2024-02-29 18:35:02   Barcodes seen once (highly inflated by sequencing errors): 80059\n",
      "2024-02-29 18:35:02   Barcodes seen twice (slightly inflated by sequencing errors): 5671\n",
      "2024-02-29 18:35:02   Barcodes seen three times or more: 10129\n",
      "2024-02-29 18:35:02   Chao estimate of population size (ones^2/2*twos): 570000\n",
      "2024-02-29 18:35:02 ---------------------\n",
      "2024-02-29 18:35:02   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB002_S1_L001_R1_001.fastq\n",
      "2024-02-29 18:35:02   Filtering reads on second index: TGTAGC\n",
      "2024-02-29 18:35:02   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-02-29 18:35:09     1279289 reads processed.\n",
      "2024-02-29 18:35:09     1228497 reads with compliant barcodes.\n",
      "2024-02-29 18:35:09     50792 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-02-29 18:35:09       20511 reads without expected Dual Index.\n",
      "2024-02-29 18:35:09       23903 reads without expected sequence before the barcode region.\n",
      "2024-02-29 18:35:09       6378 reads without expeced sequence after the barcode region.\n",
      "2024-02-29 18:35:09       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-02-29 18:35:09       0 reads with quality scores less than 10\n",
      "2024-02-29 18:35:09   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/countsFiles/KFRB002_1.counts\n",
      "2024-02-29 18:35:09   Total barcodes seen (incudes sequencing errors): 57482\n",
      "2024-02-29 18:35:09   Matching barcodes to poolfile\n",
      "2024-02-29 18:35:09     Number of barcodes from poolfile seen: 2814\n",
      "2024-02-29 18:35:09     Reads with barcodes from poolfile: 564270\n",
      "2024-02-29 18:35:09   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 108898 times.\n",
      "2024-02-29 18:35:09   Number of reads that differ from this barcode by one base pair (likely sequencing errors):5922\n",
      "2024-02-29 18:35:09   Estimated sequencing error rate for barcodes: 5.2%\n",
      "2024-02-29 18:35:09   Barcodes seen once (highly inflated by sequencing errors): 43897\n",
      "2024-02-29 18:35:09   Barcodes seen twice (slightly inflated by sequencing errors): 4207\n",
      "2024-02-29 18:35:09   Barcodes seen three times or more: 9378\n",
      "2024-02-29 18:35:09   Chao estimate of population size (ones^2/2*twos): 230000\n",
      "2024-02-29 18:35:09 ---------------------\n",
      "2024-02-29 18:35:09   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB003_S1_L001_R1_001.fastq\n",
      "2024-02-29 18:35:09   Filtering reads on second index: GCACTA\n",
      "2024-02-29 18:35:09   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-02-29 18:35:15     1179209 reads processed.\n",
      "2024-02-29 18:35:15     1127153 reads with compliant barcodes.\n",
      "2024-02-29 18:35:15     52056 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-02-29 18:35:15       23430 reads without expected Dual Index.\n",
      "2024-02-29 18:35:15       20972 reads without expected sequence before the barcode region.\n",
      "2024-02-29 18:35:15       7654 reads without expeced sequence after the barcode region.\n",
      "2024-02-29 18:35:15       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-02-29 18:35:15       0 reads with quality scores less than 10\n",
      "2024-02-29 18:35:15   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/countsFiles/KFRB003_1.counts\n",
      "2024-02-29 18:35:15   Total barcodes seen (incudes sequencing errors): 99602\n",
      "2024-02-29 18:35:15   Matching barcodes to poolfile\n",
      "2024-02-29 18:35:15     Number of barcodes from poolfile seen: 2946\n",
      "2024-02-29 18:35:15     Reads with barcodes from poolfile: 522022\n",
      "2024-02-29 18:35:15   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 122406 times.\n",
      "2024-02-29 18:35:15   Number of reads that differ from this barcode by one base pair (likely sequencing errors):7256\n",
      "2024-02-29 18:35:15   Estimated sequencing error rate for barcodes: 5.6%\n",
      "2024-02-29 18:35:15   Barcodes seen once (highly inflated by sequencing errors): 82839\n",
      "2024-02-29 18:35:15   Barcodes seen twice (slightly inflated by sequencing errors): 6083\n",
      "2024-02-29 18:35:15   Barcodes seen three times or more: 10680\n",
      "2024-02-29 18:35:15   Chao estimate of population size (ones^2/2*twos): 560000\n",
      "2024-02-29 18:35:15 ---------------------\n",
      "2024-02-29 18:35:15   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB004_S1_L001_R1_001.fastq\n",
      "2024-02-29 18:35:15   Filtering reads on second index: TGTAGC\n",
      "2024-02-29 18:35:15   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-02-29 18:35:21     1205117 reads processed.\n",
      "2024-02-29 18:35:21     1160305 reads with compliant barcodes.\n",
      "2024-02-29 18:35:21     44812 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-02-29 18:35:21       17856 reads without expected Dual Index.\n",
      "2024-02-29 18:35:21       21424 reads without expected sequence before the barcode region.\n",
      "2024-02-29 18:35:21       5532 reads without expeced sequence after the barcode region.\n",
      "2024-02-29 18:35:21       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-02-29 18:35:21       0 reads with quality scores less than 10\n",
      "2024-02-29 18:35:21   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/countsFiles/KFRB004_1.counts\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-02-29 18:35:22   Total barcodes seen (incudes sequencing errors): 50950\n",
      "2024-02-29 18:35:22   Matching barcodes to poolfile\n",
      "2024-02-29 18:35:22     Number of barcodes from poolfile seen: 2992\n",
      "2024-02-29 18:35:22     Reads with barcodes from poolfile: 566359\n",
      "2024-02-29 18:35:22   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 124263 times.\n",
      "2024-02-29 18:35:22   Number of reads that differ from this barcode by one base pair (likely sequencing errors):5571\n",
      "2024-02-29 18:35:22   Estimated sequencing error rate for barcodes: 4.3%\n",
      "2024-02-29 18:35:22   Barcodes seen once (highly inflated by sequencing errors): 37400\n",
      "2024-02-29 18:35:22   Barcodes seen twice (slightly inflated by sequencing errors): 4176\n",
      "2024-02-29 18:35:22   Barcodes seen three times or more: 9374\n",
      "2024-02-29 18:35:22   Chao estimate of population size (ones^2/2*twos): 170000\n",
      "2024-02-29 18:35:22 ---------------------\n",
      "2024-02-29 18:35:22   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB005_S1_L001_R1_001.fastq\n",
      "2024-02-29 18:35:22   Filtering reads on second index: GCACTA\n",
      "2024-02-29 18:35:22   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-02-29 18:35:28     1142319 reads processed.\n",
      "2024-02-29 18:35:28     1099239 reads with compliant barcodes.\n",
      "2024-02-29 18:35:28     43080 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-02-29 18:35:28       20284 reads without expected Dual Index.\n",
      "2024-02-29 18:35:28       17467 reads without expected sequence before the barcode region.\n",
      "2024-02-29 18:35:28       5329 reads without expeced sequence after the barcode region.\n",
      "2024-02-29 18:35:28       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-02-29 18:35:28       0 reads with quality scores less than 10\n",
      "2024-02-29 18:35:28   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/countsFiles/KFRB005_1.counts\n",
      "2024-02-29 18:35:28   Total barcodes seen (incudes sequencing errors): 88860\n",
      "2024-02-29 18:35:28   Matching barcodes to poolfile\n",
      "2024-02-29 18:35:28     Number of barcodes from poolfile seen: 3049\n",
      "2024-02-29 18:35:28     Reads with barcodes from poolfile: 506496\n",
      "2024-02-29 18:35:28   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 93462 times.\n",
      "2024-02-29 18:35:28   Number of reads that differ from this barcode by one base pair (likely sequencing errors):3958\n",
      "2024-02-29 18:35:28   Estimated sequencing error rate for barcodes: 4.1%\n",
      "2024-02-29 18:35:28   Barcodes seen once (highly inflated by sequencing errors): 74210\n",
      "2024-02-29 18:35:28   Barcodes seen twice (slightly inflated by sequencing errors): 4908\n",
      "2024-02-29 18:35:28   Barcodes seen three times or more: 9742\n",
      "2024-02-29 18:35:28   Chao estimate of population size (ones^2/2*twos): 560000\n",
      "2024-02-29 18:35:28 ---------------------\n",
      "2024-02-29 18:35:28   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/BarSeq_FASTQs_Feb_2024/KFRB006_S1_L001_R1_001.fastq\n",
      "2024-02-29 18:35:28   Filtering reads on second index: TGTAGC\n",
      "2024-02-29 18:35:28   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-02-29 18:35:34     1115874 reads processed.\n",
      "2024-02-29 18:35:34     1078435 reads with compliant barcodes.\n",
      "2024-02-29 18:35:34     37439 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-02-29 18:35:34       16098 reads without expected Dual Index.\n",
      "2024-02-29 18:35:34       16523 reads without expected sequence before the barcode region.\n",
      "2024-02-29 18:35:34       4818 reads without expeced sequence after the barcode region.\n",
      "2024-02-29 18:35:34       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-02-29 18:35:34       0 reads with quality scores less than 10\n",
      "2024-02-29 18:35:34   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/countsFiles/KFRB006_1.counts\n",
      "2024-02-29 18:35:34   Total barcodes seen (incudes sequencing errors): 57745\n",
      "2024-02-29 18:35:34   Matching barcodes to poolfile\n",
      "2024-02-29 18:35:34     Number of barcodes from poolfile seen: 3036\n",
      "2024-02-29 18:35:34     Reads with barcodes from poolfile: 519527\n",
      "2024-02-29 18:35:34   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 104635 times.\n",
      "2024-02-29 18:35:34   Number of reads that differ from this barcode by one base pair (likely sequencing errors):4001\n",
      "2024-02-29 18:35:34   Estimated sequencing error rate for barcodes: 3.7%\n",
      "2024-02-29 18:35:34   Barcodes seen once (highly inflated by sequencing errors): 44881\n",
      "2024-02-29 18:35:34   Barcodes seen twice (slightly inflated by sequencing errors): 3884\n",
      "2024-02-29 18:35:34   Barcodes seen three times or more: 8980\n",
      "2024-02-29 18:35:34   Chao estimate of population size (ones^2/2*twos): 260000\n",
      "2024-02-29 18:35:34 ---------------------\n",
      "2024-02-29 18:35:34 Saving summary statistics for fastqs to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/fastqSummaryStats.txt\n",
      "2024-02-29 18:35:34 Processed 6 sequence files corresponding to 6 biological samples.\n",
      "2024-02-29 18:35:34 Counted 3192364 total reads for 3424 mapped barcodes from the mutant pool.\n",
      "2024-02-29 18:35:34 Saving poolcount file to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp_repeat_02_29_2024/poolCount.txt\n"
     ]
    }
   ],
   "source": [
    "!python3 /usr2/people/clairedubin/barseq/latest_pipeline/RBseq_Count_BarCodes_v2.py -m {metafile_name} -l {logfile_name}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To convert the annotated file to csv and not to be separated by tab ##This section is not necessary here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB0010_1'\n",
    "y_column = 'OORB0011_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB0010_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB0011_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances (37 oC)', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_01_03_2024_37oC.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB0010_1'\n",
    "y_column = 'OORB0011_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB010_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB011_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances (37 oC)', fontsize=16, color='purple')\n",
    "\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 15000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 15000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "\n",
    "plt.savefig(\"fitness_of_TN_abundances_03_01_2024_37oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB0012_1'\n",
    "y_column = 'OORB0013_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB012_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB013_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances (45 oC)', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "#plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'OORB0012_1'\n",
    "y_column = 'OORB0013_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('OORB012_1 Abundance', fontsize=14)\n",
    "plt.ylabel('OORB013_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 6000)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 6000)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"fitness_of_TN_abundances_03_01_2024_45oC_1a.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## to normalize the Barseq pool count file based on the number of reads obtained"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### normalization based on individual sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_OORB0010_1'] = df['OORB0010_1'].sum()\n",
    "df['Total_Count_OORB0011_1'] = df['OORB0011_1'].sum()\n",
    "df['Total_Count_OORB0012_1'] = df['OORB0012_1'].sum()\n",
    "df['Total_Count_OORB0013_1'] = df['OORB0013_1'].sum()\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['OORB0010_1'] = df['OORB0010_1'] / df['Total_Count_OORB0010_1']\n",
    "df['OORB0011_1'] = df['OORB0011_1'] / df['Total_Count_OORB0011_1']\n",
    "df['OORB0012_1'] = df['OORB0012_1'] / df['Total_Count_OORB0012_1']\n",
    "df['OORB0013_1'] = df['OORB0013_1'] / df['Total_Count_OORB0013_1']\n",
    "\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts_01_03_2023.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_OORB0010_1'] = df['OORB0010_1'].sum()\n",
    "df['Total_Count_OORB0011_1'] = df['OORB0011_1'].sum()\n",
    "df['Total_Count_OORB0012_1'] = df['OORB0012_1'].sum()\n",
    "df['Total_Count_OORB0013_1'] = df['OORB0013_1'].sum()\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['Normalized_OORB0010_1'] = (df['OORB0010_1'] / df['Total_Count_OORB0010_1']).round(3)\n",
    "df['Normalized_OORB0011_1'] = (df['OORB0011_1'] / df['Total_Count_OORB0011_1']).round(3)\n",
    "df['Normalized_OORB0012_1'] = (df['OORB0012_1'] / df['Total_Count_OORB0012_1']).round(3)\n",
    "df['Normalized_OORB0013_1'] = (df['OORB0013_1'] / df['Total_Count_OORB0013_1']).round(3)\n",
    "\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts_01_03_2023_mod.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### this is the normalization that worked with the right result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/poolCount.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_OORB0010_1'] = df['OORB0010_1'].sum()\n",
    "df['Total_Count_OORB0011_1'] = df['OORB0011_1'].sum()\n",
    "df['Total_Count_OORB0012_1'] = df['OORB0012_1'].sum()\n",
    "df['Total_Count_OORB0013_1'] = df['OORB0013_1'].sum()\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['Normalized_OORB0010_1'] = (df['OORB0010_1'] / df['Total_Count_OORB0010_1'])\n",
    "df['Normalized_OORB0011_1'] = (df['OORB0011_1'] / df['Total_Count_OORB0011_1'])\n",
    "df['Normalized_OORB0012_1'] = (df['OORB0012_1'] / df['Total_Count_OORB0012_1'])\n",
    "df['Normalized_OORB0013_1'] = (df['OORB0013_1'] / df['Total_Count_OORB0013_1'])\n",
    "\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts_01_03_2023_mod1.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To make the fitness abundance agaian using the normalized values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the CSV file\n",
    "csv_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts_01_03_2023_mod1.csv'\n",
    "df = pd.read_csv(csv_file_path)\n",
    "\n",
    "# Save the DataFrame to a text file (txt)\n",
    "txt_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts.txt'\n",
    "df.to_csv(txt_file_path, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'Normalized_OORB0010_1'\n",
    "y_column = 'Normalized_OORB0011_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_OORB0010_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_OORB0011_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances (37 oC)', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"fitness_of_TN_abundances_01_03_2024_37oC_Normalized.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'Normalized_OORB0012_1'\n",
    "y_column = 'Normalized_OORB0013_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_OORB0012_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_OORB0013_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances (45 oC)', fontsize=16, color='purple')\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"fitness_of_TN_abundances_01_03_2024_45oC_Normalized.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'Normalized_OORB0010_1'\n",
    "y_column = 'Normalized_OORB0011_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_OORB0010_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_OORB0011_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_37 oC_normalized', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.05)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.05)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_37oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/TNSeq_barseq_outputs_Temp2_12_31_2023/normalized_barseq_pool_counts.txt', sep='\\t')\n",
    "\n",
    "# Select the columns you want to use for the scatter plot\n",
    "x_column = 'Normalized_OORB0012_1'\n",
    "y_column = 'Normalized_OORB0013_1'\n",
    "\n",
    "# Create a scatter plot using the selected columns\n",
    "plt.scatter(df[x_column], df[y_column], color='black')\n",
    "plt.xlabel('Normalized_OORB0012_1 Abundance', fontsize=14)\n",
    "plt.ylabel('Normalized_OORB0013_1 Abundance', fontsize=14)\n",
    "plt.title('Plot of TN_insertion abundances_45 oC_normalized', fontsize=16, color='purple')\n",
    "# Adjust the x and y-axis scale ranges\n",
    "plt.xlim(0, 0.05)  # Adjust the range for the x-axis\n",
    "plt.ylim(0, 0.05)  # Adjust the range for the y-axis\n",
    "\n",
    "# Optionally, add grid lines\n",
    "#plt.grid(True)\n",
    "#plt.grid(True)  # Optionally, add grid lines\n",
    "plt.savefig(\"/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_Temp_study_11_04_2023/histogram_plots/fitness_of_TN_abundances_01_09_2024_45oC_Normalized_1.png\", dpi=600, bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
