{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cb4ff792",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the files\n",
    "file1 = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount.txt', delimiter='\\t')\n",
    "file2 = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_06_12_2024/poolCount.txt', delimiter='\\t')\n",
    "\n",
    "# Extract the desired column from file1\n",
    "column_to_add = file1(['KFRB0029_37A1'], ['KFRB0029_37B1'])\n",
    "\n",
    "# Add this column to file2\n",
    "file2['KFRB0029_37A1'] = column_to_add\n",
    "\n",
    "# Save the result\n",
    "file2.to_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_normalized_joined_to_older_file.txt', index=False, sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8372c607",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the files\n",
    "file1 = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount.txt', delimiter='\\t')\n",
    "file2 = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_06_12_2024/poolCount.txt', delimiter='\\t')\n",
    "\n",
    "# Extract the desired columns from file1\n",
    "columns_to_add = file1[['KFRB0029_37A1', 'KFRB0029_37B1']]\n",
    "\n",
    "# Add these columns to file2\n",
    "file2 = pd.concat([file2, columns_to_add], axis=1)\n",
    "\n",
    "# Remove the specified columns from file2\n",
    "columns_to_remove = ['KFRB007_1', 'KFRB008_1', 'KFRB019_1', 'KFRB020_1']\n",
    "file2 = file2.drop(columns=columns_to_remove)\n",
    "\n",
    "# Rename columns in file2\n",
    "columns_to_rename = {\n",
    "    'KFRB0029_37A1': '37•C (A)',\n",
    "    'KFRB0029_37B1': '37•C (B)',\n",
    "    'KFRB021_1': '45•C (A)',\n",
    "    'KFRB022_1': '45•C (B)',\n",
    "    'KFRB023_1': '45•C (C)',\n",
    "    'KFRB024_1': '45•C (D)',\n",
    "    'KFRB025_1': '50•C (A)',\n",
    "    'KFRB026_1': '50•C (B)',\n",
    "    'KFRB027_1': '50•C (C)',\n",
    "    'KFRB028_1': '50•C (D)'\n",
    "}\n",
    "file2 = file2.rename(columns=columns_to_rename)\n",
    "\n",
    "# Specify the desired column order\n",
    "desired_column_order = [\n",
    "    'barcode', 'scaffold', 'pos', 'gene', '37•C (A)', '37•C (B)', \n",
    "    '45•C (A)', '45•C (B)', '45•C (C)', '45•C (D)',\n",
    "    '50•C (A)', '50•C (B)', '50•C (C)', '50•C (D)'\n",
    "]\n",
    "\n",
    "# Ensure all desired columns are present in file2\n",
    "missing_columns = [col for col in desired_column_order if col not in file2.columns]\n",
    "if missing_columns:\n",
    "    print(f\"Warning: The following columns are missing and cannot be reordered: {missing_columns}\")\n",
    "    # Adjust the desired_column_order to include only available columns\n",
    "    desired_column_order = [col for col in desired_column_order if col in file2.columns]\n",
    "\n",
    "# Reorder the columns\n",
    "file2 = file2[desired_column_order]\n",
    "\n",
    "# Save the result\n",
    "file2.to_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_37•C_repeated_09_30_2024/poolCount_new_37•C_with_older_50•C_1.txt', index=False, sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ad90bd5e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
