{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import Bio\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "thermo_gff = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Barseq_July_2023/OORB003_TnSeq/for_analysis/modified_gff_file_finalized_08_08_2023'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>scaffold</td>\n",
       "      <td>source</td>\n",
       "      <td>type</td>\n",
       "      <td>start</td>\n",
       "      <td>stop</td>\n",
       "      <td>strand</td>\n",
       "      <td>info</td>\n",
       "      <td>ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1381.0</td>\n",
       "      <td>2142.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_1</td>\n",
       "      <td>proteinId=2114025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2744.0</td>\n",
       "      <td>3343.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_2</td>\n",
       "      <td>proteinId=2293935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3344.0</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_3</td>\n",
       "      <td>proteinId=2293936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>5330.0</td>\n",
       "      <td>6385.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_4</td>\n",
       "      <td>proteinId=2051335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0       1     2       3       4       5          6  \\\n",
       "0  scaffold  source  type   start    stop  strand       info   \n",
       "1      chr1     JGI   CDS  1381.0  2142.0       +  ID=gene_1   \n",
       "2      chr1     JGI   CDS  2744.0  3343.0       -  ID=gene_2   \n",
       "3      chr1     JGI   CDS  3344.0  3817.0       -  ID=gene_3   \n",
       "4      chr1     JGI   CDS  5330.0  6385.0       -  ID=gene_4   \n",
       "\n",
       "                   7  \n",
       "0                 ID  \n",
       "1  proteinId=2114025  \n",
       "2  proteinId=2293935  \n",
       "3  proteinId=2293936  \n",
       "4  proteinId=2051335  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "thermo = pd.read_csv(thermo_gff, sep='\\t', header=None, comment='#')\n",
    "\n",
    "thermo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_path = 'Klac_Kmar_synonly.gff.txt'\n",
    "fasta_path = 'Klac_Kmar.fsa.txt'\n",
    "orthologs_path = '~/genomes_annotations/recovery_orthofinder_FINAL_wsacc.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>strand</th>\n",
       "      <th>info</th>\n",
       "      <th>ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>scaffold</td>\n",
       "      <td>source</td>\n",
       "      <td>type</td>\n",
       "      <td>start</td>\n",
       "      <td>stop</td>\n",
       "      <td>strand</td>\n",
       "      <td>info</td>\n",
       "      <td>ID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>1381.0</td>\n",
       "      <td>2142.0</td>\n",
       "      <td>+</td>\n",
       "      <td>ID=gene_1</td>\n",
       "      <td>proteinId=2114025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>2744.0</td>\n",
       "      <td>3343.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_2</td>\n",
       "      <td>proteinId=2293935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>3344.0</td>\n",
       "      <td>3817.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_3</td>\n",
       "      <td>proteinId=2293936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>JGI</td>\n",
       "      <td>CDS</td>\n",
       "      <td>5330.0</td>\n",
       "      <td>6385.0</td>\n",
       "      <td>-</td>\n",
       "      <td>ID=gene_4</td>\n",
       "      <td>proteinId=2051335</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   scaffold  source  type   start    stop  strand       info  \\\n",
       "0  scaffold  source  type   start    stop  strand       info   \n",
       "1      chr1     JGI   CDS  1381.0  2142.0       +  ID=gene_1   \n",
       "2      chr1     JGI   CDS  2744.0  3343.0       -  ID=gene_2   \n",
       "3      chr1     JGI   CDS  3344.0  3817.0       -  ID=gene_3   \n",
       "4      chr1     JGI   CDS  5330.0  6385.0       -  ID=gene_4   \n",
       "\n",
       "                  ID  \n",
       "0                 ID  \n",
       "1  proteinId=2114025  \n",
       "2  proteinId=2293935  \n",
       "3  proteinId=2293936  \n",
       "4  proteinId=2051335  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#gff = pd.read_csv(gff_path, sep='\\t', header=None)[[0,1,2,3,4,6,7,8]]\n",
    "thermo.columns = ['scaffold', 'source', 'type', 'start', 'stop', 'strand','info', 'ID']\n",
    "#gff['ID'] = gff['info'].str.split(pat=';', expand=True)\n",
    "#gff['species'] = gff['ID'].str[:2]\n",
    "\n",
    "thermo.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Add promoter annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "promoter_size = 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "can only concatenate str (not \"int\") to str",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[14], line 15\u001b[0m\n\u001b[1;32m     13\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m     14\u001b[0m     promoter_row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstart\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m promoter_row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstop\u001b[39m\u001b[38;5;124m'\u001b[39m]\n\u001b[0;32m---> 15\u001b[0m     promoter_row[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstop\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mpromoter_row\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mstop\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mpromoter_size\u001b[49m\n\u001b[1;32m     17\u001b[0m promoter_gff \u001b[38;5;241m=\u001b[39m promoter_gff\u001b[38;5;241m.\u001b[39mappend(promoter_row)\n\u001b[1;32m     19\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m i\u001b[38;5;241m>\u001b[39m\u001b[38;5;241m100\u001b[39m:\n",
      "\u001b[0;31mTypeError\u001b[0m: can only concatenate str (not \"int\") to str"
     ]
    }
   ],
   "source": [
    "promoter_gff = pd.DataFrame(columns=thermo.columns)\n",
    "\n",
    "for i, row in thermo.iterrows():\n",
    "    \n",
    "    promoter_row = row.copy()\n",
    "    promoter_row['type'] = 'promoter'\n",
    "    promoter_row['ID'] = row['ID']+'_promoter'\n",
    "    \n",
    "    if promoter_row['strand'] == '+':\n",
    "        promoter_row['start'] = promoter_row['start'] - promoter_size\n",
    "        promoter_row['stop'] = promoter_row['start']\n",
    "        \n",
    "    else:\n",
    "        promoter_row['start'] = promoter_row['stop']\n",
    "        promoter_row['stop'] = promoter_row['stop'] + promoter_size\n",
    "    \n",
    "    promoter_gff = promoter_gff.append(promoter_row)\n",
    "    \n",
    "    if i>100:\n",
    "        break\n",
    "    \n",
    "combined_gff = combined_gff.append(promoter_gff).sort_values(['scaffold', 'start'])\n",
    "combined_gff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Unable to parse string \"start\" at position 0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/_libs/lib.pyx:2315\u001b[0m, in \u001b[0;36mpandas._libs.lib.maybe_convert_numeric\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Unable to parse string \"start\"",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[15], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m promoter_gff \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame(columns\u001b[38;5;241m=\u001b[39mthermo\u001b[38;5;241m.\u001b[39mcolumns)\n\u001b[1;32m      3\u001b[0m \u001b[38;5;66;03m# Ensure numeric types for arithmetic operations\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m thermo[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstart\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mto_numeric\u001b[49m\u001b[43m(\u001b[49m\u001b[43mthermo\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mstart\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      5\u001b[0m thermo[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstop\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mto_numeric(thermo[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mstop\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m      7\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m i, row \u001b[38;5;129;01min\u001b[39;00m thermo\u001b[38;5;241m.\u001b[39miterrows():\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/core/tools/numeric.py:184\u001b[0m, in \u001b[0;36mto_numeric\u001b[0;34m(arg, errors, downcast)\u001b[0m\n\u001b[1;32m    182\u001b[0m coerce_numeric \u001b[38;5;241m=\u001b[39m errors \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m (\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mignore\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mraise\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    183\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 184\u001b[0m     values, _ \u001b[38;5;241m=\u001b[39m \u001b[43mlib\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmaybe_convert_numeric\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    185\u001b[0m \u001b[43m        \u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mset\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcoerce_numeric\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcoerce_numeric\u001b[49m\n\u001b[1;32m    186\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    187\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mValueError\u001b[39;00m, \u001b[38;5;167;01mTypeError\u001b[39;00m):\n\u001b[1;32m    188\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m errors \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mraise\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.9/site-packages/pandas/_libs/lib.pyx:2357\u001b[0m, in \u001b[0;36mpandas._libs.lib.maybe_convert_numeric\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Unable to parse string \"start\" at position 0"
     ]
    }
   ],
   "source": [
    "promoter_gff = pd.DataFrame(columns=thermo.columns)\n",
    "\n",
    "# Ensure numeric types for arithmetic operations\n",
    "thermo['start'] = pd.to_numeric(thermo['start'])\n",
    "thermo['stop'] = pd.to_numeric(thermo['stop'])\n",
    "\n",
    "for i, row in thermo.iterrows():\n",
    "    promoter_row = row.copy()\n",
    "    promoter_row['type'] = 'promoter'\n",
    "    promoter_row['ID'] = row['ID'] + '_promoter'\n",
    "    \n",
    "    if promoter_row['strand'] == '+':\n",
    "        promoter_row['start'] = promoter_row['start'] - promoter_size\n",
    "        promoter_row['stop'] = promoter_row['start']\n",
    "    else:\n",
    "        promoter_row['start'] = promoter_row['stop']\n",
    "        promoter_row['stop'] = promoter_row['stop'] + promoter_size\n",
    "    \n",
    "    promoter_gff = promoter_gff.append(promoter_row, ignore_index=True)\n",
    "    \n",
    "    if i > 100:\n",
    "        break\n",
    "\n",
    "combined_gff = combined_gff.append(promoter_gff, ignore_index=True).sort_values(['scaffold', 'start'])\n",
    "combined_gff.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_gff['scaffold'] = combined_gff['scaffold'].apply(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "combined_gff.to_csv('Klac_Kmar_CDS_and_promoters_orthologs_only.gff', sep='\\t', index=None)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
