{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import json\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as smm\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "\n",
    "from Bio import Seq\n",
    "\n",
    "pd.options.mode.chained_assignment = None\n",
    "\n",
    "#makes viewing pandas tables better\n",
    "pd.set_option('display.max_colwidth', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Change the following parameters to your output directory (where you want output files to be written), your metafile, and TNseq output. \n",
    "\n",
    "If these files aren't in the same directory as this notebook, you need use the full path. For example, the full path of 'barseq_output' would be '/usr2/people/clairedubin/barseq/barseq_output'\n",
    "\n",
    "Also change the control and experimental temperatures if needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#where the BarSeq FASTQs are\n",
    "FASTQ_directory = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## Barseq metafile\n",
    "\n",
    "You'll need to create your own metafile - easiest way is edit in Excel. A sample metafile is located at: /usr2/people/clairedubin/barseq/Kluyv_BarSeq_metadata.txt\n",
    "\n",
    "    Fastq: path to barseq FASTQ for each competition\n",
    "    SampleName: unique name for each competition, MUST include ctrl_temp or exp_temp as specified above\n",
    "    DualIndex: index from sequencing (in read name lines of FASTQ and the sequencing info file that Adam sends)\n",
    "    \n",
    "    UsePrecounted: True if you've already run the Barseq counting script on a competition, False otherwise\n",
    "    \n",
    "    poolfile: path to Tnseq output file, should be the same for each competition\n",
    "    output_dir: path to output directory\n",
    "    minRandom: number of bases between start of read and DualIndex, usually in sequence of 1,2,3,4,1,2,3,4 for each sample\n",
    "    maxRandom: same as minRandom\n",
    "    \n",
    "    BeforeBarcode: sequence preceding all barcodes\n",
    "    AfterBarcode: sequence following all barcodes\n",
    "    BarcodeLengths: lengths of barcodes to search for, with expected length of barcode first\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/37A1_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/37A2_S2_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/37B1_S3_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/37B2_S4_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/50A1_S5_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/50A2_S6_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/50B1_S7_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/50B2_S8_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB001_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB002_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB005_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB006_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB011_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB012_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB013_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB014_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB010_S1_L001_R1_001.fastq\r\n",
      "/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB011_S1_L001_R1_001.fastq\r\n"
     ]
    }
   ],
   "source": [
    "!ls {FASTQ_directory}/*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#format: sample_name, fastq_path, dual_index, offset\n",
    "#add or delete more lines as needed\n",
    "\n",
    "sample_info = [('4k_37_A1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB010_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('4k_37_A2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB011_S1_L001_R1_001.fastq', 'CGATGT', '2'), \n",
    "               ('4k_37_B1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB001_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('4k_37_B2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB002_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('4k_37_C1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB011_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('4k_37_C2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB012_S1_L001_R1_001.fastq', 'CGATGT', '2'), \n",
    "               ('4k_50_A1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB005_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('4k_50_A2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB006_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('4k_50_B1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB013_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('4k_50_B2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB014_S1_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('13k_37_A1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq', 'ATCACG', '1'),\n",
    "               ('13k_37_A2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A2_S2_L001_R1_001.fastq', 'TTAGGC', '3'),\n",
    "               ('13k_37_B1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B1_S3_L001_R1_001.fastq', 'CGATGT', '2'),\n",
    "               ('13k_37_B2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B2_S4_L001_R1_001.fastq', 'TGACCA', '4'),\n",
    "               ('13k_50_A1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A1_S5_L001_R1_001.fastq', 'ACAGTG', '1'),\n",
    "               ('13k_50_A2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq', 'CAGATC', '3'),\n",
    "               ('13k_50_B1', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B1_S7_L001_R1_001.fastq', 'GCCAAT', '2'),\n",
    "               ('13k_50_B2', '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B2_S8_L001_R1_001.fastq', 'ACTTGA', '4'),\n",
    "                ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "@M07716:180:000000000-DMYDD:1:1101:15574:1331 1:N:0:ATCACG\r\n",
      "AGCACTAGTCGACCTGCAGCGTACGGGTGGTGAATACAGGTCATCAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "CDDDDFFFFFDDGGGGGGGGGGGGGGGGGGGGHHHHHHHHHHHHHHHHHHHGHHGHHGGHHHHHHHHHHGGGGHHHHHHHGHHHGHHHHHHHHHHHHHHHHHGFHFHHHGGHHHHGGGHGHHHHHHHHHHHHGGGGF-<-\r\n",
      "@M07716:180:000000000-DMYDD:1:1101:15691:1331 1:N:0:ATCACG\r\n",
      "TGCACTAGTCGACCTGCAGCGTACGTCATCATGGACCCGTGTAATAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n",
      "+\r\n",
      "DCDDDFFFFFCDGGGGGGGGGGGGGHF225DA53BA2EE2AAAFGHHHHHHGHHGHHGGHHHHHHHHHHGGGGHHHHHHHGFF2?@F3B3F3@BBB334B3?/>/?BF0?/33?F<?<BF3?B3?F20FGHHGGGGG<<<\r\n",
      "@M07716:180:000000000-DMYDD:1:1101:15669:1331 1:N:0:ATCACG\r\n",
      "AGCACTAGTCGACCTGCAGCGTACGACGGTGTTCCAATTGTGGGCAGAGACCTCGTGGACATCAGATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAA\r\n"
     ]
    }
   ],
   "source": [
    "!head /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB010_S1_L001_R1_001.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To convert the annotated file to csv and not to be separated by tab (execute the next three lines only if necessary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "poolfile1 = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated.csv'\n",
    "df=pd.read_csv(poolfile1,sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv',index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#these variables should be the same for each replicate\n",
    "\n",
    "metafile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/BarSeq_metafile_13k_mutants_37•C_vs_50•C_12_19_2024.txt'\n",
    "logfile_name = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/BarSeq_metafile_13k_mutants_37•C_vs_50•C_12_19_2024.log'\n",
    "poolfile = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv'\n",
    "output_dir = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter'\n",
    "before_barcode = 'GTCGACCTGCAGCGTACG' #bottom strand sequence was used\n",
    "#before_barcode = 'GATGTCCACGAGGTCTCT' #what was there before \n",
    "after_barcode =  'AGAGACCTCGTGGACATC' #bottom strand sequence was used\n",
    "#after_barcode = 'CGTACGCTGCAGGTCGAC'  #what was there before \n",
    "barcode_lengths = '20,19,18,21'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#if there are any samples where you already counted barcodes, add the sample names here\n",
    "precounted_samples = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "metafile_columns = ['FileIndex','SampleName', 'Fastq', 'DualIndex', 'Poolfile', 'OutputDir', 'minRandom', 'maxRandom',\n",
    "                   'BeforeBarcode', 'AfterBarcode', 'BarcodeLengths', 'UsePrecounted']\n",
    "\n",
    "with open(metafile_name, 'w') as f:\n",
    "    \n",
    "    #write column names\n",
    "    f.write('\\t'.join(metafile_columns)+'\\n')\n",
    "    \n",
    "    #write a line for each pool\n",
    "    for sample_name, fastq, dual_index, offset in sample_info:\n",
    "        \n",
    "        #DUAL INDECES ARE BACKWARDS FROM WHAT THE FLOWCELL OUTPUT SAYS\n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index[::-1], poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'False'])\n",
    "        f.write(to_write+'\\n')\n",
    "        \n",
    "    for precounted_sample_name in precounted_samples:\n",
    "        \n",
    "        to_write = '\\t'.join([sample_name, sample_name, fastq, dual_index, poolfile, output_dir, offset, offset,\n",
    "                             before_barcode, after_barcode, barcode_lengths, 'True'])\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check metafile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileIndex</th>\n",
       "      <th>SampleName</th>\n",
       "      <th>Fastq</th>\n",
       "      <th>DualIndex</th>\n",
       "      <th>Poolfile</th>\n",
       "      <th>OutputDir</th>\n",
       "      <th>minRandom</th>\n",
       "      <th>maxRandom</th>\n",
       "      <th>BeforeBarcode</th>\n",
       "      <th>AfterBarcode</th>\n",
       "      <th>BarcodeLengths</th>\n",
       "      <th>UsePrecounted</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4k_37_A1</td>\n",
       "      <td>4k_37_A1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB010_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4k_37_A2</td>\n",
       "      <td>4k_37_A2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB011_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4k_37_B1</td>\n",
       "      <td>4k_37_B1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB001_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4k_37_B2</td>\n",
       "      <td>4k_37_B2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB002_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4k_37_C1</td>\n",
       "      <td>4k_37_C1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB011_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4k_37_C2</td>\n",
       "      <td>4k_37_C2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB012_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>4k_50_A1</td>\n",
       "      <td>4k_50_A1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB005_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4k_50_A2</td>\n",
       "      <td>4k_50_A2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB006_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4k_50_B1</td>\n",
       "      <td>4k_50_B1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB013_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4k_50_B2</td>\n",
       "      <td>4k_50_B2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB014_S1_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>13k_37_A1</td>\n",
       "      <td>13k_37_A1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq</td>\n",
       "      <td>GCACTA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>13k_37_A2</td>\n",
       "      <td>13k_37_A2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A2_S2_L001_R1_001.fastq</td>\n",
       "      <td>CGGATT</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13k_37_B1</td>\n",
       "      <td>13k_37_B1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B1_S3_L001_R1_001.fastq</td>\n",
       "      <td>TGTAGC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13k_37_B2</td>\n",
       "      <td>13k_37_B2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B2_S4_L001_R1_001.fastq</td>\n",
       "      <td>ACCAGT</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>13k_50_A1</td>\n",
       "      <td>13k_50_A1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A1_S5_L001_R1_001.fastq</td>\n",
       "      <td>GTGACA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>13k_50_A2</td>\n",
       "      <td>13k_50_A2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq</td>\n",
       "      <td>CTAGAC</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>13k_50_B1</td>\n",
       "      <td>13k_50_B1</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B1_S7_L001_R1_001.fastq</td>\n",
       "      <td>TAACCG</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>13k_50_B2</td>\n",
       "      <td>13k_50_B2</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B2_S8_L001_R1_001.fastq</td>\n",
       "      <td>AGTTCA</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv</td>\n",
       "      <td>/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>GTCGACCTGCAGCGTACG</td>\n",
       "      <td>AGAGACCTCGTGGACATC</td>\n",
       "      <td>20,19,18,21</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    FileIndex SampleName  \\\n",
       "0   4k_37_A1   4k_37_A1    \n",
       "1   4k_37_A2   4k_37_A2    \n",
       "2   4k_37_B1   4k_37_B1    \n",
       "3   4k_37_B2   4k_37_B2    \n",
       "4   4k_37_C1   4k_37_C1    \n",
       "5   4k_37_C2   4k_37_C2    \n",
       "6   4k_50_A1   4k_50_A1    \n",
       "7   4k_50_A2   4k_50_A2    \n",
       "8   4k_50_B1   4k_50_B1    \n",
       "9   4k_50_B2   4k_50_B2    \n",
       "10  13k_37_A1  13k_37_A1   \n",
       "11  13k_37_A2  13k_37_A2   \n",
       "12  13k_37_B1  13k_37_B1   \n",
       "13  13k_37_B2  13k_37_B2   \n",
       "14  13k_50_A1  13k_50_A1   \n",
       "15  13k_50_A2  13k_50_A2   \n",
       "16  13k_50_B1  13k_50_B1   \n",
       "17  13k_50_B2  13k_50_B2   \n",
       "\n",
       "                                                                                                                                                                                             Fastq  \\\n",
       "0   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB010_S1_L001_R1_001.fastq                      \n",
       "1   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB011_S1_L001_R1_001.fastq                      \n",
       "2   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB001_S1_L001_R1_001.fastq                      \n",
       "3   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB002_S1_L001_R1_001.fastq                      \n",
       "4   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB011_S1_L001_R1_001.fastq                      \n",
       "5   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB012_S1_L001_R1_001.fastq                      \n",
       "6   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB005_S1_L001_R1_001.fastq                      \n",
       "7   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB006_S1_L001_R1_001.fastq                      \n",
       "8   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB013_S1_L001_R1_001.fastq                      \n",
       "9   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB014_S1_L001_R1_001.fastq                      \n",
       "10  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq   \n",
       "11  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A2_S2_L001_R1_001.fastq   \n",
       "12  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B1_S3_L001_R1_001.fastq   \n",
       "13  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B2_S4_L001_R1_001.fastq   \n",
       "14  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A1_S5_L001_R1_001.fastq   \n",
       "15  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq   \n",
       "16  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B1_S7_L001_R1_001.fastq   \n",
       "17  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B2_S8_L001_R1_001.fastq   \n",
       "\n",
       "   DualIndex  \\\n",
       "0   GCACTA     \n",
       "1   TGTAGC     \n",
       "2   GCACTA     \n",
       "3   TGTAGC     \n",
       "4   GCACTA     \n",
       "5   TGTAGC     \n",
       "6   GCACTA     \n",
       "7   TGTAGC     \n",
       "8   GCACTA     \n",
       "9   TGTAGC     \n",
       "10  GCACTA     \n",
       "11  CGGATT     \n",
       "12  TGTAGC     \n",
       "13  ACCAGT     \n",
       "14  GTGACA     \n",
       "15  CTAGAC     \n",
       "16  TAACCG     \n",
       "17  AGTTCA     \n",
       "\n",
       "                                                                                                                                                                                                                                                            Poolfile  \\\n",
       "0   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "1   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "2   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "3   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "4   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "5   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "6   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "7   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "8   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "9   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "10  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "11  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "12  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "13  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "14  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "15  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "16  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "17  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv   \n",
       "\n",
       "                                                                                                                                                                                                                                                         OutputDir  \\\n",
       "0   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "1   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "2   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "3   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "4   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "5   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "6   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "7   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "8   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "9   /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "10  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "11  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "12  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "13  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "14  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "15  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "16  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "17  /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter   \n",
       "\n",
       "    minRandom  maxRandom       BeforeBarcode        AfterBarcode  \\\n",
       "0   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "1   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "2   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "3   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "4   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "5   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "6   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "7   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "8   1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "9   2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "10  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "11  3          3          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "12  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "13  4          4          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "14  1          1          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "15  3          3          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "16  2          2          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "17  4          4          GTCGACCTGCAGCGTACG  AGAGACCTCGTGGACATC   \n",
       "\n",
       "   BarcodeLengths  UsePrecounted  \n",
       "0   20,19,18,21    False          \n",
       "1   20,19,18,21    False          \n",
       "2   20,19,18,21    False          \n",
       "3   20,19,18,21    False          \n",
       "4   20,19,18,21    False          \n",
       "5   20,19,18,21    False          \n",
       "6   20,19,18,21    False          \n",
       "7   20,19,18,21    False          \n",
       "8   20,19,18,21    False          \n",
       "9   20,19,18,21    False          \n",
       "10  20,19,18,21    False          \n",
       "11  20,19,18,21    False          \n",
       "12  20,19,18,21    False          \n",
       "13  20,19,18,21    False          \n",
       "14  20,19,18,21    False          \n",
       "15  20,19,18,21    False          \n",
       "16  20,19,18,21    False          \n",
       "17  20,19,18,21    False          "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metafile_name, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Barseq counting\n",
    "\n",
    "Might take a few hours!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-12-19 14:09:07 RBseq_Count_BarCodes.py\n",
      "2024-12-19 14:09:07 Version: 1.1.4\n",
      "2024-12-19 14:09:07 Release Date: July 16, 2020\n",
      "2024-12-19 14:09:07 Options passed:  metafile:/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/BarSeq_metafile_13k_mutants_37•C_vs_50•C_12_19_2024.txt  logFile:/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/BarSeq_metafile_13k_mutants_37•C_vs_50•C_12_19_2024.log  minQual:10  matchBefore:6  matchAfter:6  quietMode:False \n",
      "2024-12-19 14:09:07 Logging status updates in /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/BarSeq_metafile_13k_mutants_37•C_vs_50•C_12_19_2024.log\n",
      "2024-12-19 14:09:07 Loading TnSeq library metadata from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/BarSeq_metafile_13k_mutants_37•C_vs_50•C_12_19_2024.txt\n",
      "2024-12-19 14:09:07 Setting output directory as: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter (from first line of metadata file)\n",
      "2024-12-19 14:09:07 Loading mapped barcodes in mutant pool from: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv (from first line of metadata file)\n",
      "2024-12-19 14:09:07 Read 17154 barcodes from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_mapping_output_4k_13k_mutants_all_fastqs_available/pool_4k_plus_13k_combined_poolfile_with_promoter_reannotated_1.csv\n",
      "2024-12-19 14:09:07 Finding barcodes in fastqs and counting occurances\n",
      "2024-12-19 14:09:07 ---------------------\n",
      "2024-12-19 14:09:07   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB010_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:09:07   Filtering reads on second index: GCACTA\n",
      "2024-12-19 14:09:07   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:09:19     1278676 reads processed.\n",
      "2024-12-19 14:09:19     1215383 reads with compliant barcodes.\n",
      "2024-12-19 14:09:19     63293 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:09:19       31868 reads without expected Dual Index.\n",
      "2024-12-19 14:09:19       26427 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:09:19       4998 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:09:19       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:09:19       0 reads with quality scores less than 10\n",
      "2024-12-19 14:09:19   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_37_A1.counts\n",
      "2024-12-19 14:09:20   Total barcodes seen (incudes sequencing errors): 72071\n",
      "2024-12-19 14:09:20   Matching barcodes to poolfile\n",
      "2024-12-19 14:09:20     Number of barcodes from poolfile seen: 2599\n",
      "2024-12-19 14:09:20     Reads with barcodes from poolfile: 387602\n",
      "2024-12-19 14:09:20   Most abundant barcode: TATATGTGTTGCAATGTTAC seen 119465 times.\n",
      "2024-12-19 14:09:20   Number of reads that differ from this barcode by one base pair (likely sequencing errors):4189\n",
      "2024-12-19 14:09:20   Estimated sequencing error rate for barcodes: 3.4%\n",
      "2024-12-19 14:09:20   Barcodes seen once (highly inflated by sequencing errors): 59288\n",
      "2024-12-19 14:09:20   Barcodes seen twice (slightly inflated by sequencing errors): 4462\n",
      "2024-12-19 14:09:20   Barcodes seen three times or more: 8321\n",
      "2024-12-19 14:09:20   Chao estimate of population size (ones^2/2*twos): 390000\n",
      "2024-12-19 14:09:20 ---------------------\n",
      "2024-12-19 14:09:20   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/OORB011_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:09:20   Filtering reads on second index: TGTAGC\n",
      "2024-12-19 14:09:20   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:09:32     1372401 reads processed.\n",
      "2024-12-19 14:09:32     1327397 reads with compliant barcodes.\n",
      "2024-12-19 14:09:32     45004 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:09:32       18701 reads without expected Dual Index.\n",
      "2024-12-19 14:09:32       19788 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:09:32       6515 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:09:32       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:09:32       0 reads with quality scores less than 10\n",
      "2024-12-19 14:09:32   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_37_A2.counts\n",
      "2024-12-19 14:09:32   Total barcodes seen (incudes sequencing errors): 57337\n",
      "2024-12-19 14:09:32   Matching barcodes to poolfile\n",
      "2024-12-19 14:09:32     Number of barcodes from poolfile seen: 2691\n",
      "2024-12-19 14:09:32     Reads with barcodes from poolfile: 437031\n",
      "2024-12-19 14:09:32   Most abundant barcode: TATATGTGTTGCAATGTTAC seen 124651 times.\n",
      "2024-12-19 14:09:32   Number of reads that differ from this barcode by one base pair (likely sequencing errors):3462\n",
      "2024-12-19 14:09:32   Estimated sequencing error rate for barcodes: 2.7%\n",
      "2024-12-19 14:09:32   Barcodes seen once (highly inflated by sequencing errors): 45404\n",
      "2024-12-19 14:09:32   Barcodes seen twice (slightly inflated by sequencing errors): 3765\n",
      "2024-12-19 14:09:32   Barcodes seen three times or more: 8168\n",
      "2024-12-19 14:09:32   Chao estimate of population size (ones^2/2*twos): 270000\n",
      "2024-12-19 14:09:32 ---------------------\n",
      "2024-12-19 14:09:32   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB001_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:09:32   Filtering reads on second index: GCACTA\n",
      "2024-12-19 14:09:32   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:09:42     1192746 reads processed.\n",
      "2024-12-19 14:09:42     1138095 reads with compliant barcodes.\n",
      "2024-12-19 14:09:42     54651 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:09:42       26286 reads without expected Dual Index.\n",
      "2024-12-19 14:09:42       22098 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:09:42       6267 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:09:42       2 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:09:42       2 reads with quality scores less than 10\n",
      "2024-12-19 14:09:42   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_37_B1.counts\n",
      "2024-12-19 14:09:42   Total barcodes seen (incudes sequencing errors): 95859\n",
      "2024-12-19 14:09:42   Matching barcodes to poolfile\n",
      "2024-12-19 14:09:42     Number of barcodes from poolfile seen: 2747\n",
      "2024-12-19 14:09:42     Reads with barcodes from poolfile: 463064\n",
      "2024-12-19 14:09:42   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 110463 times.\n",
      "2024-12-19 14:09:42   Number of reads that differ from this barcode by one base pair (likely sequencing errors):6921\n",
      "2024-12-19 14:09:42   Estimated sequencing error rate for barcodes: 5.9%\n",
      "2024-12-19 14:09:42   Barcodes seen once (highly inflated by sequencing errors): 80059\n",
      "2024-12-19 14:09:42   Barcodes seen twice (slightly inflated by sequencing errors): 5671\n",
      "2024-12-19 14:09:42   Barcodes seen three times or more: 10129\n",
      "2024-12-19 14:09:42   Chao estimate of population size (ones^2/2*twos): 570000\n",
      "2024-12-19 14:09:42 ---------------------\n",
      "2024-12-19 14:09:42   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB002_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:09:42   Filtering reads on second index: TGTAGC\n",
      "2024-12-19 14:09:42   Looking for sequence barcode preceeded by CGTACG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-12-19 14:09:52     1279289 reads processed.\n",
      "2024-12-19 14:09:52     1228497 reads with compliant barcodes.\n",
      "2024-12-19 14:09:52     50792 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:09:52       20511 reads without expected Dual Index.\n",
      "2024-12-19 14:09:52       23903 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:09:52       6378 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:09:52       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:09:52       0 reads with quality scores less than 10\n",
      "2024-12-19 14:09:52   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_37_B2.counts\n",
      "2024-12-19 14:09:52   Total barcodes seen (incudes sequencing errors): 57482\n",
      "2024-12-19 14:09:52   Matching barcodes to poolfile\n",
      "2024-12-19 14:09:52     Number of barcodes from poolfile seen: 2767\n",
      "2024-12-19 14:09:52     Reads with barcodes from poolfile: 504291\n",
      "2024-12-19 14:09:52   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 108898 times.\n",
      "2024-12-19 14:09:52   Number of reads that differ from this barcode by one base pair (likely sequencing errors):5922\n",
      "2024-12-19 14:09:52   Estimated sequencing error rate for barcodes: 5.2%\n",
      "2024-12-19 14:09:52   Barcodes seen once (highly inflated by sequencing errors): 43897\n",
      "2024-12-19 14:09:52   Barcodes seen twice (slightly inflated by sequencing errors): 4207\n",
      "2024-12-19 14:09:52   Barcodes seen three times or more: 9378\n",
      "2024-12-19 14:09:52   Chao estimate of population size (ones^2/2*twos): 230000\n",
      "2024-12-19 14:09:52 ---------------------\n",
      "2024-12-19 14:09:52   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB011_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:09:52   Filtering reads on second index: GCACTA\n",
      "2024-12-19 14:09:52   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:10:04     1696525 reads processed.\n",
      "2024-12-19 14:10:04     1629777 reads with compliant barcodes.\n",
      "2024-12-19 14:10:04     66748 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:10:04       30353 reads without expected Dual Index.\n",
      "2024-12-19 14:10:04       26641 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:10:04       9754 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:10:04       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:10:04       0 reads with quality scores less than 10\n",
      "2024-12-19 14:10:04   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_37_C1.counts\n",
      "2024-12-19 14:10:04   Total barcodes seen (incudes sequencing errors): 187859\n",
      "2024-12-19 14:10:04   Matching barcodes to poolfile\n",
      "2024-12-19 14:10:04     Number of barcodes from poolfile seen: 3406\n",
      "2024-12-19 14:10:04     Reads with barcodes from poolfile: 731526\n",
      "2024-12-19 14:10:04   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 265386 times.\n",
      "2024-12-19 14:10:04   Number of reads that differ from this barcode by one base pair (likely sequencing errors):17610\n",
      "2024-12-19 14:10:04   Estimated sequencing error rate for barcodes: 6.2%\n",
      "2024-12-19 14:10:04   Barcodes seen once (highly inflated by sequencing errors): 164589\n",
      "2024-12-19 14:10:04   Barcodes seen twice (slightly inflated by sequencing errors): 9478\n",
      "2024-12-19 14:10:04   Barcodes seen three times or more: 13792\n",
      "2024-12-19 14:10:04   Chao estimate of population size (ones^2/2*twos): 1400000\n",
      "2024-12-19 14:10:04 ---------------------\n",
      "2024-12-19 14:10:05   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB012_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:10:05   Filtering reads on second index: TGTAGC\n",
      "2024-12-19 14:10:05   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:10:13     1242382 reads processed.\n",
      "2024-12-19 14:10:13     1198966 reads with compliant barcodes.\n",
      "2024-12-19 14:10:13     43416 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:10:13       17099 reads without expected Dual Index.\n",
      "2024-12-19 14:10:13       19833 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:10:13       6484 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:10:13       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:10:13       0 reads with quality scores less than 10\n",
      "2024-12-19 14:10:13   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_37_C2.counts\n",
      "2024-12-19 14:10:13   Total barcodes seen (incudes sequencing errors): 68864\n",
      "2024-12-19 14:10:13   Matching barcodes to poolfile\n",
      "2024-12-19 14:10:14     Number of barcodes from poolfile seen: 2797\n",
      "2024-12-19 14:10:14     Reads with barcodes from poolfile: 593552\n",
      "2024-12-19 14:10:14   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 206274 times.\n",
      "2024-12-19 14:10:14   Number of reads that differ from this barcode by one base pair (likely sequencing errors):9533\n",
      "2024-12-19 14:10:14   Estimated sequencing error rate for barcodes: 4.4%\n",
      "2024-12-19 14:10:14   Barcodes seen once (highly inflated by sequencing errors): 55267\n",
      "2024-12-19 14:10:14   Barcodes seen twice (slightly inflated by sequencing errors): 4672\n",
      "2024-12-19 14:10:14   Barcodes seen three times or more: 8925\n",
      "2024-12-19 14:10:14   Chao estimate of population size (ones^2/2*twos): 330000\n",
      "2024-12-19 14:10:14 ---------------------\n",
      "2024-12-19 14:10:14   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB005_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:10:14   Filtering reads on second index: GCACTA\n",
      "2024-12-19 14:10:14   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:10:22     1142319 reads processed.\n",
      "2024-12-19 14:10:22     1099239 reads with compliant barcodes.\n",
      "2024-12-19 14:10:22     43080 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:10:22       20284 reads without expected Dual Index.\n",
      "2024-12-19 14:10:22       17467 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:10:22       5329 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:10:22       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:10:22       0 reads with quality scores less than 10\n",
      "2024-12-19 14:10:22   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_50_A1.counts\n",
      "2024-12-19 14:10:22   Total barcodes seen (incudes sequencing errors): 88860\n",
      "2024-12-19 14:10:22   Matching barcodes to poolfile\n",
      "2024-12-19 14:10:22     Number of barcodes from poolfile seen: 3004\n",
      "2024-12-19 14:10:22     Reads with barcodes from poolfile: 486672\n",
      "2024-12-19 14:10:22   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 93462 times.\n",
      "2024-12-19 14:10:22   Number of reads that differ from this barcode by one base pair (likely sequencing errors):3958\n",
      "2024-12-19 14:10:22   Estimated sequencing error rate for barcodes: 4.1%\n",
      "2024-12-19 14:10:22   Barcodes seen once (highly inflated by sequencing errors): 74210\n",
      "2024-12-19 14:10:22   Barcodes seen twice (slightly inflated by sequencing errors): 4908\n",
      "2024-12-19 14:10:22   Barcodes seen three times or more: 9742\n",
      "2024-12-19 14:10:22   Chao estimate of population size (ones^2/2*twos): 560000\n",
      "2024-12-19 14:10:22 ---------------------\n",
      "2024-12-19 14:10:22   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB006_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:10:22   Filtering reads on second index: TGTAGC\n",
      "2024-12-19 14:10:22   Looking for sequence barcode preceeded by CGTACG\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-12-19 14:10:30     1115874 reads processed.\n",
      "2024-12-19 14:10:30     1078435 reads with compliant barcodes.\n",
      "2024-12-19 14:10:30     37439 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:10:30       16098 reads without expected Dual Index.\n",
      "2024-12-19 14:10:30       16523 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:10:30       4818 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:10:30       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:10:30       0 reads with quality scores less than 10\n",
      "2024-12-19 14:10:30   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_50_A2.counts\n",
      "2024-12-19 14:10:30   Total barcodes seen (incudes sequencing errors): 57745\n",
      "2024-12-19 14:10:30   Matching barcodes to poolfile\n",
      "2024-12-19 14:10:30     Number of barcodes from poolfile seen: 2986\n",
      "2024-12-19 14:10:30     Reads with barcodes from poolfile: 501208\n",
      "2024-12-19 14:10:30   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 104635 times.\n",
      "2024-12-19 14:10:30   Number of reads that differ from this barcode by one base pair (likely sequencing errors):4001\n",
      "2024-12-19 14:10:30   Estimated sequencing error rate for barcodes: 3.7%\n",
      "2024-12-19 14:10:30   Barcodes seen once (highly inflated by sequencing errors): 44881\n",
      "2024-12-19 14:10:30   Barcodes seen twice (slightly inflated by sequencing errors): 3884\n",
      "2024-12-19 14:10:30   Barcodes seen three times or more: 8980\n",
      "2024-12-19 14:10:30   Chao estimate of population size (ones^2/2*twos): 260000\n",
      "2024-12-19 14:10:30 ---------------------\n",
      "2024-12-19 14:10:30   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB013_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:10:30   Filtering reads on second index: GCACTA\n",
      "2024-12-19 14:10:30   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:10:37     942141 reads processed.\n",
      "2024-12-19 14:10:37     901944 reads with compliant barcodes.\n",
      "2024-12-19 14:10:37     40197 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:10:37       19322 reads without expected Dual Index.\n",
      "2024-12-19 14:10:37       15817 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:10:37       5058 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:10:37       56 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:10:37       57 reads with quality scores less than 10\n",
      "2024-12-19 14:10:37   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_50_B1.counts\n",
      "2024-12-19 14:10:37   Total barcodes seen (incudes sequencing errors): 62622\n",
      "2024-12-19 14:10:37   Matching barcodes to poolfile\n",
      "2024-12-19 14:10:37     Number of barcodes from poolfile seen: 2949\n",
      "2024-12-19 14:10:37     Reads with barcodes from poolfile: 472888\n",
      "2024-12-19 14:10:37   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 147705 times.\n",
      "2024-12-19 14:10:37   Number of reads that differ from this barcode by one base pair (likely sequencing errors):6676\n",
      "2024-12-19 14:10:37   Estimated sequencing error rate for barcodes: 4.3%\n",
      "2024-12-19 14:10:37   Barcodes seen once (highly inflated by sequencing errors): 49692\n",
      "2024-12-19 14:10:37   Barcodes seen twice (slightly inflated by sequencing errors): 4397\n",
      "2024-12-19 14:10:37   Barcodes seen three times or more: 8533\n",
      "2024-12-19 14:10:37   Chao estimate of population size (ones^2/2*twos): 280000\n",
      "2024-12-19 14:10:37 ---------------------\n",
      "2024-12-19 14:10:37   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/BarSeq_FASTQs_4k_plus_13k_mutants_11_04_2024/KFRB014_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:10:37   Filtering reads on second index: TGTAGC\n",
      "2024-12-19 14:10:37   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:10:42     774089 reads processed.\n",
      "2024-12-19 14:10:42     749116 reads with compliant barcodes.\n",
      "2024-12-19 14:10:42     24973 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:10:42       11048 reads without expected Dual Index.\n",
      "2024-12-19 14:10:42       11281 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:10:42       2644 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:10:42       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:10:42       0 reads with quality scores less than 10\n",
      "2024-12-19 14:10:42   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/4k_50_B2.counts\n",
      "2024-12-19 14:10:42   Total barcodes seen (incudes sequencing errors): 32077\n",
      "2024-12-19 14:10:42   Matching barcodes to poolfile\n",
      "2024-12-19 14:10:42     Number of barcodes from poolfile seen: 2950\n",
      "2024-12-19 14:10:42     Reads with barcodes from poolfile: 422989\n",
      "2024-12-19 14:10:42   Most abundant barcode: ACGGTGTTCCAAGTGTGGGC seen 136151 times.\n",
      "2024-12-19 14:10:42   Number of reads that differ from this barcode by one base pair (likely sequencing errors):4961\n",
      "2024-12-19 14:10:42   Estimated sequencing error rate for barcodes: 3.5%\n",
      "2024-12-19 14:10:42   Barcodes seen once (highly inflated by sequencing errors): 22599\n",
      "2024-12-19 14:10:42   Barcodes seen twice (slightly inflated by sequencing errors): 2517\n",
      "2024-12-19 14:10:42   Barcodes seen three times or more: 6961\n",
      "2024-12-19 14:10:42   Chao estimate of population size (ones^2/2*twos): 100000\n",
      "2024-12-19 14:10:42 ---------------------\n",
      "2024-12-19 14:10:42   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq\n",
      "2024-12-19 14:10:42   Filtering reads on second index: GCACTA\n",
      "2024-12-19 14:10:42   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:16:10     40263692 reads processed.\n",
      "2024-12-19 14:16:10     38359575 reads with compliant barcodes.\n",
      "2024-12-19 14:16:10     1904117 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:16:10       1209030 reads without expected Dual Index.\n",
      "2024-12-19 14:16:10       575556 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:16:10       119531 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:16:10       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:16:10       0 reads with quality scores less than 10\n",
      "2024-12-19 14:16:10   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_37_A1.counts\n",
      "2024-12-19 14:16:11   Total barcodes seen (incudes sequencing errors): 708741\n",
      "2024-12-19 14:16:11   Matching barcodes to poolfile\n",
      "2024-12-19 14:16:12     Number of barcodes from poolfile seen: 11079\n",
      "2024-12-19 14:16:12     Reads with barcodes from poolfile: 17716519\n",
      "2024-12-19 14:16:12   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 6063821 times.\n",
      "2024-12-19 14:16:12   Number of reads that differ from this barcode by one base pair (likely sequencing errors):92012\n",
      "2024-12-19 14:16:12   Estimated sequencing error rate for barcodes: 1.5%\n",
      "2024-12-19 14:16:12   Barcodes seen once (highly inflated by sequencing errors): 558436\n",
      "2024-12-19 14:16:12   Barcodes seen twice (slightly inflated by sequencing errors): 47736\n",
      "2024-12-19 14:16:12   Barcodes seen three times or more: 102569\n",
      "2024-12-19 14:16:12   Chao estimate of population size (ones^2/2*twos): 3300000\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-12-19 14:16:12 ---------------------\n",
      "2024-12-19 14:16:12   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A2_S2_L001_R1_001.fastq\n",
      "2024-12-19 14:16:12   Filtering reads on second index: CGGATT\n",
      "2024-12-19 14:16:12   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:21:21     39219999 reads processed.\n",
      "2024-12-19 14:21:21     36060633 reads with compliant barcodes.\n",
      "2024-12-19 14:21:21     3159366 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:21:21       2394780 reads without expected Dual Index.\n",
      "2024-12-19 14:21:21       569716 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:21:21       194870 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:21:21       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:21:21       0 reads with quality scores less than 10\n",
      "2024-12-19 14:21:21   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_37_A2.counts\n",
      "2024-12-19 14:21:24   Total barcodes seen (incudes sequencing errors): 509441\n",
      "2024-12-19 14:21:24   Matching barcodes to poolfile\n",
      "2024-12-19 14:21:24     Number of barcodes from poolfile seen: 10667\n",
      "2024-12-19 14:21:24     Reads with barcodes from poolfile: 17376268\n",
      "2024-12-19 14:21:24   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 7099893 times.\n",
      "2024-12-19 14:21:24   Number of reads that differ from this barcode by one base pair (likely sequencing errors):101191\n",
      "2024-12-19 14:21:24   Estimated sequencing error rate for barcodes: 1.4%\n",
      "2024-12-19 14:21:24   Barcodes seen once (highly inflated by sequencing errors): 383234\n",
      "2024-12-19 14:21:24   Barcodes seen twice (slightly inflated by sequencing errors): 38492\n",
      "2024-12-19 14:21:24   Barcodes seen three times or more: 87715\n",
      "2024-12-19 14:21:24   Chao estimate of population size (ones^2/2*twos): 1900000\n",
      "2024-12-19 14:21:24 ---------------------\n",
      "2024-12-19 14:21:24   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B1_S3_L001_R1_001.fastq\n",
      "2024-12-19 14:21:24   Filtering reads on second index: TGTAGC\n",
      "2024-12-19 14:21:24   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:28:38     57187067 reads processed.\n",
      "2024-12-19 14:28:38     54545091 reads with compliant barcodes.\n",
      "2024-12-19 14:28:38     2641976 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:28:38       1564299 reads without expected Dual Index.\n",
      "2024-12-19 14:28:38       821968 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:28:38       255709 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:28:38       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:28:38       0 reads with quality scores less than 10\n",
      "2024-12-19 14:28:38   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_37_B1.counts\n",
      "2024-12-19 14:28:40   Total barcodes seen (incudes sequencing errors): 1127688\n",
      "2024-12-19 14:28:40   Matching barcodes to poolfile\n",
      "2024-12-19 14:28:41     Number of barcodes from poolfile seen: 11151\n",
      "2024-12-19 14:28:41     Reads with barcodes from poolfile: 24553332\n",
      "2024-12-19 14:28:41   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 8145103 times.\n",
      "2024-12-19 14:28:41   Number of reads that differ from this barcode by one base pair (likely sequencing errors):179699\n",
      "2024-12-19 14:28:41   Estimated sequencing error rate for barcodes: 2.2%\n",
      "2024-12-19 14:28:41   Barcodes seen once (highly inflated by sequencing errors): 905418\n",
      "2024-12-19 14:28:41   Barcodes seen twice (slightly inflated by sequencing errors): 75619\n",
      "2024-12-19 14:28:41   Barcodes seen three times or more: 146651\n",
      "2024-12-19 14:28:41   Chao estimate of population size (ones^2/2*twos): 5400000\n",
      "2024-12-19 14:28:41 ---------------------\n",
      "2024-12-19 14:28:41   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B2_S4_L001_R1_001.fastq\n",
      "2024-12-19 14:28:41   Filtering reads on second index: ACCAGT\n",
      "2024-12-19 14:28:41   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:33:37     42029828 reads processed.\n",
      "2024-12-19 14:33:37     39507992 reads with compliant barcodes.\n",
      "2024-12-19 14:33:37     2521836 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:33:37       1397496 reads without expected Dual Index.\n",
      "2024-12-19 14:33:37       915194 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:33:37       209146 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:33:37       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:33:37       0 reads with quality scores less than 10\n",
      "2024-12-19 14:33:37   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_37_B2.counts\n",
      "2024-12-19 14:33:39   Total barcodes seen (incudes sequencing errors): 522282\n",
      "2024-12-19 14:33:39   Matching barcodes to poolfile\n",
      "2024-12-19 14:33:40     Number of barcodes from poolfile seen: 10877\n",
      "2024-12-19 14:33:40     Reads with barcodes from poolfile: 18732930\n",
      "2024-12-19 14:33:40   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 7048011 times.\n",
      "2024-12-19 14:33:40   Number of reads that differ from this barcode by one base pair (likely sequencing errors):82397\n",
      "2024-12-19 14:33:40   Estimated sequencing error rate for barcodes: 1.2%\n",
      "2024-12-19 14:33:40   Barcodes seen once (highly inflated by sequencing errors): 393213\n",
      "2024-12-19 14:33:40   Barcodes seen twice (slightly inflated by sequencing errors): 37933\n",
      "2024-12-19 14:33:40   Barcodes seen three times or more: 91136\n",
      "2024-12-19 14:33:40   Chao estimate of population size (ones^2/2*twos): 2000000\n",
      "2024-12-19 14:33:40 ---------------------\n",
      "2024-12-19 14:33:40   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A1_S5_L001_R1_001.fastq\n",
      "2024-12-19 14:33:40   Filtering reads on second index: GTGACA\n",
      "2024-12-19 14:33:40   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:38:23     39675304 reads processed.\n",
      "2024-12-19 14:38:23     37446323 reads with compliant barcodes.\n",
      "2024-12-19 14:38:23     2228981 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:38:23       1261894 reads without expected Dual Index.\n",
      "2024-12-19 14:38:23       783014 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:38:23       184073 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:38:23       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:38:23       0 reads with quality scores less than 10\n",
      "2024-12-19 14:38:23   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_50_A1.counts\n",
      "2024-12-19 14:38:23   Total barcodes seen (incudes sequencing errors): 671325\n",
      "2024-12-19 14:38:23   Matching barcodes to poolfile\n",
      "2024-12-19 14:38:24     Number of barcodes from poolfile seen: 11004\n",
      "2024-12-19 14:38:24     Reads with barcodes from poolfile: 13523027\n",
      "2024-12-19 14:38:24   Most abundant barcode: GTACCAGATAAACTCTAAAA seen 1950930 times.\n",
      "2024-12-19 14:38:24   Number of reads that differ from this barcode by one base pair (likely sequencing errors):24254\n",
      "2024-12-19 14:38:24   Estimated sequencing error rate for barcodes: 1.2%\n",
      "2024-12-19 14:38:24   Barcodes seen once (highly inflated by sequencing errors): 522237\n",
      "2024-12-19 14:38:24   Barcodes seen twice (slightly inflated by sequencing errors): 45652\n",
      "2024-12-19 14:38:24   Barcodes seen three times or more: 103436\n",
      "2024-12-19 14:38:24   Chao estimate of population size (ones^2/2*twos): 3000000\n",
      "2024-12-19 14:38:24 ---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-12-19 14:38:24   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq\n",
      "2024-12-19 14:38:24   Filtering reads on second index: CTAGAC\n",
      "2024-12-19 14:38:24   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:42:53     38141465 reads processed.\n",
      "2024-12-19 14:42:53     35591327 reads with compliant barcodes.\n",
      "2024-12-19 14:42:53     2550138 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:42:53       1637925 reads without expected Dual Index.\n",
      "2024-12-19 14:42:53       710264 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:42:53       201949 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:42:53       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:42:53       0 reads with quality scores less than 10\n",
      "2024-12-19 14:42:53   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_50_A2.counts\n",
      "2024-12-19 14:42:55   Total barcodes seen (incudes sequencing errors): 612189\n",
      "2024-12-19 14:42:55   Matching barcodes to poolfile\n",
      "2024-12-19 14:42:56     Number of barcodes from poolfile seen: 10963\n",
      "2024-12-19 14:42:56     Reads with barcodes from poolfile: 13289732\n",
      "2024-12-19 14:42:56   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 2310076 times.\n",
      "2024-12-19 14:42:56   Number of reads that differ from this barcode by one base pair (likely sequencing errors):41320\n",
      "2024-12-19 14:42:56   Estimated sequencing error rate for barcodes: 1.8%\n",
      "2024-12-19 14:42:56   Barcodes seen once (highly inflated by sequencing errors): 457592\n",
      "2024-12-19 14:42:56   Barcodes seen twice (slightly inflated by sequencing errors): 46714\n",
      "2024-12-19 14:42:56   Barcodes seen three times or more: 107883\n",
      "2024-12-19 14:42:56   Chao estimate of population size (ones^2/2*twos): 2200000\n",
      "2024-12-19 14:42:56 ---------------------\n",
      "2024-12-19 14:42:56   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B1_S7_L001_R1_001.fastq\n",
      "2024-12-19 14:42:56   Filtering reads on second index: TAACCG\n",
      "2024-12-19 14:42:56   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:50:29     47561233 reads processed.\n",
      "2024-12-19 14:50:29     44656442 reads with compliant barcodes.\n",
      "2024-12-19 14:50:29     2904791 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:50:29       1909672 reads without expected Dual Index.\n",
      "2024-12-19 14:50:29       772333 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:50:29       222786 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:50:29       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:50:29       0 reads with quality scores less than 10\n",
      "2024-12-19 14:50:29   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_50_B1.counts\n",
      "2024-12-19 14:50:30   Total barcodes seen (incudes sequencing errors): 775386\n",
      "2024-12-19 14:50:30   Matching barcodes to poolfile\n",
      "2024-12-19 14:50:31     Number of barcodes from poolfile seen: 11044\n",
      "2024-12-19 14:50:31     Reads with barcodes from poolfile: 16399351\n",
      "2024-12-19 14:50:31   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 2484714 times.\n",
      "2024-12-19 14:50:31   Number of reads that differ from this barcode by one base pair (likely sequencing errors):33131\n",
      "2024-12-19 14:50:31   Estimated sequencing error rate for barcodes: 1.3%\n",
      "2024-12-19 14:50:31   Barcodes seen once (highly inflated by sequencing errors): 611116\n",
      "2024-12-19 14:50:31   Barcodes seen twice (slightly inflated by sequencing errors): 52536\n",
      "2024-12-19 14:50:31   Barcodes seen three times or more: 111734\n",
      "2024-12-19 14:50:31   Chao estimate of population size (ones^2/2*twos): 3600000\n",
      "2024-12-19 14:50:31 ---------------------\n",
      "2024-12-19 14:50:31   Mapping reads from /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B2_S8_L001_R1_001.fastq\n",
      "2024-12-19 14:50:31   Filtering reads on second index: AGTTCA\n",
      "2024-12-19 14:50:31   Looking for sequence barcode preceeded by CGTACG\n",
      "2024-12-19 14:55:50     38221312 reads processed.\n",
      "2024-12-19 14:55:50     36020255 reads with compliant barcodes.\n",
      "2024-12-19 14:55:50     2201057 reads without recognizable, compliant barcodes. Of those:\n",
      "2024-12-19 14:55:50       1211704 reads without expected Dual Index.\n",
      "2024-12-19 14:55:50       810064 reads without expected sequence before the barcode region.\n",
      "2024-12-19 14:55:50       179289 reads without expeced sequence after the barcode region.\n",
      "2024-12-19 14:55:50       0 reads with noncompliant barcdes. (Contains Ns, etc).\n",
      "2024-12-19 14:55:50       0 reads with quality scores less than 10\n",
      "2024-12-19 14:55:50   Saving barcode counts to /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/countsFiles/13k_50_B2.counts\n",
      "2024-12-19 14:55:51   Total barcodes seen (incudes sequencing errors): 552661\n",
      "2024-12-19 14:55:51   Matching barcodes to poolfile\n",
      "2024-12-19 14:55:52     Number of barcodes from poolfile seen: 10889\n",
      "2024-12-19 14:55:52     Reads with barcodes from poolfile: 13546927\n",
      "2024-12-19 14:55:52   Most abundant barcode: CGTAGACCATAGTGGACCAA seen 2351497 times.\n",
      "2024-12-19 14:55:52   Number of reads that differ from this barcode by one base pair (likely sequencing errors):30210\n",
      "2024-12-19 14:55:52   Estimated sequencing error rate for barcodes: 1.3%\n",
      "2024-12-19 14:55:52   Barcodes seen once (highly inflated by sequencing errors): 414974\n",
      "2024-12-19 14:55:52   Barcodes seen twice (slightly inflated by sequencing errors): 40572\n",
      "2024-12-19 14:55:52   Barcodes seen three times or more: 97115\n",
      "2024-12-19 14:55:52   Chao estimate of population size (ones^2/2*twos): 2100000\n",
      "2024-12-19 14:55:52 ---------------------\n",
      "2024-12-19 14:55:52 Saving summary statistics for fastqs to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/fastqSummaryStats.txt\n",
      "2024-12-19 14:55:52 Processed 18 sequence files corresponding to 18 biological samples.\n",
      "2024-12-19 14:55:52 Counted 140138909 total reads for 15081 mapped barcodes from the mutant pool.\n",
      "2024-12-19 14:55:52 Saving poolcount file to: /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/Barseq_recounted_with_promoter/poolCount.txt\n"
     ]
    }
   ],
   "source": [
    "!python3 /usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/RBseq_Count_BarCodes_v2.py -m {metafile_name} -l {logfile_name}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### To change the Barseq pool count file from txt to csv for making plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the data from the TSV file\n",
    "tsv_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_12_19_2024/Barseq_recounted_with_promoter/poolCount.txt'\n",
    "df = pd.read_csv(tsv_file_path, sep='\\t')\n",
    "\n",
    "# Save the data to a CSV file\n",
    "csv_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_12_19_2024/Barseq_recounted_with_promoter/poolCount.csv'\n",
    "df.to_csv(csv_file_path, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Making Scatter Plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount.txt', sep='\\t')  # Update with your file path\n",
    "\n",
    "# Define the pairs of columns for scatter plots\n",
    "column_pairs = [\n",
    "    ('4k_37_A1', '4k_37_A2'),\n",
    "    ('4k_37_B1', '4k_37_B2'),\n",
    "    ('4k_37_C1', '4k_37_C2'),\n",
    "    ('4k_50_A1', '4k_50_A2'),\n",
    "    ('4k_50_B1', '4k_50_B2'),\n",
    "    ('13k_37_A1', '13k_37_B2'),\n",
    "    ('13k_37_B1', '13k_37_B2'),\n",
    "    ('13k_50_A1', '13k_50_B2'),\n",
    "    ('13k_50_B1', '13k_50_B2')\n",
    "]\n",
    "\n",
    "# Plotting each pair of columns\n",
    "for x_column, y_column in column_pairs:\n",
    "    plt.figure(figsize=(8, 6))\n",
    "    plt.scatter(df[x_column], df[y_column], color='black')\n",
    "    plt.xlabel(f'{x_column} Abundance', fontsize=11)\n",
    "    plt.ylabel(f'{y_column} Abundance', fontsize=11)\n",
    "    plt.title(f'Scatter Plot: {x_column} vs {y_column}', fontsize=13, color='purple')\n",
    "    plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ajusting the axis of the scatter plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount.txt', sep='\\t')  # Update with your file path\n",
    "\n",
    "# Define the pairs of columns for scatter plots\n",
    "column_pairs = [\n",
    "    ('4k_37_A1', '4k_37_A2'),\n",
    "    ('4k_37_B1', '4k_37_B2'),\n",
    "    ('4k_37_C1', '4k_37_C2'),\n",
    "    ('4k_50_A1', '4k_50_A2'),\n",
    "    ('4k_50_B1', '4k_50_B2'),\n",
    "    ('13k_37_A1', '13k_37_A2'),\n",
    "    ('13k_37_B1', '13k_37_B2'),\n",
    "    ('13k_50_A1', '13k_50_A2'),\n",
    "    ('13k_50_B1', '13k_50_B2')\n",
    "]\n",
    "\n",
    "\n",
    "# Set custom axis limits for each pair (adjust as needed)\n",
    "axis_limits = {\n",
    "    '4k_37_A1_4k_37_A2': {'xlim': (0, 20000), 'ylim': (0, 20000)},\n",
    "    '4k_37_B1_4k_37_B2': {'xlim': (0, 10000), 'ylim': (0, 10000)},\n",
    "    '4k_37_C1_4k_37_C2': {'xlim': (0, 10000), 'ylim': (0, 10000)},\n",
    "    '4k_50_A1_4k_50_A2': {'xlim': (0, 6000), 'ylim': (0, 6000)},\n",
    "    '4k_50_B1_4k_50_B2': {'xlim': (0, 6000), 'ylim': (0, 6000)},\n",
    "    '13k_37_A1_13k_37_A2': {'xlim': (0, 250000), 'ylim': (0, 250000)},\n",
    "    '13k_37_B1_13k_37_B2': {'xlim': (0, 250000), 'ylim': (0, 250000)},\n",
    "    '13k_50_A1_13k_50_A2': {'xlim': (0, 250000), 'ylim': (0, 250000)},\n",
    "    '13k_50_B1_13k_50_B2': {'xlim': (0, 250000), 'ylim': (0, 250000)},\n",
    "}\n",
    "\n",
    "# Plotting each pair of columns with custom axis limits\n",
    "for x_column, y_column in column_pairs:\n",
    "    key = f'{x_column}_{y_column}'\n",
    "    plt.figure(figsize=(8, 6))\n",
    "    plt.scatter(df[x_column], df[y_column], color='black')\n",
    "    plt.xlabel(f'{x_column} Abundance', fontsize=11)\n",
    "    plt.ylabel(f'{y_column} Abundance', fontsize=11)\n",
    "    plt.title(f'Scatter Plot: {x_column} vs {y_column}', fontsize=13, color='purple')\n",
    "    \n",
    "    # Apply axis limits if available\n",
    "    if key in axis_limits:\n",
    "        plt.xlim(axis_limits[key]['xlim'])\n",
    "        plt.ylim(axis_limits[key]['ylim'])\n",
    "\n",
    "    plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## to normalize the Barseq pool count file based on the number of reads obtained"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### this is the normalization that worked with the right result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the count file into a DataFrame\n",
    "count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount.txt'\n",
    "df = pd.read_csv(count_file_path, sep='\\t')\n",
    "\n",
    "# Calculate the total counts for each replicate\n",
    "df['Total_Count_4k_37_A1'] = df['4k_37_A1'].sum()\n",
    "df['Total_Count_4k_37_A2'] = df['4k_37_A2'].sum()\n",
    "df['Total_Count_4k_37_B1'] = df['4k_37_B1'].sum()\n",
    "df['Total_Count_4k_37_B2'] = df['4k_37_B2'].sum()\n",
    "df['Total_Count_4k_37_C1'] = df['4k_37_C1'].sum()\n",
    "df['Total_Count_4k_37_C2'] = df['4k_37_C2'].sum()\n",
    "df['Total_Count_4k_50_A1'] = df['4k_50_A1'].sum()\n",
    "df['Total_Count_4k_50_A2'] = df['4k_50_A2'].sum()\n",
    "df['Total_Count_4k_50_B1'] = df['4k_50_B1'].sum()\n",
    "df['Total_Count_4k_50_B2'] = df['4k_50_B2'].sum()\n",
    "df['Total_Count_13k_37_A1'] = df['13k_37_A1'].sum()\n",
    "df['Total_Count_13k_37_A2'] = df['13k_37_A2'].sum()\n",
    "df['Total_Count_13k_37_B1'] = df['13k_37_B1'].sum()\n",
    "df['Total_Count_13k_37_B2'] = df['13k_37_B2'].sum()\n",
    "df['Total_Count_13k_50_A1'] = df['13k_50_A1'].sum()\n",
    "df['Total_Count_13k_50_A2'] = df['13k_50_A2'].sum()\n",
    "df['Total_Count_13k_50_B1'] = df['13k_50_B1'].sum()\n",
    "df['Total_Count_13k_50_B2'] = df['13k_50_B2'].sum()\n",
    "\n",
    "\n",
    "\n",
    "# Normalize counts for each replicate by dividing by its own Total_Count\n",
    "df['normalized_4k_37_A1'] = df['4k_37_A1'] / df['Total_Count_4k_37_A1']\n",
    "df['normalized_4k_37_A2'] = df['4k_37_A2'] / df['Total_Count_4k_37_A2']\n",
    "df['normalized_4k_37_B1'] = df['4k_37_B1'] / df['Total_Count_4k_37_B1']\n",
    "df['normalized_4k_37_B2'] = df['4k_37_B2'] / df['Total_Count_4k_37_B2']\n",
    "df['normalized_4k_37_C1'] = df['4k_37_C1'] / df['Total_Count_4k_37_C1']\n",
    "df['normalized_4k_37_C2'] = df['4k_37_C2'] / df['Total_Count_4k_37_C2']\n",
    "df['normalized_4k_50_A1'] = df['4k_50_A1'] / df['Total_Count_4k_50_A1']\n",
    "df['normalized_4k_50_A2'] = df['4k_50_A2'] / df['Total_Count_4k_50_A2']\n",
    "df['normalized_4k_50_B1'] = df['4k_50_B1'] / df['Total_Count_4k_50_B1']\n",
    "df['normalized_4k_50_B2'] = df['4k_50_B2'] / df['Total_Count_4k_50_B2']\n",
    "df['normalized_13k_37_A1'] = df['13k_37_A1'] / df['Total_Count_13k_37_A1']\n",
    "df['normalized_13k_37_A2'] = df['13k_37_A2'] / df['Total_Count_13k_37_A2']\n",
    "df['normalized_13k_37_B1'] = df['13k_37_B1'] / df['Total_Count_13k_37_B1']\n",
    "df['normalized_13k_37_B2'] = df['13k_37_B2'] / df['Total_Count_13k_37_B2']\n",
    "df['normalized_13k_50_A1'] = df['13k_50_A1'] / df['Total_Count_13k_50_A1']\n",
    "df['normalized_13k_50_A2'] = df['13k_50_A2'] / df['Total_Count_13k_50_A2']\n",
    "df['normalized_13k_50_B1'] = df['13k_50_B1'] / df['Total_Count_13k_50_B1']\n",
    "df['normalized_13k_50_B2'] = df['13k_50_B2'] / df['Total_Count_13k_50_B2']\n",
    "\n",
    "# Save the normalized count file\n",
    "normalized_count_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount_4k_plus_13k_mutants_37•C_vs_50•C.csv'\n",
    "df.to_csv(normalized_count_file_path, index=False)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To make the fitness abundance agaian using the normalized values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Load the CSV file\n",
    "csv_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount_4k_plus_13k_mutants_37•C_vs_50•C.csv'\n",
    "df = pd.read_csv(csv_file_path)\n",
    "\n",
    "# Save the DataFrame to a text file (txt)\n",
    "txt_file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount_4k_plus_13k_mutants_37•C_vs_50•C.txt'\n",
    "df.to_csv(txt_file_path, sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/poolCount_37•C_vs_50•C.txt', sep='\\t')\n",
    "\n",
    "# Define the pairs of columns for scatter plots\n",
    "column_pairs = [\n",
    "    ('normalized_37_A1', 'normalized_37_A2'),\n",
    "    ('normalized_37_B1', 'normalized_37_B2'),\n",
    "    ('normalized_50_A1', 'normalized_50_A2'),\n",
    "    ('normalized_50_B1', 'normalized_50_B2')\n",
    "]\n",
    "\n",
    "# Plotting each pair of columns\n",
    "for x_column, y_column in column_pairs:\n",
    "    plt.figure(figsize=(8, 6))\n",
    "    plt.scatter(df[x_column], df[y_column], color='black')\n",
    "    plt.xlabel(f'{x_column} Abundance', fontsize=11)\n",
    "    plt.ylabel(f'{y_column} Abundance', fontsize=11)\n",
    "    plt.title(f'Scatter Plot: {x_column} vs {y_column}', fontsize=13, color='purple')\n",
    "    plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Load the data from the CSV file into a DataFrame\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/poolCount_37•C_vs_50•C.txt', sep='\\t')\n",
    "\n",
    "# Define the pairs of columns and their axis limits for scatter plots\n",
    "column_pairs = [\n",
    "    ('normalized_37_A1', 'normalized_37_A2'),\n",
    "    ('normalized_37_B1', 'normalized_37_B2'),\n",
    "    ('normalized_50_A1', 'normalized_50_A2'),\n",
    "    ('normalized_50_B1', 'normalized_50_B2')\n",
    "]\n",
    "\n",
    "# Set custom axis limits for each pair (adjust as needed)\n",
    "axis_limits = {\n",
    "    'normalized_37_A1_normalized_37_A2': {'xlim': (0, 0.02), 'ylim': (0, 0.025)},\n",
    "    'normalized_37_B1_normalized_37_B2': {'xlim': (0, 0.02), 'ylim': (0, 0.025)},\n",
    "    'normalized_50_A1_normalized_50_A2': {'xlim': (0, 0.02), 'ylim': (0, 0.02)},\n",
    "    'normalized_50_B1_normalized_50_B2': {'xlim': (0, 0.02), 'ylim': (0, 0.02)}\n",
    "}\n",
    "\n",
    "# Plotting each pair of columns with custom axis limits\n",
    "for x_column, y_column in column_pairs:\n",
    "    key = f'{x_column}_{y_column}'\n",
    "    plt.figure(figsize=(8, 6))\n",
    "    plt.scatter(df[x_column], df[y_column], color='black')\n",
    "    plt.xlabel(f'{x_column} Abundance', fontsize=11)\n",
    "    plt.ylabel(f'{y_column} Abundance', fontsize=11)\n",
    "    plt.title(f'Scatter Plot: {x_column} vs {y_column}', fontsize=13, color='purple')\n",
    "    \n",
    "    # Apply axis limits if available\n",
    "    if key in axis_limits:\n",
    "        plt.xlim(axis_limits[key]['xlim'])\n",
    "        plt.ylim(axis_limits[key]['ylim'])\n",
    "\n",
    "    plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# To do statistical t-test"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### To handle rows where some values are zero and convert them to NaN before performing the Mann-Whitney U test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import mannwhitneyu\n",
    "\n",
    "# Load the dataset\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount_4k_plus_13k_mutants_37•C_vs_50•C.csv')\n",
    "\n",
    "# Convert all columns to numeric, coercing errors to NaN\n",
    "for col in df.columns:\n",
    "    df[col] = pd.to_numeric(df[col], errors='coerce')\n",
    "\n",
    "# Replace zeros with NaN\n",
    "df.replace(0, np.nan, inplace=True)\n",
    "\n",
    "# Define the data columns for the existing and new conditions\n",
    "data_37C = df[['normalized_4k_37_A1', 'normalized_4k_37_A2', 'normalized_4k_37_B1', 'normalized_4k_37_B2', 'normalized_4k_37_C1', 'normalized_4k_37_C2', 'normalized_13k_37_A1', 'normalized_13k_37_A2', 'normalized_13k_37_B1', 'normalized_13k_37_B2']]\n",
    "data_50C = df[['normalized_4k_50_A1', 'normalized_4k_50_A2', 'normalized_4k_50_B1', 'normalized_4k_50_B2', 'normalized_13k_50_A1', 'normalized_13k_50_A2', 'normalized_13k_50_B1', 'normalized_13k_50_B2']]\n",
    "            \n",
    "\n",
    "\n",
    "\n",
    "# List to store p-values for existing conditions\n",
    "p_values_37_vs_50 = []\n",
    "\n",
    "# Iterate over rows and perform Mann-Whitney U test for each\n",
    "for index, row in df.iterrows():\n",
    "    # Drop NaN values from the datasets\n",
    "    data_37C_values = row[data_37C.columns].dropna()\n",
    "    \n",
    "    \n",
    "\n",
    "    # Perform the Mann-Whitney U test for 37C vs. 50C\n",
    "    #data_37C_values = row[data_25C.columns].dropna()\n",
    "    data_50C_values = row[data_50C.columns].dropna()\n",
    "\n",
    "    if len(data_37C_values) > 0 and len(data_50C_values) > 0:\n",
    "        statistic, p_value = mannwhitneyu(data_37C_values, data_50C_values, alternative='two-sided')\n",
    "        p_values_37_vs_50.append(p_value)\n",
    "    else:\n",
    "        p_values_37_vs_50.append(np.nan)\n",
    "\n",
    "# Add the p-values to the DataFrame\n",
    "\n",
    "df['MannWhitneyU_p_value_37_vs_50'] = p_values_37_vs_50\n",
    "\n",
    "# Save the DataFrame with the calculated p-values to a new CSV file\n",
    "output_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_4k_plus_13k_mutants/for_analysis/pool_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_4k_plus_13k_mutants_37•C_vs_50•C_11_05_2024/poolCount_4k_plus_13k_mutants_37•C_vs_50•C_output_with_mannwhitneyu_p_values_zero_values_removed_combined.csv'\n",
    "df.to_csv(output_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from scipy.stats import mannwhitneyu\n",
    "\n",
    "# Load the dataset\n",
    "df = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/poolCount_37•C_vs_50•C.csv')\n",
    "\n",
    "# Convert all columns to numeric, coercing errors to NaN\n",
    "for col in df.columns:\n",
    "    df[col] = pd.to_numeric(df[col], errors='coerce')\n",
    "\n",
    "# Replace zeros with NaN\n",
    "df.replace(0, np.nan, inplace=True)\n",
    "\n",
    "# Define the data columns for the existing and new conditions\n",
    "data_37C = df[['37_A1', '37_A2', '37_B1', '37_B2']]\n",
    "data_50C = df[['50_A1', '50_A2', '50_B1', '50_B2']]\n",
    "            \n",
    "\n",
    "\n",
    "\n",
    "# List to store p-values for existing conditions\n",
    "p_values_37_vs_50 = []\n",
    "\n",
    "# Iterate over rows and perform Mann-Whitney U test for each\n",
    "for index, row in df.iterrows():\n",
    "    # Drop NaN values from the datasets\n",
    "    data_37C_values = row[data_37C.columns].dropna()\n",
    "    \n",
    "    \n",
    "\n",
    "    # Perform the Mann-Whitney U test for 37C vs. 50C\n",
    "    #data_37C_values = row[data_25C.columns].dropna()\n",
    "    data_50C_values = row[data_50C.columns].dropna()\n",
    "\n",
    "    if len(data_37C_values) > 0 and len(data_50C_values) > 0:\n",
    "        statistic, p_value = mannwhitneyu(data_37C_values, data_50C_values, alternative='two-sided')\n",
    "        p_values_37_vs_50.append(p_value)\n",
    "    else:\n",
    "        p_values_37_vs_50.append(np.nan)\n",
    "\n",
    "# Add the p-values to the DataFrame\n",
    "\n",
    "df['MannWhitneyU_p_value_37_vs_50'] = p_values_37_vs_50\n",
    "\n",
    "# Save the DataFrame with the calculated p-values to a new CSV file\n",
    "output_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/for_analysis/all_fastqs_available/TNSeq_barseq_outputs_using_all_fastqs_combined_13k_mutants_37•C_vs_50•C_10_28_2024/poolCount__raw_37•C_vs_50•C_output_with_mannwhitneyu_p_values_zero_values_removed_combined.csv'\n",
    "df.to_csv(output_path, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
