{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e4024a25",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 556852\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/KFRB019_S1_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "31b1f8fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 1331161\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/Tnseq_fastq_files_jumpstart_polymerase/TNSeq_merge_FASTQ/fastq_file_for_two_sequencing_batches_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "175649dd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 3848664\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/TnSeq_11k_diff_digestion_time/TNSeq_merge_FASTQ/all_digestion_time_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4ec346b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 5179825\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/Tnseq_fastq_files/all_available_fastq_merged/all_batches_together_06_12_2024_merged.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f55903b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "BARSEQ FILES WITH 500 MILLION reads generated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "02c55908",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 40263692\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A1_S1_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e41af355",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total number of reads: 39219999\n"
     ]
    }
   ],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37A2_S2_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc4dcb04",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B1_S3_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06c2a13c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/37B2_S4_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43ec5dac",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A1_S5_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38226832",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50A2_S6_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a06c8aa3",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B1_S7_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7719b250",
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_reads_in_fastq(file_path):\n",
    "    with open(file_path, 'r') as file:\n",
    "        count = 0\n",
    "        for line in file:\n",
    "            if line.startswith('@'):\n",
    "                count += 1\n",
    "    return count\n",
    "\n",
    "# Example usage:\n",
    "file_path = '/usr2/people/shollyt22/shollyt22/TnSeq_BarSeq_sequencings/Sequencing_with_the_13k_mutants/BarSeq_FASTQs_13k_mutants_03_27_2024/Barseq_37•C_vs_50•C_500_million_reads/50B2_S8_L001_R1_001.fastq'\n",
    "total_reads = count_reads_in_fastq(file_path)\n",
    "print(f'Total number of reads: {total_reads}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0ec529e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
