{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "901c1de0",
   "metadata": {},
   "source": [
    "### To read through the file and extract columns with gene only and write it into a new output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "dfe167c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "68be7876",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  0           1                2          3          4  5  6  \\\n",
      "0      chromosome_1  prediction             gene     1381.0     2142.0  0  +   \n",
      "1      chromosome_1  prediction             mRNA     1381.0     2142.0  .  +   \n",
      "2      chromosome_1  prediction             exon     1381.0     2142.0  .  +   \n",
      "3      chromosome_1  prediction              CDS     1381.0     2142.0  .  +   \n",
      "4      chromosome_1  prediction             gene     2744.0     3343.0  0  -   \n",
      "...             ...         ...              ...        ...        ... .. ..   \n",
      "83236  chromosome_7  prediction             exon  4103222.0  4103301.0  .  -   \n",
      "83237  chromosome_7  prediction              CDS  4103222.0  4103301.0  .  -   \n",
      "83238  chromosome_7  prediction             exon  4101637.0  4103168.0  .  -   \n",
      "83239  chromosome_7  prediction              CDS  4101762.0  4103168.0  .  -   \n",
      "83240  chromosome_7  prediction  three_prime_UTR  4101637.0  4101761.0  .  -   \n",
      "\n",
      "       7                                                  8  \n",
      "0      .  ID=gene_1;Name=jgi.p|Spoth2|2114025;portal_id=...  \n",
      "1      .  ID=mRNA_1;Name=jgi.p|Spoth2|2114025;Parent=gen...  \n",
      "2      .                          ID=exon_1_1;Parent=mRNA_1  \n",
      "3      0                             ID=CDS_1;Parent=mRNA_1  \n",
      "4      .  ID=gene_2;Name=jgi.p|Spoth2|2293935;portal_id=...  \n",
      "...   ..                                                ...  \n",
      "83236  .                  ID=exon_10841_2;Parent=mRNA_10841  \n",
      "83237  2                     ID=CDS_10841;Parent=mRNA_10841  \n",
      "83238  .                  ID=exon_10841_3;Parent=mRNA_10841  \n",
      "83239  0                     ID=CDS_10841;Parent=mRNA_10841  \n",
      "83240  .                    ID=UTR3_10841;Parent=mRNA_10841  \n",
      "\n",
      "[83241 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "# Read the TnSeq file\n",
    "data_frame = pd.read_csv('/usr2/people/shollyt22/shollyt22/TnSeq_Results_may_2023/Myceliopthora_Whole_Genome_sequence/download.20230512.134223/MycthetATCC42464_2/Mycocosm/Annotation/Filtered_Models___best__/Proteins/Spoth2.filtered_proteins.FilteredModels2.gff3_mod', sep='\\t', header=None)\n",
    "print(data_frame)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "55ee14de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  0           1     2          3          4  5  6  7  \\\n",
      "0      chromosome_1  prediction  gene     1381.0     2142.0  0  +  .   \n",
      "4      chromosome_1  prediction  gene     2744.0     3343.0  0  -  .   \n",
      "10     chromosome_1  prediction  gene     3344.0     3817.0  0  -  .   \n",
      "16     chromosome_1  prediction  gene     5330.0     6385.0  0  -  .   \n",
      "20     chromosome_1  prediction  gene    15458.0    25533.0  0  +  .   \n",
      "...             ...         ...   ...        ...        ... .. .. ..   \n",
      "83193  chromosome_7  prediction  gene  4089438.0  4091337.0  0  +  .   \n",
      "83207  chromosome_7  prediction  gene  4095568.0  4097284.0  0  +  .   \n",
      "83213  chromosome_7  prediction  gene  4098122.0  4098907.0  0  -  .   \n",
      "83219  chromosome_7  prediction  gene  4100606.0  4101293.0  0  -  .   \n",
      "83231  chromosome_7  prediction  gene  4101637.0  4103479.0  0  -  .   \n",
      "\n",
      "                                                       8  \n",
      "0      ID=gene_1;Name=jgi.p|Spoth2|2114025;portal_id=...  \n",
      "4      ID=gene_2;Name=jgi.p|Spoth2|2293935;portal_id=...  \n",
      "10     ID=gene_3;Name=jgi.p|Spoth2|2293936;portal_id=...  \n",
      "16     ID=gene_4;Name=jgi.p|Spoth2|2051335;portal_id=...  \n",
      "20     ID=gene_5;Name=jgi.p|Spoth2|2121898;portal_id=...  \n",
      "...                                                  ...  \n",
      "83193  ID=gene_10836;Name=jgi.p|Spoth2|103797;portal_...  \n",
      "83207  ID=gene_10837;Name=jgi.p|Spoth2|2071098;portal...  \n",
      "83213  ID=gene_10838;Name=jgi.p|Spoth2|2114022;portal...  \n",
      "83219  ID=gene_10840;Name=jgi.p|Spoth2|2114023;portal...  \n",
      "83231  ID=gene_10841;Name=jgi.p|Spoth2|2313233;portal...  \n",
      "\n",
      "[9110 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "# Find the rows that contain the desired word in a specific column\n",
    "condition = data_frame[2].str.contains('gene', case=False).fillna(False)\n",
    "# Extract the corresponding columns for the rows that match the condition\n",
    "extracted_data = data_frame.loc[condition, [0, 1, 2, 3, 4, 5, 6, 7, 8]]  # Replace column_index1 and column_index2 with the indices of the desired columns\n",
    "extracted_data.to_csv('extracted_gff_files_from_protein_file.txt', sep='\\t', index=False, header=None)  # Replace 'sorted_mapped_reads.txt' with the desired filename\n",
    "print(extracted_data)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b16db009",
   "metadata": {},
   "source": [
    "### to change the source column from prediction to JGI . same was also done to replace prediction to JGI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "a5c3654b",
   "metadata": {},
   "outputs": [],
   "source": [
    "data_frame1 = pd.read_csv('/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/extracted_gff_files_from_protein_file.txt', sep='\\t', header=None)\n",
    "#print(data_frame1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "6cf29262",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 0    1     2          3          4  5  6  7  \\\n",
      "0     chromosome_1  JGI  gene     1381.0     2142.0  0  +  .   \n",
      "1     chromosome_1  JGI  gene     2744.0     3343.0  0  -  .   \n",
      "2     chromosome_1  JGI  gene     3344.0     3817.0  0  -  .   \n",
      "3     chromosome_1  JGI  gene     5330.0     6385.0  0  -  .   \n",
      "4     chromosome_1  JGI  gene    15458.0    25533.0  0  +  .   \n",
      "...            ...  ...   ...        ...        ... .. .. ..   \n",
      "9105  chromosome_7  JGI  gene  4089438.0  4091337.0  0  +  .   \n",
      "9106  chromosome_7  JGI  gene  4095568.0  4097284.0  0  +  .   \n",
      "9107  chromosome_7  JGI  gene  4098122.0  4098907.0  0  -  .   \n",
      "9108  chromosome_7  JGI  gene  4100606.0  4101293.0  0  -  .   \n",
      "9109  chromosome_7  JGI  gene  4101637.0  4103479.0  0  -  .   \n",
      "\n",
      "                                                      8  \n",
      "0     ID=gene_1;Name=jgi.p|Spoth2|2114025;portal_id=...  \n",
      "1     ID=gene_2;Name=jgi.p|Spoth2|2293935;portal_id=...  \n",
      "2     ID=gene_3;Name=jgi.p|Spoth2|2293936;portal_id=...  \n",
      "3     ID=gene_4;Name=jgi.p|Spoth2|2051335;portal_id=...  \n",
      "4     ID=gene_5;Name=jgi.p|Spoth2|2121898;portal_id=...  \n",
      "...                                                 ...  \n",
      "9105  ID=gene_10836;Name=jgi.p|Spoth2|103797;portal_...  \n",
      "9106  ID=gene_10837;Name=jgi.p|Spoth2|2071098;portal...  \n",
      "9107  ID=gene_10838;Name=jgi.p|Spoth2|2114022;portal...  \n",
      "9108  ID=gene_10840;Name=jgi.p|Spoth2|2114023;portal...  \n",
      "9109  ID=gene_10841;Name=jgi.p|Spoth2|2313233;portal...  \n",
      "\n",
      "[9110 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "# Update all occurrences of 'prediction' to 'JGI' in column 2 using apply and lambda function\n",
    "data_frame1[1] = data_frame1[1].apply(lambda x: 'JGI' if 'prediction' in str(x).lower() else x)\n",
    "# Save the DataFrame to a CSV file. The default delimiter in CSV file is (,) it can be changed \n",
    "data_frame1.to_csv('updated_extracted_gff_files_from_protein_file.csv', sep='\\t', index=False, header=False)\n",
    "print(data_frame1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4839ba82",
   "metadata": {},
   "source": [
    "### To split the content of the last column in order to retain only the protein ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "8526114e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 0           1     2          3          4  5  6  7  \\\n",
      "0     chromosome_1  prediction  gene     1381.0     2142.0  0  +  .   \n",
      "1     chromosome_1  prediction  gene     2744.0     3343.0  0  -  .   \n",
      "2     chromosome_1  prediction  gene     3344.0     3817.0  0  -  .   \n",
      "3     chromosome_1  prediction  gene     5330.0     6385.0  0  -  .   \n",
      "4     chromosome_1  prediction  gene    15458.0    25533.0  0  +  .   \n",
      "...            ...         ...   ...        ...        ... .. .. ..   \n",
      "9105  chromosome_7  prediction  gene  4089438.0  4091337.0  0  +  .   \n",
      "9106  chromosome_7  prediction  gene  4095568.0  4097284.0  0  +  .   \n",
      "9107  chromosome_7  prediction  gene  4098122.0  4098907.0  0  -  .   \n",
      "9108  chromosome_7  prediction  gene  4100606.0  4101293.0  0  -  .   \n",
      "9109  chromosome_7  prediction  gene  4101637.0  4103479.0  0  -  .   \n",
      "\n",
      "          Retained_Part  \n",
      "0     proteinId=2114025  \n",
      "1     proteinId=2293935  \n",
      "2     proteinId=2293936  \n",
      "3     proteinId=2051335  \n",
      "4     proteinId=2121898  \n",
      "...                 ...  \n",
      "9105   proteinId=103797  \n",
      "9106  proteinId=2071098  \n",
      "9107  proteinId=2114022  \n",
      "9108  proteinId=2114023  \n",
      "9109  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 9 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "def split_cell_content(input_file, column_index, delimiter, portion_index):\n",
    "    # Read the input table into a pandas DataFrame without header names\n",
    "    df = pd.read_csv(input_file, sep='\\t', header=None)\n",
    "\n",
    "    # Convert the column with float64 values to strings\n",
    "    df[column_index] = df[column_index].astype(str)\n",
    "\n",
    "    # Split the content of the cell using the delimiter\n",
    "    df[column_index] = df[column_index].str.split(delimiter)\n",
    "\n",
    "    # Extract the portion to retain from the split content using indexing and create a new column\n",
    "    df['Retained_Part'] = df[column_index].apply(lambda x: x[portion_index])\n",
    "\n",
    "    # Create a new column for each part of the split content\n",
    "    #for i in range(len(df[column_index].iloc[0])):\n",
    "     #   df[f'Split_Part_{i + 1}'] = df[column_index].apply(lambda x: x[i] if len(x) > i else \"\")\n",
    "\n",
    "    # Drop the original column containing the split content (optional)\n",
    "    df.drop(column_index, axis=1, inplace=True)\n",
    "\n",
    "    # Print the resulting DataFrame\n",
    "    print(df)\n",
    "    # Save the resulting DataFrame to the output file as a text file\n",
    "    df.to_csv(output_file, sep='\\t', index=False, header=None)  # Use '\\t' as the delimiter for a tab-separated file\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    # Example usage:\n",
    "    input_file = \"/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/updated_extracted_gff_files_from_protein_file.csv\"  # Replace with the path to your input file\n",
    "    output_file = \"modified_gff_file_4_annotation\" \n",
    "    column_index = 8  # Replace with the index of the column containing the cell content (0-based index)\n",
    "    delimiter = \";\"  # Replace with the appropriate delimiter used to split the cell content\n",
    "    portion_index = 3  # Replace with the index of the split portion to retain (0-based index)\n",
    "\n",
    "    split_cell_content(input_file, column_index, delimiter, portion_index)\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "ce9a9b3e",
   "metadata": {},
   "source": [
    "# code to extracted two portions from the splitted column to retains the geneID and protein ID in new columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "bac36000",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 0           1     2          3          4  5  6  7  \\\n",
      "0     chromosome_1  prediction  gene     1381.0     2142.0  0  +  .   \n",
      "1     chromosome_1  prediction  gene     2744.0     3343.0  0  -  .   \n",
      "2     chromosome_1  prediction  gene     3344.0     3817.0  0  -  .   \n",
      "3     chromosome_1  prediction  gene     5330.0     6385.0  0  -  .   \n",
      "4     chromosome_1  prediction  gene    15458.0    25533.0  0  +  .   \n",
      "...            ...         ...   ...        ...        ... .. .. ..   \n",
      "9105  chromosome_7  prediction  gene  4089438.0  4091337.0  0  +  .   \n",
      "9106  chromosome_7  prediction  gene  4095568.0  4097284.0  0  +  .   \n",
      "9107  chromosome_7  prediction  gene  4098122.0  4098907.0  0  -  .   \n",
      "9108  chromosome_7  prediction  gene  4100606.0  4101293.0  0  -  .   \n",
      "9109  chromosome_7  prediction  gene  4101637.0  4103479.0  0  -  .   \n",
      "\n",
      "     First_Fragment    Second_Fragment  \n",
      "0         ID=gene_1  proteinId=2114025  \n",
      "1         ID=gene_2  proteinId=2293935  \n",
      "2         ID=gene_3  proteinId=2293936  \n",
      "3         ID=gene_4  proteinId=2051335  \n",
      "4         ID=gene_5  proteinId=2121898  \n",
      "...             ...                ...  \n",
      "9105  ID=gene_10836   proteinId=103797  \n",
      "9106  ID=gene_10837  proteinId=2071098  \n",
      "9107  ID=gene_10838  proteinId=2114022  \n",
      "9108  ID=gene_10840  proteinId=2114023  \n",
      "9109  ID=gene_10841  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "def split_cell_content(input_file, output_file, column_index, delimiter):\n",
    "    # Read the input table into a pandas DataFrame without header names\n",
    "    df = pd.read_csv(input_file, sep='\\t', header=None)\n",
    "\n",
    "    # Convert the column with float64 values to strings\n",
    "    df[column_index] = df[column_index].astype(str)\n",
    "\n",
    "    # Split the content of the cell using the delimiter\n",
    "    df[column_index] = df[column_index].str.split(delimiter)\n",
    "\n",
    "    # Create new columns for the first and second fragments from the split content\n",
    "    df['First_Fragment'] = df[column_index].apply(lambda x: x[0] if len(x) > 0 else \"\")\n",
    "    df['Second_Fragment'] = df[column_index].apply(lambda x: x[3] if len(x) > 1 else \"\")\n",
    "\n",
    "    # Drop the original column containing the split content\n",
    "    df.drop(column_index, axis=1, inplace=True)\n",
    "    print(df)\n",
    "    # Save the resulting DataFrame to the output file as a text file\n",
    "    df.to_csv(output_file, sep='\\t', index=False, header=None)  # Use '\\t' as the delimiter for a tab-separated file\n",
    "\n",
    "if __name__ == \"__main__\":\n",
    "    # Example usage:\n",
    "    input_file = \"/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/extracted_gff_files_from_protein_file.txt\"  # Replace with the path to your input file\n",
    "    output_file = \"modified_gff_file_4_annotation_2\" \n",
    "    column_index = 8  # Replace with the index of the column containing the cell content (0-based index)\n",
    "    delimiter = \";\"  # Replace with the appropriate delimiter used to split the cell content\n",
    "\n",
    "    split_cell_content(input_file, output_file, column_index, delimiter)\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "6c56b24f",
   "metadata": {},
   "source": [
    "# Update all occurrences of 'chromosome_' to 'chr' in column 2 using apply and lambda function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "efe46aeb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                 0           1     2          3          4  5  6  7  \\\n",
      "0     chromosome_1  prediction  gene     1381.0     2142.0  0  +  .   \n",
      "1     chromosome_1  prediction  gene     2744.0     3343.0  0  -  .   \n",
      "2     chromosome_1  prediction  gene     3344.0     3817.0  0  -  .   \n",
      "3     chromosome_1  prediction  gene     5330.0     6385.0  0  -  .   \n",
      "4     chromosome_1  prediction  gene    15458.0    25533.0  0  +  .   \n",
      "...            ...         ...   ...        ...        ... .. .. ..   \n",
      "9105  chromosome_7  prediction  gene  4089438.0  4091337.0  0  +  .   \n",
      "9106  chromosome_7  prediction  gene  4095568.0  4097284.0  0  +  .   \n",
      "9107  chromosome_7  prediction  gene  4098122.0  4098907.0  0  -  .   \n",
      "9108  chromosome_7  prediction  gene  4100606.0  4101293.0  0  -  .   \n",
      "9109  chromosome_7  prediction  gene  4101637.0  4103479.0  0  -  .   \n",
      "\n",
      "                  8                  9  \n",
      "0         ID=gene_1  proteinId=2114025  \n",
      "1         ID=gene_2  proteinId=2293935  \n",
      "2         ID=gene_3  proteinId=2293936  \n",
      "3         ID=gene_4  proteinId=2051335  \n",
      "4         ID=gene_5  proteinId=2121898  \n",
      "...             ...                ...  \n",
      "9105  ID=gene_10836   proteinId=103797  \n",
      "9106  ID=gene_10837  proteinId=2071098  \n",
      "9107  ID=gene_10838  proteinId=2114022  \n",
      "9108  ID=gene_10840  proteinId=2114023  \n",
      "9109  ID=gene_10841  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "data_frame2 = pd.read_csv('/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/modified_gff_file_4_annotation_2', sep='\\t', header=None)\n",
    "print(data_frame2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "19c2bfba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         0           1    2          3          4  5  6  7              8  \\\n",
      "0     chr1  prediction  CDS     1381.0     2142.0  0  +  .      ID=gene_1   \n",
      "1     chr1  prediction  CDS     2744.0     3343.0  0  -  .      ID=gene_2   \n",
      "2     chr1  prediction  CDS     3344.0     3817.0  0  -  .      ID=gene_3   \n",
      "3     chr1  prediction  CDS     5330.0     6385.0  0  -  .      ID=gene_4   \n",
      "4     chr1  prediction  CDS    15458.0    25533.0  0  +  .      ID=gene_5   \n",
      "...    ...         ...  ...        ...        ... .. .. ..            ...   \n",
      "9105  chr7  prediction  CDS  4089438.0  4091337.0  0  +  .  ID=gene_10836   \n",
      "9106  chr7  prediction  CDS  4095568.0  4097284.0  0  +  .  ID=gene_10837   \n",
      "9107  chr7  prediction  CDS  4098122.0  4098907.0  0  -  .  ID=gene_10838   \n",
      "9108  chr7  prediction  CDS  4100606.0  4101293.0  0  -  .  ID=gene_10840   \n",
      "9109  chr7  prediction  CDS  4101637.0  4103479.0  0  -  .  ID=gene_10841   \n",
      "\n",
      "                      9  \n",
      "0     proteinId=2114025  \n",
      "1     proteinId=2293935  \n",
      "2     proteinId=2293936  \n",
      "3     proteinId=2051335  \n",
      "4     proteinId=2121898  \n",
      "...                 ...  \n",
      "9105   proteinId=103797  \n",
      "9106  proteinId=2071098  \n",
      "9107  proteinId=2114022  \n",
      "9108  proteinId=2114023  \n",
      "9109  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr1' if 'chromosome_1' in str(x).lower() else x)\n",
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr2' if 'chromosome_2' in str(x).lower() else x)\n",
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr3' if 'chromosome_3' in str(x).lower() else x)\n",
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr4' if 'chromosome_4' in str(x).lower() else x)\n",
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr5' if 'chromosome_5' in str(x).lower() else x)\n",
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr6' if 'chromosome_6' in str(x).lower() else x)\n",
    "data_frame2[0] = data_frame2[0].apply(lambda x: 'chr7' if 'chromosome_7' in str(x).lower() else x)\n",
    "data_frame2[2] = data_frame2[2].apply(lambda x: 'CDS' if 'gene' in str(x).lower() else x)\n",
    "data_frame2.to_csv('modified_gff_file_4_annotation_latest', sep='\\t', index=False)\n",
    "print(data_frame2)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "5992fde9",
   "metadata": {},
   "source": [
    "# Code to remove some unwanted columns from the modified_gff_file_4_annotation_latest file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "f5260e4b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         0           1    2          3          4  6              8  \\\n",
      "0     chr1  prediction  CDS     1381.0     2142.0  +      ID=gene_1   \n",
      "1     chr1  prediction  CDS     2744.0     3343.0  -      ID=gene_2   \n",
      "2     chr1  prediction  CDS     3344.0     3817.0  -      ID=gene_3   \n",
      "3     chr1  prediction  CDS     5330.0     6385.0  -      ID=gene_4   \n",
      "4     chr1  prediction  CDS    15458.0    25533.0  +      ID=gene_5   \n",
      "...    ...         ...  ...        ...        ... ..            ...   \n",
      "9105  chr7  prediction  CDS  4089438.0  4091337.0  +  ID=gene_10836   \n",
      "9106  chr7  prediction  CDS  4095568.0  4097284.0  +  ID=gene_10837   \n",
      "9107  chr7  prediction  CDS  4098122.0  4098907.0  -  ID=gene_10838   \n",
      "9108  chr7  prediction  CDS  4100606.0  4101293.0  -  ID=gene_10840   \n",
      "9109  chr7  prediction  CDS  4101637.0  4103479.0  -  ID=gene_10841   \n",
      "\n",
      "                      9  \n",
      "0     proteinId=2114025  \n",
      "1     proteinId=2293935  \n",
      "2     proteinId=2293936  \n",
      "3     proteinId=2051335  \n",
      "4     proteinId=2121898  \n",
      "...                 ...  \n",
      "9105   proteinId=103797  \n",
      "9106  proteinId=2071098  \n",
      "9107  proteinId=2114022  \n",
      "9108  proteinId=2114023  \n",
      "9109  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 8 columns]\n",
      "Columns removed successfully!\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Read your table into a DataFrame\n",
    "# Replace 'your_table.csv' with your actual file name or use other functions like read_excel for different formats\n",
    "data = pd.read_csv('/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/modified_gff_file_4_annotation_latest', sep='\\t')\n",
    "\n",
    "# List of columns you want to remove\n",
    "columns_to_remove = ['5', '7']\n",
    "\n",
    "# Remove the specified columns\n",
    "data = data.drop(columns=columns_to_remove, axis=1)\n",
    "print(data)\n",
    "# Optionally, you can save the modified data back to a file\n",
    "data.to_csv('modified_gff_file_4_annotation_latest1.csv', sep='\\t', index=False)\n",
    "# data.to_excel('modified_table.xlsx', index=False)\n",
    "\n",
    "print(\"Columns removed successfully!\")"
   ]
  },
  {
   "cell_type": "raw",
   "id": "c0b357e6",
   "metadata": {},
   "source": [
    "# To change the names of columns to the right gff format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "9d32b35a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     scaffold      source type      start       stop strand           info  \\\n",
      "0        chr1  prediction  CDS     1381.0     2142.0      +      ID=gene_1   \n",
      "1        chr1  prediction  CDS     2744.0     3343.0      -      ID=gene_2   \n",
      "2        chr1  prediction  CDS     3344.0     3817.0      -      ID=gene_3   \n",
      "3        chr1  prediction  CDS     5330.0     6385.0      -      ID=gene_4   \n",
      "4        chr1  prediction  CDS    15458.0    25533.0      +      ID=gene_5   \n",
      "...       ...         ...  ...        ...        ...    ...            ...   \n",
      "9105     chr7  prediction  CDS  4089438.0  4091337.0      +  ID=gene_10836   \n",
      "9106     chr7  prediction  CDS  4095568.0  4097284.0      +  ID=gene_10837   \n",
      "9107     chr7  prediction  CDS  4098122.0  4098907.0      -  ID=gene_10838   \n",
      "9108     chr7  prediction  CDS  4100606.0  4101293.0      -  ID=gene_10840   \n",
      "9109     chr7  prediction  CDS  4101637.0  4103479.0      -  ID=gene_10841   \n",
      "\n",
      "                     ID  \n",
      "0     proteinId=2114025  \n",
      "1     proteinId=2293935  \n",
      "2     proteinId=2293936  \n",
      "3     proteinId=2051335  \n",
      "4     proteinId=2121898  \n",
      "...                 ...  \n",
      "9105   proteinId=103797  \n",
      "9106  proteinId=2071098  \n",
      "9107  proteinId=2114022  \n",
      "9108  proteinId=2114023  \n",
      "9109  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 8 columns]\n",
      "Columns renamed successfully!\n"
     ]
    }
   ],
   "source": [
    "data1 = pd.read_csv('/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/modified_gff_file_4_annotation_latest1.csv', sep='\\t')\n",
    "\n",
    "# Define a dictionary that maps old column names to new column names\n",
    "column_name_mapping = {\n",
    "    '0': 'scaffold',\n",
    "    '1': 'source',\n",
    "    '2': 'type',\n",
    "    '3': 'start',\n",
    "    '4': 'stop',\n",
    "    '6': 'strand',\n",
    "    '8': 'info',\n",
    "    '9': 'ID'\n",
    "}\n",
    "\n",
    "# Rename the columns using the dictionary\n",
    "data1 = data1.rename(columns=column_name_mapping)\n",
    "print(data1)\n",
    "data1.to_csv('modified_gff_file_finalized_08_08_2023', sep='\\t', index=False)\n",
    "\n",
    "print(\"Columns renamed successfully!\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "0b0cb457",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "data2 = pd.read_csv('/usr2/people/shollyt22/shollyt22/Barseq_July_2023/OORB003_TnSeq/for_analysis/modified_gff_file_finalized_08_08_2023', sep='\\t')\n",
    "#print(data2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "72bb9a7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     scaffold source type      start       stop strand           info  \\\n",
      "0        chr1    JGI  CDS     1381.0     2142.0      +      ID=gene_1   \n",
      "1        chr1    JGI  CDS     2744.0     3343.0      -      ID=gene_2   \n",
      "2        chr1    JGI  CDS     3344.0     3817.0      -      ID=gene_3   \n",
      "3        chr1    JGI  CDS     5330.0     6385.0      -      ID=gene_4   \n",
      "4        chr1    JGI  CDS    15458.0    25533.0      +      ID=gene_5   \n",
      "...       ...    ...  ...        ...        ...    ...            ...   \n",
      "9105     chr7    JGI  CDS  4089438.0  4091337.0      +  ID=gene_10836   \n",
      "9106     chr7    JGI  CDS  4095568.0  4097284.0      +  ID=gene_10837   \n",
      "9107     chr7    JGI  CDS  4098122.0  4098907.0      -  ID=gene_10838   \n",
      "9108     chr7    JGI  CDS  4100606.0  4101293.0      -  ID=gene_10840   \n",
      "9109     chr7    JGI  CDS  4101637.0  4103479.0      -  ID=gene_10841   \n",
      "\n",
      "                     ID  \n",
      "0     proteinId=2114025  \n",
      "1     proteinId=2293935  \n",
      "2     proteinId=2293936  \n",
      "3     proteinId=2051335  \n",
      "4     proteinId=2121898  \n",
      "...                 ...  \n",
      "9105   proteinId=103797  \n",
      "9106  proteinId=2071098  \n",
      "9107  proteinId=2114022  \n",
      "9108  proteinId=2114023  \n",
      "9109  proteinId=2313233  \n",
      "\n",
      "[9110 rows x 8 columns]\n"
     ]
    }
   ],
   "source": [
    "# Update all occurrences of 'prediction' to 'JGI' in column 2 using apply and lambda function\n",
    "data2['source'] = data2['source'].apply(lambda x: 'JGI' if 'prediction' in str(x).lower() else x)\n",
    "# Save the DataFrame to a CSV file. The default delimiter in CSV file is (,) it can be changed \n",
    "data2.to_csv('modified_gff_file_finalized_08_08_2023', sep='\\t', index=False)\n",
    "print(data2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8c6f975c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "solaenv",
   "language": "python",
   "name": "solaenv"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
